---
title: "Izar 2020 PDX DE Analysis"
author: "Jesslyn Goh"
date: "5/28/2020"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_knit$set(root.dir = here::here())
knitr::opts_chunk$set(echo = TRUE)
set.seed(10)
```

```{r init}
# Load packages
library(tidyverse) # ggplot, dplyr, read_r (read_xlsx)
library(Matrix) # interact with sparse data
library(Seurat)
library(patchwork) # package to combine plots
library(glue)
library(beepr)
library(readxl)
library(ggplot2)
library(gt)
library(msigdbr)
library(here)
library(dplyr)
#library(devtools)
# devtools::install_github("immunogenomics/presto")
library(presto)
library(fgsea)


# Read in Seurat objects
PDX = readRDS(file = "data/Izar_2020/jesslyn_PDX_processed.RDS")

#load in a list of hallmark names that we are interested in 
hallmark_names = read_lines("gene_lists/hallmarks.txt")
#load in all hallmarks, category H stands for hallmark geneset 
#filter out genesets we're interested in, and only select for columns with the geneset name and gene symbols
hallmark = msigdbr(species = "Homo sapiens", category = "H") %>% filter(gs_name %in% hallmark_names) %>% select(gs_name, gene_symbol) 
#convert hallmark into a list
hallmark.list = vector(mode = "list")
for(hm in unique(hallmark$gs_name)){
  hallmark.list = c(hallmark.list, select(filter(hallmark, gs_name %in% hm),gene_symbol) %>% as.list())
}
names(hallmark.list) <- unique(hallmark$gs_name)
```

DE ANALYSIS TYPE #1. TEST DIFFERENTIAL EXPRESSION ON A PREDEFINED GO GENESET
Visualize differential expression of each hallmark across models and treatment groups, generally. 
```{r hallmark expression}
#examining whether certain hallmarks are expressed differentially across models and treatment groups

#across models
Idents(PDX) <- "model_ID"
hallmark_names = names(PDX@meta.data)[12:45]
DotPlot(PDX, features = hallmark_names) + coord_flip() + labs(title = "DE Expression of Hallmark Signatures across Models", y = "Model ID", x = "Hallmarks") + theme(plot.title = element_text(hjust = 0.5)) 
ggsave("PDX_hallmark_DEAnalysis_AcrossModels.png", path = "jesslyn_plots/PDX", width = 10, height = 10)

#across treatment conditions
Idents(PDX) <- "treatment.status"
DotPlot(PDX, features = hallmark_names) + coord_flip() + labs(title = "DE Expression of Hallmark Signatures across Treatment Statuses", y = "Treatment Status", x = "Hallmarks") + theme(plot.title = element_text(hjust = 0.5))
ggsave("PDX_hallmark_DEAnalysis_AcrossTreatment.png", path = "jesslyn_plots/PDX", width = 10, height = 10)

#across treatment conditions within each model
models <- levels(PDX$model_ID)
for(model in models){
  obj <- subset(PDX, subset = (model_ID == model))
  obj %>% DotPlot(features = hallmark_names) + coord_flip() + labs(title = glue("DE Expression of Hallmark Signatures across Treatment Statuses for {model} Mouse Model"), y - "Treatment Status", x = "Hallmarks") + theme(plot.title = element_text(hjust = 0.5))
  ggsave(glue("PDX_hallmark_DEAnalysis_AcrossTreatment({model}).png"), path = "jesslyn_plots/PDX", width = 10, height = 10)
}

#SEE EVERYTHING TOGETHER (SPECIFICALLY FOR OXPHOS AND UTR)
#make Vln plots
sigs_of_interest <- c("HALLMARK_OXIDATIVE_PHOSPHORYLATION25", "HALLMARK_UNFOLDED_PROTEIN_RESPONSE33")
Idents(PDX) <- "model_ID"

plots <- VlnPlot(PDX, features = sigs_of_interest, split.by = "treatment.status", combine = F)
plot_names <- c("Oxphos", "UTR")
for (i in c(1,2)){
  name <- plot_names[i]
  plots[[i]] + labs(title = glue("Comparing of {name} Gene Expression Across Treatment Conditions witin Each Model"), x = "Model", fill = "Treatment Status") + theme(plot.title = element_text(size = 6))
  ggsave(glue("PDX_{name}_VlnPlot.png"), path = "jesslyn_plots/PDX")
}

#make DotPlot with mike's code
source(here('mike_code','DotPlot2.R'))
Dotplots = DotPlot2(PDX, features = sigs_of_interest, group1 = "treatment.status", group2 = "model_ID")
Dotplots[[1]] = Dotplots[[1]] + theme(plot.title = element_text(size = 10))
Dotplots[[2]] = Dotplots[[2]] + theme(plot.title = element_text(size = 10))
Dotplots[[1]] + Dotplots[[2]] + plot_layout(guides = 'collect')
ggsave(glue("PDX_SummaryDotPlot.png"), path = "jesslyn_plots/PDX", width = 10, height = 6)
```

Interested in seeing how oxphos and upr genes are differentially expressed across models and treatment conditions. Qualitative results above shows that there is some difference. Now, we want to use quantitative measures to determine how significant this difference is. 
```{r GSEA across treatment statuses}
source(here('jesslyn_code','GSEA_code.R'))
conditions <- levels(PDX$treatment.status)
fgsea_genesets <- hallmark.list[c("HALLMARK_OXIDATIVE_PHOSPHORYLATION", "HALLMARK_UNFOLDED_PROTEIN_RESPONSE")]

for(condition in conditions){
  PDX_condition.results <- GSEA_code(PDX, group.by = "treatment.status", fgsea_genesets, condition) #calculate enrichment score for each gene set you're interested in 
  PDX_condition.ranks <- GSEA_code(PDX, group.by = "treatment.status", fgsea_genesets, condition, ranks = TRUE)
    
    #graph fgsea results for each treatment condition 
    plotEnrichment(fgsea_genesets[["HALLMARK_OXIDATIVE_PHOSPHORYLATION"]], PDX_condition.ranks) + labs(title = glue("Oxidative Phosphorylation GSEA Results For PDX {condition}"))
    ggsave(glue("PDX_All_{condition}_Oxphos_GSEA_EnrichmentPlot.png"), path = "jesslyn_plots/PDX")
    
    plotEnrichment(fgsea_genesets[["HALLMARK_UNFOLDED_PROTEIN_RESPONSE"]], PDX_condition.ranks) + labs(title = glue("Unfolded Protein Response GSEA Results For PDX {condition}"))
    ggsave(glue("PDX_All_{condition}_UTR_GSEA_EnrichmentPlot.png"), path = "jesslyn_plots/PDX")
}

```

```{r GSEA for each treatment condition within each model}
models <- levels(PDX$model_ID)
conditions <- c("vehicle", "MRD", "relapse")

for(model in models){
  #subset Seurat object to contain only one model ID
  obj <- subset(PDX, subset = (model_ID == model))
  
  #fgsea results for each treatment condition (relative to each other) within each model 
  for (condition in conditions){
    obj_condition.results <- GSEA_code(obj, group.by = "treatment.status", fgsea_genesets, condition)
    obj_condition.ranks <- GSEA_code(obj, group.by = "treatment.status", fgsea_genesets, condition, ranks = TRUE)
    
    #graph fgsea results for each treatment condition 
    plotEnrichment(fgsea_genesets[["HALLMARK_OXIDATIVE_PHOSPHORYLATION"]], obj_condition.ranks) + labs(title = glue("Oxidative Phosphorylation GSEA Results For {model} {condition}")) + theme(plot.title = element_text(size = 10))
    ggsave(glue("PDX_{model}_{condition}_Oxphos_GSEA_EnrichmentPlot.png"), path = "jesslyn_plots/PDX")
    
    plotEnrichment(fgsea_genesets[["HALLMARK_UNFOLDED_PROTEIN_RESPONSE"]], obj_condition.ranks) + labs(title = glue("Unfolded Protein Response GSEA Results For {model} {condition}")) + theme(plot.title = element_text(size = 10))
    ggsave(glue("PDX_{model}_{condition}_UTR_GSEA_EnrichmentPlot.png"), path = "jesslyn_plots/PDX")
  }
}
```

DE ANALYSIS TYPE #2. FIND DIFFERENTIALLY EXPRESSED GENES FROM SCRATCH AND EVALUATE IF THEY MAKE UP A GO TERM COLLECTIVELY

```{r DE Analysis between treatment conditions}
source(here('jesslyn_code','DEAnalysis_code.R'))

#DE Analysis between PDX Treatment conditions Vehicle vs. MRD and MRD vs. Relapse
conditions <- c("vehicle", "relapse")
for(condition in conditions){
  #find all DE genes
  DEgenes <- DEAnalysis_code(PDX, group.by= "treatment.status", group.1 = condition, group.2 = "MRD")
  #volcano plot of all genes
  DEAnalysis_code(PDX, markers = DEgenes, group.by = "treatment.status", group.1 = condition, group.2 = "MRD", graph = TRUE)
  ggsave(glue("PDX_{condition}vs.MRD_allDEGenes.png"), path= "jesslyn_plots/PDX", width = 15)
  
  #volcano plots with only oxphos genes (from all DE genes)
  DEAnalysis_code(PDX, markers = DEgenes, group.by = "treatment.status", group.1 = condition, group.2 = "MRD", oxphosGenes = hallmark.list$HALLMARK_OXIDATIVE_PHOSPHORYLATION)
  ggsave(glue("PDX_{condition}vs.MRD_oxphosDEGenes.png"), path= "jesslyn_plots/PDX", width = 15)
  
  #volcano plots with only UTR genes (from all DE genes)
  DEAnalysis_code(PDX, markers = DEgenes, group.by = "treatment.status", group.1 = condition, group.2 = "MRD", utrGenes = hallmark.list$HALLMARK_UNFOLDED_PROTEIN_RESPONSE)
  ggsave(glue("PDX_{condition}vs.MRD_utrDEGenes.png"), path= "jesslyn_plots/PDX", width = 15)
}

beep(3)
```

```{r DE Analysis between treatment conditions within each model}

models <- levels(PDX$model_ID)
conditions <- c("vehicle", "relapse")

for(model in models){
  #subset Seurat object to contain only one model ID
  obj <- subset(PDX, subset = (model_ID == model))
  
  #DE analysis between treatment conditions within each model 
  for (condition in conditions){
    #find all DE genes
    DEgenes <- DEAnalysis_code(obj, group.by= "treatment.status", group.1 = condition, group.2 = "MRD")
    #volcano plot of all genes
    DEAnalysis_code(obj, markers = DEgenes, group.by = "treatment.status", group.1 = condition, group.2 = "MRD", graph = TRUE)
    ggsave(glue("PDX_{model}_{condition}vs.MRD_allDEGenes.png"), path= "jesslyn_plots/PDX", width = 15)
  
    #volcano plots with only oxphos genes (from all DE genes)
    DEAnalysis_code(obj, markers = DEgenes, group.by = "treatment.status", group.1 = condition, group.2 = "MRD", oxphosGenes = hallmark.list$HALLMARK_OXIDATIVE_PHOSPHORYLATION)
    ggsave(glue("PDX_{model}_{condition}vs.MRD_oxphosDEGenes.png"), path= "jesslyn_plots/PDX", width = 15)
  
    #volcano plots with only UTR genes (from all DE genes)
    DEAnalysis_code(obj, markers = DEgenes, group.by = "treatment.status", group.1 = condition, group.2 = "MRD", utrGenes = hallmark.list$HALLMARK_UNFOLDED_PROTEIN_RESPONSE)
    ggsave(glue("PDX_{model}_{condition}vs.MRD_utrDEGenes.png"), path= "jesslyn_plots/PDX", width = 15)
  }
}

```