---
title: "Izar 2020 PDX DE Analysis"
author: "Jesslyn Goh"
date: "5/28/2020"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_knit$set(root.dir = here::here())
knitr::opts_chunk$set(echo = TRUE)
set.seed(1)
```

```{r init}
# Load packages
source(here::here('packages.R'))

# Read in Seurat objects
PDX = readRDS(file = "data/Izar_2020/jesslyn_PDX_processed.RDS")

#load in a list of hallmark names that we are interested in 
hallmark_names = read_lines("gene_lists/hallmarks.txt")
#load updated hallmarks 
hallmark.list <- vector(mode = "list", length = length(hallmark_names))
names(hallmark.list) <- hallmark_names
for(hm in hallmark_names){
  file <- read_lines(glue("hallmarks/{hm}_updated.txt"), skip = 1)
  hallmark.list[[hm]] <- file
}
```

DE ANALYSIS TYPE #1. TEST DIFFERENTIAL EXPRESSION ON A PREDEFINED GO GENESET
Visualize differential expression of each hallmark across models and treatment groups, generally. 
```{r hallmark expression}
#examining whether certain hallmarks are expressed differentially across models and treatment groups

#across models
Idents(PDX) <- "model_ID"
hallmark_names = names(PDX@meta.data)[12:45]
DotPlot(PDX, features = hallmark_names) + coord_flip() + labs(title = "DE Expression of Hallmark Signatures across Models", y = "Model ID", x = "Hallmarks") + theme(plot.title = element_text(hjust = 0.5)) 
ggsave("PDX_hallmark_DEAnalysis_AcrossModels.png", path = "jesslyn_plots/PDX", width = 10, height = 10)

#across treatment conditions
Idents(PDX) <- "treatment.status"
DotPlot(PDX, features = hallmark_names) + coord_flip() + labs(title = "DE Expression of Hallmark Signatures across Treatment Statuses", y = "Treatment Status", x = "Hallmarks") + theme(plot.title = element_text(hjust = 0.5))
ggsave("PDX_hallmark_DEAnalysis_AcrossTreatment.png", path = "jesslyn_plots/PDX", width = 10, height = 10)

#across treatment conditions within each model
models <- levels(PDX$model_ID)
for(model in models){
  obj <- subset(PDX, subset = (model_ID == model))
  obj %>% DotPlot(features = hallmark_names) + coord_flip() + labs(title = glue("DE Expression of Hallmark Signatures across Treatment Statuses for {model} Mouse Model"), y - "Treatment Status", x = "Hallmarks") + theme(plot.title = element_text(hjust = 0.5))
  ggsave(glue("PDX_hallmark_DEAnalysis_AcrossTreatment({model}).png"), path = "jesslyn_plots/PDX", width = 10, height = 10)
}

#SEE EVERYTHING TOGETHER (SPECIFICALLY FOR OXPHOS AND UTR)
#make Vln plots
sigs_of_interest <- c("HALLMARK_OXIDATIVE_PHOSPHORYLATION25", "HALLMARK_UNFOLDED_PROTEIN_RESPONSE33")
Idents(PDX) <- "model_ID"

plots <- VlnPlot(PDX, features = sigs_of_interest, split.by = "treatment.status", combine = F)
plot_names <- c("Oxphos", "UTR")
for (i in c(1,2)){
  name <- plot_names[i]
  plots[[i]]<- plots[[i]] + labs(title = glue("Comparing of {name} Gene Expression Across Treatment Conditions witin Each Model"), x = "Model", fill = "Treatment Status") + theme(plot.title = element_text(size = 6))
  ggsave(glue("PDX_{name}_VlnPlot.png"), path = "jesslyn_plots/PDX")
}

#make DotPlot with mike's code
source(here('mike_code','DotPlot2.R'))
Dotplots = DotPlot2(PDX, features = sigs_of_interest, group1 = "model_ID", group2 = "treatment.status")
Dotplots[[1]] = Dotplots[[1]] + theme(plot.title = element_text(size = 10))
Dotplots[[2]] = Dotplots[[2]] + theme(plot.title = element_text(size = 10))
Dotplots[[1]] + Dotplots[[2]] + plot_layout(guides = 'collect')
ggsave(glue("PDX_SummaryDotPlot.png"), path = "jesslyn_plots/PDX", width = 10, height = 6)
```

Interested in seeing how oxphos and upr genes are differentially expressed across models and treatment conditions. Qualitative results above shows that there is some difference. Now, we want to use quantitative measures to determine how significant this difference is. 

##UNPAIRED GSEA - testing across all three conditions at once
#SPECIFIC FOR OXPHOS AND UTR
```{r GSEA across treatment statuses}
source(here('jesslyn_code','GSEA_code.R'))
conditions <- levels(PDX$treatment.status)
fgsea_genesets <- hallmark.list[c("HALLMARK_OXIDATIVE_PHOSPHORYLATION", "HALLMARK_UNFOLDED_PROTEIN_RESPONSE")]
gs <- c("HALLMARK_OXIDATIVE_PHOSPHORYLATION", "HALLMARK_UNFOLDED_PROTEIN_RESPONSE")

for(condition in conditions){
  PDX_condition.ranks <- GSEA_code(PDX, group.by = "treatment.status", fgsea_genesets, condition)
  
  PDX_condition.results <- GSEA_code(PDX, group.by = "treatment.status", fgsea_genesets, condition, ranks = PDX_condition.ranks) #calculate enrichment score for each gene set you're interested in for this particular condition relative to all other conditions
    
    #graph fgsea results for each hallmark of interest for this particular condition relative to all other condition
  for (geneset in gs){
    plotEnrichment(hallmark.list[[geneset]], PDX_condition.ranks) + labs(title = glue("{geneset} GSEA Results For PDX {condition}")) + theme(plot.title = element_text(size = 8))
    ggsave(glue("PDX_UNPAIRED_All_{condition}_{geneset}_GSEA.png"), path = "jesslyn_plots/PDX/GSEA_unpaired")
    
  }
}

```

```{r GSEA for each treatment condition within each model}
models <- levels(PDX$model_ID)
conditions <- c("vehicle", "MRD", "relapse")

for(model in models){
  #subset Seurat object to contain only one model ID
  obj <- subset(PDX, subset = (model_ID == model))
  
  #fgsea results for each treatment condition (relative to each other) within each model 
  for (condition in conditions){
    obj_condition.ranks <- GSEA_code(obj, group.by = "treatment.status", fgsea_genesets, condition)
    
    obj_condition.results <- GSEA_code(obj, group.by = "treatment.status", fgsea_genesets, condition, ranks = obj_condition.ranks)
    
    #graph fgsea results for each treatment condition 
    for (geneset in gs){
    plotEnrichment(hallmark.list[[geneset]], obj_condition.ranks) + labs(title = glue("{geneset} GSEA Results For {model} {condition}")) + theme(plot.title = element_text(size = 10))
    ggsave(glue("PDX_UNPAIRED_{model}_{condition}_{geneset}_GSEA.png"), path = "jesslyn_plots/PDX/GSEA_unpaired")
    }
  }
}
```

##PAIRED GSEA - testing MRD vs. vehicle, MRD vs. relapse and vehicle vs. relapse separately 
#FOR ALL HALLMARKS IN THE LIST
```{r paired GSEA}
#ACROSS TREATMENT STATUSES
conditions <- c("vehicle", "relapse")
hallmarks_names <- names(hallmark.list)
condition.results <- vector("list", length(conditions))
names(condition.results) <- conditions

for(condition in conditions){
  #calculate the enrichment score for each hallmark specifically for group.1 in comparison to group.2
  condition.ranks <- GSEA_code(PDX, group.by = "treatment.status", hallmark.list, paired = TRUE, group.1 = "MRD", group.2 = condition, ranks = NULL)
  
  condition.results[[condition]] <- GSEA_code(PDX, group.by = "treatment.status", hallmark.list, ranks = condition.ranks) 
  
  #graph fgsea for each hallmark for group.1 in comparison to group.2
  for(hallmark in hallmarks_names){
    plotEnrichment(hallmark.list[[hallmark]], condition.ranks) + labs(title = glue("{hallmark} GSEA results for PDX MRD vs. {condition}")) + theme(plot.title = element_text(size = 8))
    ggsave(glue("PDX_PAIRED_MRDvs.{condition}_{hallmark}_GSEA.png"), path = "jesslyn_plots/PDX/GSEA_paired/All")
  }
}
names(condition.results) <- c("MRD vs. vehicle", "MRD vs. relapse")
condition.results

#WITHIN EACH MODEL 
models <- levels(PDX$model_ID)
obj.results <- vector("list", length(models))
names(obj.results) <- levels(PDX$model_ID)
obj.condition.results <- vector("list", length(conditions))
names(obj.condition.results) <- conditions

#MRD vs. vehicle and MRD vs. relapse
for(model in models){
  #subset Seurat object to contain only one model ID
  obj <- subset(PDX, subset = (model_ID == model))
  
  #within each model, calculate fgsea results for MRD vs. vehicle and MRD vs. relapse
  for (condition in conditions){
    obj.condition.ranks <- GSEA_code(obj, group.by = "treatment.status", hallmark.list, paired = TRUE, group.1 = "MRD", group.2 = condition, ranks = NULL)
  
  obj.condition.results[[condition]] <- GSEA_code(obj, group.by = "treatment.status", hallmark.list, ranks = obj.condition.ranks) 
  
  #graph fgsea for each hallmark for each specified comparison within specified model
    for(hallmark in hallmarks_names){
      plotEnrichment(hallmark.list[[hallmark]], obj.condition.ranks) + labs(title = glue("{hallmark} GSEA results for {model} MRD vs. {condition}")) + theme(plot.title = element_text(size = 8))
      ggsave(glue("{model}_PAIRED_MRDvs.{condition}_{hallmark}_GSEA.png"), path = glue("jesslyn_plots/PDX/GSEA_paired/{model}"))
    }
  }
  obj.results[[model]] <- obj.condition.results #save all gsea results into a nested list
}

#vehicle vs. relapse within each model
models <- levels(PDX$model_ID)
obj.vrresults <- vector("list", length(models))
genesetsOI <- c("HALLMARK_OXIDATIVE_PHOSPHORYLATION", "HALLMARK_UNFOLDED_PROTEIN_RESPONSE")

for (model in models){
  obj <- subset(PDX, subset = (model_ID == model))
  #within each model, calculate GSEA for vehicle vs. relapse 
  obj.ranks <- GSEA_code(obj, group.by = "treatment.status", fgseaGS = hallmark.list, paired = TRUE, group.1 = "vehicle", group.2 = "relapse", ranks = NULL)
  
  obj.vrresults[[model]] <- GSEA_code(obj, group.by = "treatment.status", fgseaGS= hallmark.list, ranks = obj.ranks) 
  
  #graph fgsea results for vehicle vs. relapse (oxphos and UTR)
  for (geneset in genesetsOI){
    plotEnrichment(hallmark.list[[geneset]], obj.ranks) + labs(title = glue("{geneset} GSEA Results For {model} vehicle vs. relapse")) + theme(plot.title = element_text(size = 10))
    ggsave(glue("{model}_vehicleVSrelapse_{geneset}_GSEA.png"), path = "jesslyn_plots/PDX/GSEA_paired")
  }
}
```

# TRYING OUT DIFFERENT STATISTICAL TESTS ON OXPHOS and UTR GENES FOR DF20 MRD VS. VEHICLE
```{r try diff stat tests}
tests <- c("wilcox", "t", "MAST", "LR")
DF20.results <- vector("list", length(tests))
names(DF20.results) <- tests
DF20 <- subset(PDX, subset = (model_ID == "DF20"))

for (test in tests){
  DF20.ranks <- GSEA_code(DF20, stattest = test, group.by = "treatment.status", fgseaGS = hallmark.list, paired = TRUE, group.1 = "MRD", group.2 = "vehicle", ranks = NULL)
  
  DF20.results[[test]] <- GSEA_code(DF20, group.by = "treatment.status", fgseaGS= hallmark.list, ranks = DF20.ranks) 
  
  #graph fgsea results for MRD vs. vehicle (oxphos and UTR)
  for (geneset in genesetsOI){
    plotEnrichment(hallmark.list[[geneset]], DF20.ranks) + labs(title = glue("{geneset} GSEA Results For DF20 MRD vs. vehicle {test}")) + theme(plot.title = element_text(size = 10))
    ggsave(glue("DF20_{test}_MRDVSvehicle_{geneset}_GSEA.png"), path = "jesslyn_plots/PDX/GSEA_paired")
  }
}

#print results 
for(test in tests){
  print(DF20.results[[test]] %>% filter(`pathway` == "HALLMARK_OXIDATIVE_PHOSPHORYLATION"))
  print(DF20.results[[test]] %>% filter(`pathway` == "HALLMARK_UNFOLDED_PROTEIN_RESPONSE"))
}
```

DE ANALYSIS TYPE #2. FIND DIFFERENTIALLY EXPRESSED GENES FROM SCRATCH AND EVALUATE IF THEY MAKE UP A GO TERM COLLECTIVELY

```{r DE Analysis between treatment conditions}
source(here('jesslyn_code','DEAnalysis_code.R'))
genesets <- c("HALLMARK_OXIDATIVE_PHOSPHORYLATION", "HALLMARK_UNFOLDED_PROTEIN_RESPONSE")

#DE Analysis between PDX Treatment conditions Vehicle vs. MRD and MRD vs. Relapse
conditions <- c("vehicle", "relapse")
for(condition in conditions){
  #find all DE genes
  DEgenes <- DEAnalysis_code(PDX, group.by= "treatment.status", group.1 = condition, group.2 = "MRD")
  
  #volcano plot of all genes
  DEAnalysis_code(PDX, markers = DEgenes, group.by = "treatment.status", group.1 = condition, group.2 = "MRD", graph = TRUE)
  ggsave(glue("PDX_{condition}vs.MRD_allDEGenes.png"), path= "jesslyn_plots/PDX/Volcano/All", width = 15)
  
  #volcano plots for specific genesets
  for (geneset in genesets){
  DEAnalysis_code(PDX, markers = DEgenes, group.by = "treatment.status", group.1 = condition, group.2 = "MRD", geneset = hallmark.list[[geneset]]) + labs(title = glue("PDX {condition} vs. MRD DE Genes for {geneset}"))
  ggsave(glue("PDX_{condition}vs.MRD_{geneset}DEGenes.png"), path= "jesslyn_plots/PDX/Volcano/All", width = 15)
  }
}
```

```{r DE Analysis between treatment conditions within each model}

models <- levels(PDX$model_ID)
conditions <- c("vehicle", "relapse")
genesets <- c("HALLMARK_OXIDATIVE_PHOSPHORYLATION", "HALLMARK_UNFOLDED_PROTEIN_RESPONSE")

for(model in models){
  #subset Seurat object to contain only one model ID
  obj <- subset(PDX, subset = (model_ID == model))
  
  #DE analysis between treatment conditions within each model 
  for (condition in conditions){
    #find all DE genes
    DEgenes <- DEAnalysis_code(obj, group.by= "treatment.status", group.1 = condition, group.2 = "MRD")
    #volcano plot of all genes
    DEAnalysis_code(obj, markers = DEgenes, group.by = "treatment.status", group.1 = condition, group.2 = "MRD", graph = TRUE) + labs(title = glue("PDX {model} {condition} vs. MRD All DE Genes"))
    ggsave(glue("PDX_{model}_{condition}vs.MRD_allDEGenes.png"), path= glue( "jesslyn_plots/PDX/Volcano/{model}"), width = 15)
  
    #volcano plots for specific genesets 
    for(geneset in genesets){
    DEAnalysis_code(obj, markers = DEgenes, group.by = "treatment.status", group.1 = condition, group.2 = "MRD", geneset = hallmark.list[[geneset]]) + labs(title = glue("PDX {model} {condition} vs. MRD {geneset} DE Genes"))
    ggsave(glue("PDX_{model}_{condition}vs.MRD_{geneset}DEGenes.png"), path= glue("jesslyn_plots/PDX/Volcano/{model}"), width = 15)
    }
  }
}
```