---
title: "PDX DE Analysis (test version)"
author: "Jesslyn Goh"
date: "6/18/2020"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_knit$set(root.dir = here::here())
knitr::opts_chunk$set(echo = TRUE)
```

# PART 2. PDX DIFFERENTIAL GENE EXPRESSION ANALYSIS
In this R Markdown file, we will examine Differential Expression across **treatment statuses** within each **model** by: 
    1) Type #1 DE Analysis: Finding DE Genes from scratch and evaluate if they make up a GO term collectively using **Volcano Plots**
    2) Type #2 DE Analysis: Visualizing and Quantifying DE on **predefined** GO Genesets through: 
        a) Violin and Dot Plots 
        b) Gene Set Enrichment Analysis (GSEA)

## STEP 1. LOAD IN PROCESSED PDX SEURAT OBJECTS AND HALLMARK GENESETS
```{r init, message= FALSE}
# Load packages
source(here::here('packages.R'))

#Read in PDX RDS object
PDX_All = readRDS("data/Izar_2020/test/jesslyn_PDX_All_processed.RDS")
PDX_DF20 = readRDS("data/Izar_2020/test/jesslyn_PDX_DF20_processed.RDS")
PDX_DF101 = readRDS("data/Izar_2020/test/jesslyn_PDX_DF101_processed.RDS")
PDX_DF68 = readRDS("data/Izar_2020/test/jesslyn_PDX_DF68_processed.RDS")

#Read in hallmarks of interest
hallmark_names = read_lines("gene_lists/hallmarks.txt")
hallmark.list <- vector(mode = "list", length = length(hallmark_names))
names(hallmark.list) <- hallmark_names
for(hm in hallmark_names){
  file <- read_lines(glue("hallmarks/{hm}_updated.txt"), skip = 1)
  hallmark.list[[hm]] <- file
}
```

## STEP 2. DE ANALYSIS TYPE #2. TEST DIFFERENTIAL EXPRESSION ON A PREDEFINED GO GENESET
Visualize differential expression of each hallmark across treatment groups within each model 
```{r DE hallmark visualization}
hms <- c("HALLMARK_OXIDATIVE_PHOSPHORYLATION25", "HALLMARK_UNFOLDED_PROTEIN_RESPONSE33")

#DF20 
VlnPlot(PDX_DF20, features = hms, group.by = "treatment.status") #not centered because AddModuleScore uses normalized but unscaled, uncentered data
#try to scale module score afterwards and reassign it back to the Seurat object, then plot VlnPlot
oxphos.scaled <- scale(PDX_DF20$HALLMARK_OXIDATIVE_PHOSPHORYLATION25)
PDX_DF20 <- AddMetaData(PDX_DF20, oxphos.scaled, col.name = "oxphos.scaled")
VlnPlot(PDX_DF20, features = "oxphos.scaled", group.by = "treatment.status")

#DF101

```