<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jesslyn Goh and Mike Cuoco" />


<title>Izar 2020 PDX (Cohort 3) DE Analysis</title>

<script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.14/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">jesslyn_ovca</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="coding.html">Coding convention</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/mikecuoco/jesslyn_ovca">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Izar 2020 PDX (Cohort 3) DE Analysis</h1>
<h4 class="author">Jesslyn Goh and Mike Cuoco</h4>
<h4 class="date">7/20/2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-08-04
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>jesslyn_ovca/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200713code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200713)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200713code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200713)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjgoh2jesslynovcatree6d23590f4a7c34ac917fb57c39c328d1f4d1dc31targetblank6d23590a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/jgoh2/jesslyn_ovca/tree/6d23590f4a7c34ac917fb57c39c328d1f4d1dc31" target="_blank">6d23590</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomjgoh2jesslynovcatree6d23590f4a7c34ac917fb57c39c328d1f4d1dc31targetblank6d23590a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/jgoh2/jesslyn_ovca/tree/6d23590f4a7c34ac917fb57c39c328d1f4d1dc31" target="_blank">6d23590</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/HTAPP/
    Ignored:    data/Izar_2020/
    Ignored:    data/gene_lists/.DS_Store
    Ignored:    data/gene_lists/extra/.DS_Store
    Ignored:    jesslyn_plots/
    Ignored:    mike_plots/
    Ignored:    old/.DS_Store
    Ignored:    old/edited/.DS_Store
    Ignored:    renv/.DS_Store
    Ignored:    renv/library/
    Ignored:    renv/python/
    Ignored:    renv/staging/
    Ignored:    vignettes/

Unstaged changes:
    Modified:   old/edited/02_Izar2020_SS2_Load_Plots.Rmd
    Modified:   old/edited/PDX_choices.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/03.1_Izar2020_PDX_DEAnalysis.Rmd</code>) and HTML (<code>docs/03.1_Izar2020_PDX_DEAnalysis.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgoh2/jesslyn_ovca/26f64a3c5a9ca82d37db30b427f88739847f3db5/docs/03.1_Izar2020_PDX_DEAnalysis.html" target="_blank">26f64a3</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/eaa900ebf0881bea5dc0b4a7395e3b2b0da3a17e/analysis/03.1_Izar2020_PDX_DEAnalysis.Rmd" target="_blank">eaa900e</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-03
</td>
<td>
workflowr::wflow_publish(files = files)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgoh2/jesslyn_ovca/2dc1bee7caab8ef67312b7d286e864b74e8f3028/docs/03.1_Izar2020_PDX_DEAnalysis.html" target="_blank">2dc1bee</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/284aad41a1a7e82fdd119034dd23aeb385fa2f03/analysis/03.1_Izar2020_PDX_DEAnalysis.Rmd" target="_blank">284aad4</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-31
</td>
<td>
workflowr::wflow_publish(files = files)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/c8bb9fc065ed0d52d9fff97708768a8502747ff6/analysis/03.1_Izar2020_PDX_DEAnalysis.Rmd" target="_blank">c8bb9fc</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-30
</td>
<td>
PDX Exploratory + DE + Cell Cycle Analyses
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgoh2/jesslyn_ovca/c8bb9fc065ed0d52d9fff97708768a8502747ff6/docs/03.1_Izar2020_PDX_DEAnalysis.html" target="_blank">c8bb9fc</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-30
</td>
<td>
PDX Exploratory + DE + Cell Cycle Analyses
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgoh2/jesslyn_ovca/6be6c853e8ff1ecf2ad96e905cf37ee9b0412dda/docs/03.1_Izar2020_PDX_DEAnalysis.html" target="_blank">6be6c85</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/cdd10f957d932d708833a7b496e26c575b9e65fd/analysis/03.1_Izar2020_PDX_DEAnalysis.Rmd" target="_blank">cdd10f9</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-28
</td>
<td>
SS2 DE Analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgoh2/jesslyn_ovca/cdd10f957d932d708833a7b496e26c575b9e65fd/docs/03.1_Izar2020_PDX_DEAnalysis.html" target="_blank">cdd10f9</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-28
</td>
<td>
SS2 DE Analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgoh2/jesslyn_ovca/35c794768b06ac80182d293ea94c867226d20d49/docs/03.1_Izar2020_PDX_DEAnalysis.html" target="_blank">35c7947</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-27
</td>
<td>
SS2 Analysis Part 1 and 2
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/8ca1e011ab3510091358c41f362229573a9741f9/analysis/03.1_Izar2020_PDX_DEAnalysis.Rmd" target="_blank">8ca1e01</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-27
</td>
<td>
PDX analysis edits
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgoh2/jesslyn_ovca/8ca1e011ab3510091358c41f362229573a9741f9/docs/03.1_Izar2020_PDX_DEAnalysis.html" target="_blank">8ca1e01</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-27
</td>
<td>
PDX analysis edits
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgoh2/jesslyn_ovca/f1acd7b63f7d61d83ee574a9d77c21cffeeb5cb9/docs/03.1_Izar2020_PDX_DEAnalysis.html" target="_blank">f1acd7b</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-24
</td>
<td>
Move PDX_choices.Rmd to old
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/bc21d3a4bbfeaaab3c3701621e489ba430369973/analysis/03.1_Izar2020_PDX_DEAnalysis.Rmd" target="_blank">bc21d3a</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-23
</td>
<td>
PDX DE Analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgoh2/jesslyn_ovca/bc21d3a4bbfeaaab3c3701621e489ba430369973/docs/03.1_Izar2020_PDX_DEAnalysis.html" target="_blank">bc21d3a</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-23
</td>
<td>
PDX DE Analysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/e27cfd1903b17888c35d127d87d78c6bed38792f/analysis/03.1_Izar2020_PDX_DEAnalysis.Rmd" target="_blank">e27cfd1</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-22
</td>
<td>
Moved files out of the analysis folder + AddModulescore in read_Izar_2020.R
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/979ae91ff2afca3e9d691aaefb3d31128bc5fc24/analysis/03.1_Izar2020_PDX_DEAnalysis.Rmd" target="_blank">979ae91</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-20
</td>
<td>
Reorganize PDX code and add to the analysis folder
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="izar-2020-pdx-cohort-3-data-differantial-expresison-analysis" class="section level1">
<h1>IZAR 2020 PDX (COHORT 3) DATA DIFFERANTIAL EXPRESISON ANALYSIS</h1>
<div id="overview" class="section level3">
<h3>OVERVIEW</h3>
<ul>
<li>The PDX data from Izar2020 consists of <strong>only Malignant cells</strong> from three HGSOC PDX models derived from patients with different treatment histories were selected for implantation:
<ul>
<li>DF20 (BRCA WT treatment-naive, clinically platinum sensitive)</li>
<li>DF101 (BRCA1 mutant, 2 lines of prior therapy, clinically platinum resistant)</li>
<li>DF68 (BRCA1 mutant, 6 lines of prior therapy, clinically platinum resistant)</li>
</ul></li>
<li>After tumors were established, animals were divided into two groups per model:
<ul>
<li>Vehicle (treated with DMSO)</li>
<li>Carboplatin (treated with IP carboplatin)
<ul>
<li>Carboplatin-treated mice for minimal residual disease (<strong>MRD</strong>) group were harvested for scRNA-seq</li>
<li>The remaining carboplatin-treated mice were harvested at endpoint (<strong>vehicle</strong>)</li>
</ul></li>
</ul></li>
<li>In our 5-part analysis of the Izar 2020 PDX data, we are interested in identifying differentially expressed genes and hallmark genesets between treatment statuses <strong>within</strong> each model.</li>
<li>We split our PDX analysis into three parts:
<ol style="list-style-type: decimal">
<li><strong>Load Data and Create PDX Seurat Object</strong>
<ol style="list-style-type: lower-alpha">
<li>The code to this part of our analysis is in the <strong>read_Izar_2020.R</strong> file in the <strong>code</strong> folder. During this part of our analysis we:
<ol style="list-style-type: lower-roman">
<li>Load in PDX count matrix and Create Seurat Object</li>
<li>Assign Metadata identities including:
<ul>
<li>Mouse ID</li>
<li>Model ID</li>
<li>Treatment Status</li>
</ul></li>
<li>Score cells for <strong>cell cycle</strong> and <strong>hallmark genesets</strong> - <strong>Note</strong>: It does not matter whether we call AddModuleScore before or after subsetting and scaling each model because AddModuleScore uses the <strong>data</strong> slot.</li>
<li>Save Seurat Object</li>
</ol></li>
</ol></li>
<li><strong>Process Data and Exploratory Data Analysis</strong>
<ol style="list-style-type: lower-alpha">
<li>The code to this part of our analysis can be found in the <strong>03_Izar2020_PDX_Load</strong> file in the <strong>old/edited</strong> folder. During this part of our analysis we:
<ol style="list-style-type: lower-roman">
<li>Load in PDX Seurat Object from Part 1 and <strong>subset by model</strong>. Continue analysis separately for each model.</li>
<li>Scale and FindVariableFeatures (prepares data for dimensionality reduction)</li>
<li>Dimensionality Reduction (PCA + UMAP)</li>
<li>Save Seurat Objects</li>
</ol></li>
</ol></li>
<li><strong>Exploratory Data Analysis</strong>
<ol style="list-style-type: lower-alpha">
<li>The code to this part of our analysis can be found in the <strong>03.0_Izar2020_PDX_Exploratory Analysis</strong> file in the <strong>analysis</strong> folder. During this part of our analysis we:
<ol style="list-style-type: lower-roman">
<li>Load in PDX Seurat Object from Part 2. Analyze separately for each model.</li>
<li>Compute summary metrics for PDX data such as:
<ul>
<li>Number of cells per model per treatment</li>
<li>Number of cells per treatment per cell cycle phase</li>
</ul></li>
<li>Visualize how cells separate based on metadata identities via UMAP - Intermodel heterogeneity: How do cells separate by model? - Intramodel heterogeneity:
<ul>
<li>How do cells separate by treament status?</li>
<li>How do cells separate by cell cycle phase?</li>
</ul></li>
</ol></li>
</ol></li>
<li><strong>DE Analysis</strong>
<ol style="list-style-type: lower-alpha">
<li><strong>TYPE #1 DE ANALYSIS</strong>: Visualizing and Quantifying Differentially Expression on <strong>Predefined</strong> GO Genesets
<ol style="list-style-type: lower-roman">
<li>Violin Plots and UMAP</li>
<li>Gene Set Enrichment Analysis (GSEA)</li>
</ol></li>
<li><strong>TYPE #2 DE ANALYSIS</strong>: Finding DE Genes from scratch
<ol style="list-style-type: lower-roman">
<li>Volcano Plots</li>
</ol></li>
</ol></li>
<li><strong>CELL CYCLE ANALYSIS</strong>
<ol style="list-style-type: lower-alpha">
<li><strong>TYPE #1 CELL CYCLE ANALYSIS</strong>: Examine correlation between treatment condition and cell cycle phase</li>
<li>Evaluate the idea that cell cycle might influence expression of signatures</li>
</ol></li>
</ol></li>
</ul>
<p>This is the fourth part of our 5-part analysis of the Izar 2020 PDX (Cohort 3) data.</p>
</div>
<div id="differential-expression-analysis-in-depth-explanation" class="section level3">
<h3>DIFFERENTIAL EXPRESSION ANALYSIS IN-DEPTH EXPLANATION</h3>
<p>We are interested in answering a few questions for our DE Analysis:</p>
<p><strong>DE ANALYSIS #1. Visualizing and Quantifying DE Hallmark Genesets</strong></p>
<ul>
<li><strong>QUESTION</strong> Are modules (<strong>OXPHOS and UPR</strong>) differentially expressed across treatment conditions within each model?
<ul>
<li><strong>APPROACH #1</strong> Violin Plots and UMAP
<ul>
<li>Visualize differences in hallmark score across treatment conditions with:
<ul>
<li>Violin Plot</li>
<li>UMAP by treatment vs. UMAP by hallmark score</li>
</ul></li>
<li>Statistical test: is the difference in hallmark score across treatment conditions statistically significant?</li>
</ul></li>
<li><strong>APPROACH #2</strong> GSEA
<ul>
<li>GSEA enrichment plots for hallmarks of interest between condition.1 vs. condition.2</li>
<li>Rank genes and compute GSEA enrichment scores</li>
<li>Statistical test: how significant are the enrichment scores?</li>
</ul></li>
</ul></li>
</ul>
<p><strong>DE ANALYSIS #2. Identifying Individual DE Genes</strong></p>
</div>
<div id="step-1-load-in-seurat-objects-and-genesets" class="section level3">
<h3>STEP 1 LOAD IN SEURAT OBJECTS AND GENESETS</h3>
<pre class="r"><code># Load packages
source(here::here(&#39;packages.R&#39;))

#Read in PDX RDS object
PDX_All = readRDS(&quot;data/Izar_2020/test/jesslyn_PDX_All_processed.RDS&quot;)
PDX_DF20 = readRDS(&quot;data/Izar_2020/test/jesslyn_PDX_DF20_processed.RDS&quot;)
PDX_DF101 = readRDS(&quot;data/Izar_2020/test/jesslyn_PDX_DF101_processed.RDS&quot;)
PDX_DF68 = readRDS(&quot;data/Izar_2020/test/jesslyn_PDX_DF68_processed.RDS&quot;)

#Read in hallmarks of interest
hallmark_names = read_lines(&quot;data/gene_lists/hallmarks.txt&quot;)
hallmark.list &lt;- vector(mode = &quot;list&quot;, length = length(hallmark_names) + 4)
names(hallmark.list) &lt;- c(hallmark_names, &quot;GO.OXPHOS&quot;, &quot;KEGG.OXPHOS&quot;, &quot;UNUPDATED.OXPHOS&quot;, &quot;UNUPDATED.UPR&quot;)
for(hm in hallmark_names){
  file &lt;- read_lines(glue(&quot;data/gene_lists/hallmarks/{hm}_updated.txt&quot;), skip = 1)
  hallmark.list[[hm]] &lt;- file
}

hallmark.list[[&quot;GO.OXPHOS&quot;]] &lt;- read_lines(&quot;data/gene_lists/extra/GO.OXPHOS.txt&quot;, skip = 1)
hallmark.list[[&quot;KEGG.OXPHOS&quot;]] &lt;- read_lines(&quot;data/gene_lists/extra/KEGG.OXPHOS.txt&quot;, skip = 2)
hallmark.list[[&quot;UNUPDATED.OXPHOS&quot;]] &lt;- read_lines(&quot;data/gene_lists/oxphos.txt&quot;, skip =1)
hallmark.list[[&quot;UNUPDATED.UPR&quot;]] &lt;- read_lines(&quot;data/gene_lists/upr.txt&quot;, skip =1)

#center module and cell cycle scores and reassign to the metadata of each Seurat object
hm.names &lt;- names(PDX_All@meta.data)[9:48]

for(i in hm.names){
    DF20.hm.centered &lt;- scale(PDX_DF20[[i]], center = TRUE, scale = FALSE)
    PDX_DF20 &lt;- AddMetaData(PDX_DF20, DF20.hm.centered, col.name = glue(&quot;{i}.centered&quot;))
    
    DF101.hm.centered &lt;- scale(PDX_DF101[[i]], center = TRUE, scale = FALSE)
    PDX_DF101 &lt;- AddMetaData(PDX_DF101, DF101.hm.centered, col.name = glue(&quot;{i}.centered&quot;))
    
    DF68.hm.centered &lt;- scale(PDX_DF68[[i]], center = TRUE, scale = FALSE)
    PDX_DF68 &lt;- AddMetaData(PDX_DF68, DF68.hm.centered, col.name = glue(&quot;{i}.centered&quot;))
}</code></pre>
</div>
<div id="step-2-determining-oxphos-geneset" class="section level3">
<h3>STEP 2 DETERMINING OXPHOS GENESET</h3>
<ul>
<li>Before doing any DE Analysis on the genesets, we investigate which of the three OXPHOS genesets we found is best for our data. The <strong>OXPHOS</strong> genesets that we test are listed as follows:
<ul>
<li>Unupdated Version of HALLMARK_OXIDATIVE_PHOSPHORYLATION (200 genes) <a href="https://www.gsea-msigdb.org/gsea/msigdb/cards/HALLMARK_OXIDATIVE_PHOSPHORYLATION" class="uri">https://www.gsea-msigdb.org/gsea/msigdb/cards/HALLMARK_OXIDATIVE_PHOSPHORYLATION</a></li>
<li>Updated Version of HALLMARK_OXIDATIVE_PHOSPHORYLATION (200 genes) <a href="https://www.gsea-msigdb.org/gsea/msigdb/cards/HALLMARK_OXIDATIVE_PHOSPHORYLATION" class="uri">https://www.gsea-msigdb.org/gsea/msigdb/cards/HALLMARK_OXIDATIVE_PHOSPHORYLATION</a></li>
<li>GO_OXIDATIVE_PHOSPHORYLATION (144 genes) <a href="https://www.gsea-msigdb.org/gsea/msigdb/cards/GO_OXIDATIVE_PHOSPHORYLATION" class="uri">https://www.gsea-msigdb.org/gsea/msigdb/cards/GO_OXIDATIVE_PHOSPHORYLATION</a></li>
<li>KEGG_OXIDATIVE_PHOSPHORYLATION (131 genes) <a href="https://www.gsea-msigdb.org/gsea/msigdb/cards/KEGG_OXIDATIVE_PHOSPHORYLATION" class="uri">https://www.gsea-msigdb.org/gsea/msigdb/cards/KEGG_OXIDATIVE_PHOSPHORYLATION</a></li>
</ul></li>
<li>We determine which geneset to use by asking these questions:
<ol style="list-style-type: decimal">
<li>How many and which genes are not found in the PDX Seurat Object for each geneset?</li>
<li>What makes the genesets different from each other? Which genes in each geneset are the most DE (or have the higest logFC)? Are they the same?
<ol style="list-style-type: lower-alpha">
<li>Volcano Plot: Call FindMarkers on each model, plot onto Volcano Plot, label cells by each geneset</li>
</ol></li>
<li>Which geneset gives us the most statistically significant results?
<ol style="list-style-type: lower-alpha">
<li>VlnPlot: Plot hallmark scores for each geneset group by treatment status for each model</li>
<li>Label the VlnPlot with the number of genes found in PDX Seurat Object for each OXPHOS geneset</li>
</ol></li>
</ol></li>
</ul>
<p><strong>ANSWERING QUESTION #1: How many and which genes are not found in the PDX Seurat Object for each geneset?</strong></p>
<pre class="r"><code>hm.length.df &lt;- data.frame(
  &quot;UNUPDATED.OXPHOS&quot; = length(hallmark.list[[&quot;UNUPDATED.OXPHOS&quot;]]),
  &quot;HALLMARK.OXPHOS&quot; = length(hallmark.list[[&quot;HALLMARK_OXIDATIVE_PHOSPHORYLATION&quot;]]), 
  &quot;GO.OXPHOS&quot; = length(hallmark.list[[&quot;GO.OXPHOS&quot;]]), 
  &quot;KEGG.OXPHOS&quot; = length(hallmark.list[[&quot;KEGG.OXPHOS&quot;]])
)

Found.df &lt;- data.frame(
  &quot;UNUPDATED.OXPHOS&quot; = sum((hallmark.list[[&quot;UNUPDATED.OXPHOS&quot;]] %in% rownames(PDX_All))),
  &quot;HALLMARK.OXPHOS&quot; = sum((hallmark.list[[&quot;HALLMARK_OXIDATIVE_PHOSPHORYLATION&quot;]] %in% rownames(PDX_All))), 
  &quot;GO.OXPHOS&quot; = sum((hallmark.list[[&quot;GO.OXPHOS&quot;]] %in% rownames(PDX_All))), 
  &quot;KEGG.OXPHOS&quot; = sum((hallmark.list[[&quot;KEGG.OXPHOS&quot;]] %in% rownames(PDX_All)))
)

PFound.df &lt;- data.frame(
  &quot;UNUPDATED.OXPHOS&quot; = (sum((hallmark.list[[&quot;UNUPDATED.OXPHOS&quot;]] %in% rownames(PDX_All)))/length(hallmark.list[[&quot;UNUPDATED.OXPHOS&quot;]]))*100,
  &quot;HALLMARK.OXPHOS&quot; = (sum((hallmark.list[[&quot;HALLMARK_OXIDATIVE_PHOSPHORYLATION&quot;]] %in% rownames(PDX_All)))/length(hallmark.list[[&quot;HALLMARK_OXIDATIVE_PHOSPHORYLATION&quot;]])) * 100, 
  &quot;GO.OXPHOS&quot; = (sum((hallmark.list[[&quot;GO.OXPHOS&quot;]] %in% rownames(PDX_All)))/length(hallmark.list[[&quot;GO.OXPHOS&quot;]]))*100, 
  &quot;KEGG.OXPHOS&quot; = (sum((hallmark.list[[&quot;KEGG.OXPHOS&quot;]] %in% rownames(PDX_All)))/length(hallmark.list[[&quot;KEGG.OXPHOS&quot;]]))*100
)

NA.df &lt;- data.frame(
  &quot;UNUPDATED.OXPHOS&quot; = sum(!(hallmark.list[[&quot;UNUPDATED.OXPHOS&quot;]] %in% rownames(PDX_All))),
  &quot;HALLMARK.OXPHOS&quot; = sum(!(hallmark.list[[&quot;HALLMARK_OXIDATIVE_PHOSPHORYLATION&quot;]] %in% rownames(PDX_All))), 
  &quot;GO.OXPHOS&quot; = sum(!(hallmark.list[[&quot;GO.OXPHOS&quot;]] %in% rownames(PDX_All))), 
  &quot;KEGG.OXPHOS&quot; = sum(!(hallmark.list[[&quot;KEGG.OXPHOS&quot;]] %in% rownames(PDX_All)))
)

all.df &lt;- rbind(hm.length.df, Found.df, PFound.df, NA.df)
rownames(all.df) &lt;- c(&quot;NumGenes&quot;, &quot;Found&quot;, &quot;%Found&quot;, &quot;Not Found&quot;)
all.df[,&quot;GO.OXPHOS&quot;] &lt;- round(all.df[,&quot;GO.OXPHOS&quot;])
all.df[,&quot;KEGG.OXPHOS&quot;] &lt;- round(all.df[,&quot;KEGG.OXPHOS&quot;])
all.df</code></pre>
<pre><code>          UNUPDATED.OXPHOS HALLMARK.OXPHOS GO.OXPHOS KEGG.OXPHOS
NumGenes               200             200       144         131
Found                  184             182       107         101
%Found                  92              91        74          77
Not Found               16              18        37          30</code></pre>
<pre class="r"><code># IDENTIFY GENES THAT ARE NOT FOUND -----------
NA.genes.df &lt;- vector(&quot;list&quot;, length = 4)
names &lt;- c(&quot;UNUPDATED.OXPHOS&quot;, &quot;HALLMARK_OXIDATIVE_PHOSPHORYLATION&quot;, &quot;GO.OXPHOS&quot;, &quot;KEGG.OXPHOS&quot;)
names(NA.genes.df) &lt;- names
for(i in names){
  NA.genes.df[[i]] &lt;- (hallmark.list[[i]])[which(!(hallmark.list[[i]] %in% rownames(PDX_All)))]
}</code></pre>
<ul>
<li><strong>OBSERVATIONS</strong>
<ul>
<li>The <strong>UNUPDATED.OXPHOS</strong> genelist actually has the most genes that are found in the PDX object.</li>
<li>Tried to delete the dash (-) in genes with (“MT-”), and also tried to delete the MT part of the gene name, but those genes remain not to be found in the PDX object.</li>
<li>Since the <strong>UNUPDATED.OXPHOS</strong> genelist works better than the updated version (HALLMARK_OXIDATIVE_PHOSPHORYLATION), we will continue our comparison of genesets using the UNUPDATED version.</li>
</ul></li>
<li>We now wonder:
<ul>
<li>Which OXPHOS genes are the most DE if we call FindMarkers?</li>
<li>Which geneset(s) do the most DE OXPHOS genes belong to?</li>
</ul></li>
</ul>
<p><strong>ANSWERING QUESTION 2: Which genes in each geneset are the most DE? Are they the same?</strong> * Used the <strong>wilcoxon rank sum test</strong> for FindMarkers</p>
<pre class="r"><code>PDXs &lt;- c(PDX_DF20, PDX_DF101, PDX_DF68)
PDX.names &lt;- c(&quot;DF20&quot;, &quot;DF101&quot;, &quot;DF68&quot;)
oxphos.hm &lt;- c(&quot;UNUPDATED.OXPHOS&quot;, &quot;GO.OXPHOS&quot;, &quot;KEGG.OXPHOS&quot;)
PDX.hm.plots &lt;- vector(&quot;list&quot;, length = 3)
names(PDX.hm.plots) &lt;- PDX.names

markers &lt;- vector(&quot;list&quot;, length = 3)
names(markers) &lt;- PDX.names

markers[[&quot;DF20&quot;]] &lt;- FindMarkers(PDX_DF20, group.by = &quot;treatment.status&quot;, ident.1 = &quot;MRD&quot;, ident.2 = &quot;vehicle&quot;, test.use = &quot;wilcox&quot;, logfc.threshold = 0) 
markers[[&quot;DF101&quot;]]  &lt;- FindMarkers(PDX_DF101, group.by = &quot;treatment.status&quot;, ident.1 = &quot;MRD&quot;, ident.2 = &quot;vehicle&quot;, test.use = &quot;wilcox&quot;, logfc.threshold = 0)
markers[[&quot;DF68&quot;]] &lt;- FindMarkers(PDX_DF68, group.by = &quot;treatment.status&quot;, ident.1 = &quot;MRD&quot;, ident.2 = &quot;vehicle&quot;, test.use = &quot;wilcox&quot;, logfc.threshold = 0)

for(i in 1:length(PDXs)){
    PDX &lt;- PDX.names[[i]]
    DF.hm.plot &lt;- vector(&quot;list&quot;, length = 3)
    names(DF.hm.plot) &lt;- oxphos.hm
    marker &lt;- markers[[PDX]]
    
  for(oxphos in oxphos.hm){
    
    avgLFC &lt;- rownames(marker[which(abs(marker$avg_logFC) &gt; 0.5),])
    keyvals &lt;- ifelse(!(rownames(marker) %in% hallmark.list[[oxphos]]), &#39;black&#39;, 
                      ifelse(abs(marker$avg_logFC) &gt; 0.5, &#39;red&#39;, &#39;grey&#39;))
    names(keyvals)[keyvals == &#39;red&#39;] &lt;- &#39;OXPHOS &amp; logFC&#39;
    names(keyvals)[keyvals == &#39;grey&#39;] &lt;- &#39;OXPHOS x logFC&#39;
    names(keyvals)[keyvals == &#39;black&#39;] &lt;- &#39;Not OXPHOS&#39;
    found = length(avgLFC[which(avgLFC %in% hallmark.list[[oxphos]])])
    
    p &lt;- EnhancedVolcano(marker, 
           lab = rownames(marker),
           selectLab = avgLFC[which(avgLFC %in% hallmark.list[[oxphos]])],
           labCol = &quot;red&quot;,
           x=&#39;avg_logFC&#39;, y=&#39;p_val_adj&#39;, pCutoff = 0.05, 
           FCcutoff = 0.5, 
           colCustom = keyvals,
           pointSize = c(ifelse(rownames(marker) %in% hallmark.list[[oxphos]], 2.5,1)), 
           drawConnectors = TRUE,
           boxedLabels = TRUE,
           labvjust = 1,
           title= glue(&quot;{PDX.names[[i]]} MRD vs. vehicle&quot;), subtitle= &quot;LogFC cutoff: 0.5, p cutoff: 0.05&quot;,
           caption = glue(&quot;{oxphos}: {found} high LFC oxphos genes found&quot;)
           )
    DF.hm.plot[[oxphos]] &lt;- p
  }
  
  PDX.hm.plots[[PDX]] &lt;- DF.hm.plot[[&quot;UNUPDATED.OXPHOS&quot;]] + DF.hm.plot[[&quot;GO.OXPHOS&quot;]] + DF.hm.plot[[&quot;KEGG.OXPHOS&quot;]]
}

PDX.hm.plots[[&quot;DF20&quot;]]</code></pre>
<p><img src="figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/oxphosVolcano-1.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-oxphosVolcano-1">
Past versions of oxphosVolcano-1.png
</button>
</p>
<div id="fig-oxphosVolcano-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/26f64a3c5a9ca82d37db30b427f88739847f3db5/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/oxphosVolcano-1.png" target="_blank">26f64a3</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/2dc1bee7caab8ef67312b7d286e864b74e8f3028/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/oxphosVolcano-1.png" target="_blank">2dc1bee</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-31
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/c8bb9fc065ed0d52d9fff97708768a8502747ff6/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/oxphosVolcano-1.png" target="_blank">c8bb9fc</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>PDX.hm.plots[[&quot;DF101&quot;]]</code></pre>
<p><img src="figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/oxphosVolcano-2.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-oxphosVolcano-2">
Past versions of oxphosVolcano-2.png
</button>
</p>
<div id="fig-oxphosVolcano-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/26f64a3c5a9ca82d37db30b427f88739847f3db5/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/oxphosVolcano-2.png" target="_blank">26f64a3</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/2dc1bee7caab8ef67312b7d286e864b74e8f3028/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/oxphosVolcano-2.png" target="_blank">2dc1bee</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-31
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/c8bb9fc065ed0d52d9fff97708768a8502747ff6/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/oxphosVolcano-2.png" target="_blank">c8bb9fc</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>PDX.hm.plots[[&quot;DF68&quot;]]</code></pre>
<p><img src="figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/oxphosVolcano-3.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-oxphosVolcano-3">
Past versions of oxphosVolcano-3.png
</button>
</p>
<div id="fig-oxphosVolcano-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/26f64a3c5a9ca82d37db30b427f88739847f3db5/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/oxphosVolcano-3.png" target="_blank">26f64a3</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/2dc1bee7caab8ef67312b7d286e864b74e8f3028/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/oxphosVolcano-3.png" target="_blank">2dc1bee</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-31
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/c8bb9fc065ed0d52d9fff97708768a8502747ff6/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/oxphosVolcano-3.png" target="_blank">c8bb9fc</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># number of DE oxphos genes found in each geneset within each model -----------------
DF20.de.df &lt;- data.frame(
  &quot;UNUPDATED.OXPHOS&quot; = sum(rownames(markers[[&quot;DF20&quot;]][which(abs(markers[[&quot;DF20&quot;]]$avg_logFC) &gt; 0.5),]) %in% hallmark.list[[&quot;UNUPDATED.OXPHOS&quot;]]),
  &quot;GO.OXPHOS&quot; = sum(rownames(markers[[&quot;DF20&quot;]][which(abs(markers[[&quot;DF20&quot;]]$avg_logFC) &gt; 0.5),]) %in% hallmark.list[[&quot;GO.OXPHOS&quot;]]),
  &quot;KEGG.OXPHOS&quot; = sum(rownames(markers[[&quot;DF20&quot;]][which(abs(markers[[&quot;DF20&quot;]]$avg_logFC) &gt; 0.5),]) %in% hallmark.list[[&quot;KEGG.OXPHOS&quot;]])
)

DF101.de.df &lt;- data.frame(
  &quot;UNUPDATED.OXPHOS&quot; = sum(rownames(markers[[&quot;DF101&quot;]][which(abs(markers[[&quot;DF101&quot;]]$avg_logFC) &gt; 0.5),]) %in% hallmark.list[[&quot;UNUPDATED.OXPHOS&quot;]]),
  &quot;GO.OXPHOS&quot; = sum(rownames(markers[[&quot;DF101&quot;]][which(abs(markers[[&quot;DF101&quot;]]$avg_logFC) &gt; 0.5),]) %in% hallmark.list[[&quot;GO.OXPHOS&quot;]]),
  &quot;KEGG.OXPHOS&quot; = sum(rownames(markers[[&quot;DF101&quot;]][which(abs(markers[[&quot;DF101&quot;]]$avg_logFC) &gt; 0.5),]) %in% hallmark.list[[&quot;KEGG.OXPHOS&quot;]])
)

DF68.de.df &lt;- data.frame(
  &quot;UNUPDATED.OXPHOS&quot; = sum(rownames(markers[[&quot;DF68&quot;]][which(abs(markers[[&quot;DF68&quot;]]$avg_logFC) &gt; 0.5),]) %in% hallmark.list[[&quot;UNUPDATED.OXPHOS&quot;]]),
  &quot;GO.OXPHOS&quot; = sum(rownames(markers[[&quot;DF68&quot;]][which(abs(markers[[&quot;DF68&quot;]]$avg_logFC) &gt; 0.5),]) %in% hallmark.list[[&quot;GO.OXPHOS&quot;]]),
  &quot;KEGG.OXPHOS&quot; = sum(rownames(markers[[&quot;DF68&quot;]][which(abs(markers[[&quot;DF68&quot;]]$avg_logFC) &gt; 0.5),]) %in% hallmark.list[[&quot;KEGG.OXPHOS&quot;]])
)

all.de.df &lt;- rbind(DF20.de.df, DF101.de.df, DF68.de.df)
rownames(all.de.df) &lt;- c(&quot;DF20.MRDvVehicle&quot;, &quot;DF101.MRDvVehicle&quot;, &quot;DF68.MRDvVehicle&quot;)
all.de.df</code></pre>
<pre><code>                  UNUPDATED.OXPHOS GO.OXPHOS KEGG.OXPHOS
DF20.MRDvVehicle                24        15           9
DF101.MRDvVehicle               65        50          38
DF68.MRDvVehicle                47        25          20</code></pre>
<pre class="r"><code>percent.logFC &lt;- data.frame(
  &quot;UNUPDATED.OXPHOS&quot; = round(all.de.df[, &quot;UNUPDATED.OXPHOS&quot;]/all.df[&quot;Found&quot;, &quot;UNUPDATED.OXPHOS&quot;]*100, 2), 
  &quot;GO.OXPHOS&quot; = round(all.de.df[, &quot;GO.OXPHOS&quot;]/all.df[&quot;Found&quot;, &quot;GO.OXPHOS&quot;]*100, 2), 
  &quot;KEGG.OXPHOS&quot; = round(all.de.df[, &quot;KEGG.OXPHOS&quot;]/all.df[&quot;Found&quot;, &quot;KEGG.OXPHOS&quot;]*100, 2)
)
rownames(percent.logFC) &lt;- c(&quot;DF20 %Found highLFC&quot;, &quot;DF101 %Found highLFC&quot;, &quot;DF68 %Found highLFC&quot;)
rbind(all.df[,names(all.df) != &quot;HALLMARK.OXPHOS&quot;], percent.logFC)</code></pre>
<pre><code>                     UNUPDATED.OXPHOS GO.OXPHOS KEGG.OXPHOS
NumGenes                       200.00    144.00      131.00
Found                          184.00    107.00      101.00
%Found                          92.00     74.00       77.00
Not Found                       16.00     37.00       30.00
DF20 %Found highLFC             13.04     14.02        8.91
DF101 %Found highLFC            35.33     46.73       37.62
DF68 %Found highLFC             25.54     23.36       19.80</code></pre>
<ul>
<li><strong>OBSERVATIONS</strong>
<ul>
<li><strong>UNUPDATED.OXPHOS</strong> geneset consistently has the most OXPHOS genes with <strong>high logFC</strong> (logFC &gt; 0.5, regardless of padj) relative to the other two genesets for each model comparison between MRD and vehicle. However, <strong>GO.OXPHOS</strong> seems to have the highest percentage of OXPHOS genes with high logFC out of the OXPHOS genes Found on average.</li>
<li>Although it seems that there are a few overlaps of OXPHOS genes found, it is hard to visualize which ones overlap, and whether the ones that overlap are among the top genes with high logFC (regardless of padj). We therefore create tables for each model, extract the top 5 high logFC genes from each geneset and compare if they’re the same.</li>
</ul></li>
</ul>
<pre class="r"><code>DF20.top5 &lt;- markers[[&quot;DF20&quot;]] %&gt;% arrange(-abs(avg_logFC))
DF101.top5 &lt;- markers[[&quot;DF101&quot;]] %&gt;% arrange(-abs(avg_logFC))
DF68.top5 &lt;- markers[[&quot;DF68&quot;]] %&gt;% arrange(-abs(avg_logFC))

DF20.gs.df &lt;- data.frame(
  &quot;UNUPDATED.OXPHOS&quot; = head(rownames(DF20.top5)[which(rownames(DF20.top5) %in% hallmark.list[[&quot;UNUPDATED.OXPHOS&quot;]])], 5), 
  &quot;UNUPDATED.OXPHOS&quot; = select(DF20.top5[head(rownames(DF20.top5)[which(rownames(DF20.top5) %in% hallmark.list[[&quot;UNUPDATED.OXPHOS&quot;]])], 5),], avg_logFC),
  &quot;GO.OXPHOS&quot; = head(rownames(DF20.top5)[which(rownames(DF20.top5) %in% hallmark.list[[&quot;GO.OXPHOS&quot;]])], 5),
  &quot;GO.OXPHOS&quot; = select(DF20.top5[head(rownames(DF20.top5)[which(rownames(DF20.top5) %in% hallmark.list[[&quot;GO.OXPHOS&quot;]])], 5),], avg_logFC),
  &quot;KEGG.OXPHOS&quot; = head(rownames(DF20.top5)[which(rownames(DF20.top5) %in% hallmark.list[[&quot;KEGG.OXPHOS&quot;]])], 5), 
  &quot;KEGG.OXPHOS&quot; = select(DF20.top5[head(rownames(DF20.top5)[which(rownames(DF20.top5) %in% hallmark.list[[&quot;KEGG.OXPHOS&quot;]])], 5),], avg_logFC)
)
rownames(DF20.gs.df) &lt;- seq(from = 1, length = nrow(DF20.gs.df))

DF101.gs.df &lt;- data.frame(
  &quot;UNUPDATED.OXPHOS&quot; = head(rownames(DF101.top5)[which(rownames(DF101.top5) %in% hallmark.list[[&quot;UNUPDATED.OXPHOS&quot;]])], 5), 
  &quot;UNUPDATED.OXPHOS&quot; = select(DF101.top5[head(rownames(DF101.top5)[which(rownames(DF101.top5) %in% hallmark.list[[&quot;UNUPDATED.OXPHOS&quot;]])], 5),], avg_logFC),
  &quot;GO.OXPHOS&quot; = head(rownames(DF101.top5)[which(rownames(DF101.top5) %in% hallmark.list[[&quot;GO.OXPHOS&quot;]])], 5),
  &quot;GO.OXPHOS&quot; = select(DF101.top5[head(rownames(DF101.top5)[which(rownames(DF101.top5) %in% hallmark.list[[&quot;GO.OXPHOS&quot;]])], 5),], avg_logFC),
  &quot;KEGG.OXPHOS&quot; = head(rownames(DF101.top5)[which(rownames(DF101.top5) %in% hallmark.list[[&quot;KEGG.OXPHOS&quot;]])], 5), 
  &quot;KEGG.OXPHOS&quot; = select(DF101.top5[head(rownames(DF101.top5)[which(rownames(DF101.top5) %in% hallmark.list[[&quot;KEGG.OXPHOS&quot;]])], 5),], avg_logFC)
)
rownames(DF101.gs.df) &lt;- seq(from = 1, length = nrow(DF20.gs.df))


DF68.gs.df &lt;- data.frame(
  &quot;UNUPDATED.OXPHOS&quot; = head(rownames(DF68.top5)[which(rownames(DF68.top5) %in% hallmark.list[[&quot;UNUPDATED.OXPHOS&quot;]])], 5), 
  &quot;UNUPDATED.OXPHOS&quot; = select(DF68.top5[head(rownames(DF68.top5)[which(rownames(DF68.top5) %in% hallmark.list[[&quot;UNUPDATED.OXPHOS&quot;]])], 5),], avg_logFC),
  &quot;GO.OXPHOS&quot; = head(rownames(DF68.top5)[which(rownames(DF68.top5) %in% hallmark.list[[&quot;GO.OXPHOS&quot;]])], 5),
  &quot;GO.OXPHOS&quot; = select(DF68.top5[head(rownames(DF68.top5)[which(rownames(DF68.top5) %in% hallmark.list[[&quot;GO.OXPHOS&quot;]])], 5),], avg_logFC),
  &quot;KEGG.OXPHOS&quot; = head(rownames(DF68.top5)[which(rownames(DF68.top5) %in% hallmark.list[[&quot;KEGG.OXPHOS&quot;]])], 5), 
  &quot;KEGG.OXPHOS&quot; = select(DF68.top5[head(rownames(DF68.top5)[which(rownames(DF68.top5) %in% hallmark.list[[&quot;KEGG.OXPHOS&quot;]])], 5),], avg_logFC)
)
rownames(DF68.gs.df) &lt;- seq(from = 1, length = nrow(DF20.gs.df))

DF20.gs.df</code></pre>
<pre><code>  UNUPDATED.OXPHOS  avg_logFC GO.OXPHOS avg_logFC.1 KEGG.OXPHOS avg_logFC.2
1           TIMM10  1.1454854     NUPR1   1.9607774    NDUFA4L2  -1.0646767
2             IDH1  1.0249810     SURF1  -0.9575538      NDUFB3   0.7948077
3            SURF1 -0.9575538      TEFM   0.9397055       COX10  -0.6567115
4           ACADVL -0.9479864     CCNB1   0.9113314       COX15  -0.6314722
5           MRPL35 -0.8855315    NDUFB3   0.7948077       COX5A   0.6066344</code></pre>
<pre class="r"><code>DF101.gs.df</code></pre>
<pre><code>  UNUPDATED.OXPHOS avg_logFC GO.OXPHOS avg_logFC.1 KEGG.OXPHOS avg_logFC.2
1          ATP6V0C  2.803572    NDUFA9  -1.0849423     ATP6V0C   2.8035722
2              BAX  1.723575      SDHD  -1.0338880      NDUFA9  -1.0849423
3             ECI1 -1.563555   NDUFA12  -0.9821144        SDHD  -1.0338880
4             OPA1  1.259500    NDUFB1  -0.9460268      NDUFB1  -0.9460268
5          ALDH6A1  1.184952     COX7B  -0.9355418       COX7B  -0.9355418</code></pre>
<pre class="r"><code>DF68.gs.df</code></pre>
<pre><code>  UNUPDATED.OXPHOS avg_logFC GO.OXPHOS avg_logFC.1 KEGG.OXPHOS avg_logFC.2
1             DLST -1.687756     SURF1   1.5060351    ATP6V1G2   -1.518730
2            SURF1  1.506035     PINK1   1.1743957     ATP6V0C   -1.376521
3          ATP6V0C -1.376521   NDUFAF1   1.0715203    ATP6V1B2   -1.226365
4            TIMM9 -1.204707    NDUFB3   1.0008525      TCIRG1    1.057882
5           TCIRG1  1.057882     COX15   0.8470823      NDUFB3    1.000852</code></pre>
<ul>
<li><strong>OBSERVATIONS</strong>
<ul>
<li>Note: All of these top 5 high logFC OXPHOS genes (logFC &gt; 0.5) have a padj of 1.00</li>
<li>There are not a lot of overlaps of high logFC OXPHOS genes across all three genesets for all three models</li>
<li>Seems like a trend where the top high logFC OXPHOS genes present within the <strong>UNUPDATED OXPHOS</strong> geneset have higher logFC values in comparison to the other two genesets.</li>
</ul></li>
<li>Our answers to <strong>QUESTION #1 and #2</strong> seem to suggest that <strong>UNUPDATED.OXPHOS</strong> hallmark geneset is the best our of all the genesets tested because:
<ol style="list-style-type: decimal">
<li>It has the least number of genes that are not found within the PDX Seurat Object, and the highest percentage of genes that are found.</li>
<li>It has the most number of high logFC OXPHOS genes (logFC &gt; 0.5, regardless of padj) relative to the other two genesets for each model comparison between MRD and vehicle</li>
<li>The high logFC OXPHOS genes present within this geneset have relatively higher logFC values (they are more DE, regardless of padj) in comparison to the high logFC OXPHOS genes present in the other two genesets.</li>
</ol></li>
<li>We answer our 3rd question to confirm whether the <strong>UNUPDATED.OXPHOS</strong> geneset is indeed the most appropriate one for our data relative to the other two genesets.</li>
</ul>
<p><strong>ANSWERING QUESTION #3: which geneset gives us the most statistically significant results</strong></p>
<pre class="r"><code>oxphos.centered &lt;- c(&quot;UNUPDATED.OXPHOS37.centered&quot;, &quot;GO.OXPHOS35.centered&quot;, &quot;KEGG.OXPHOS36.centered&quot;)
Oxphos.Vln.plots &lt;- vector(&quot;list&quot;, length(PDXs))
names(Oxphos.Vln.plots) &lt;- PDX.names

for (i in 1:length(PDXs)){
  obj &lt;- PDXs[[i]]
  name &lt;- PDX.names[[i]]
  numCells &lt;- nrow(PDXs[[i]]@meta.data)
  
  my_comparisons &lt;- list(
    c(&quot;MRD&quot;, &quot;vehicle&quot;),
    c(&quot;MRD&quot;, &quot;relapse&quot;), 
    c(&quot;vehicle&quot;, &quot;relapse&quot;)
  )
  
if(name == &quot;DF68&quot;){
  p &lt;- VlnPlot(obj, features = oxphos.centered, group.by = &quot;treatment.status&quot;, pt.size = 0, cols = c(&quot;#00AFBB&quot;, &quot;#E7B800&quot;, &quot;#FC4E07&quot;), combine = F, y.max = 1.5)
}
if(name == &quot;DF101&quot;){
   p &lt;- VlnPlot(obj, features = oxphos.centered, group.by = &quot;treatment.status&quot;, pt.size = 0, cols = c(&quot;#00AFBB&quot;, &quot;#E7B800&quot;, &quot;#FC4E07&quot;), combine = F, y.max = 1.8)
}
else{
  p &lt;- VlnPlot(obj, features = oxphos.centered, group.by = &quot;treatment.status&quot;, pt.size = 0, cols = c(&quot;#00AFBB&quot;, &quot;#E7B800&quot;, &quot;#FC4E07&quot;), combine = F, y.max = 2.0)
}
  
  unupdated.found &lt;- sum(hallmark.list[[&quot;UNUPDATED.OXPHOS&quot;]] %in% rownames(obj))
  unupdated.length &lt;- length(hallmark.list[[&quot;UNUPDATED.OXPHOS&quot;]])
  unupdated.pFound &lt;- round((unupdated.found / unupdated.length)*100, 2)
  
  p[[1]] &lt;- p[[1]] + labs(title = glue(&quot;{name} UNUPDATED_OXPHOS scores across treatment&quot;), x = name, subtitle = glue(&quot;{numCells} Malignant Cells, {unupdated.found} out of {unupdated.length} OXPHOS genes found ({unupdated.pFound}%)&quot;)) + 
    theme(plot.title = element_text(size = 12), plot.caption = element_text(size = 10)) + 
    geom_boxplot(width = 0.15, position = position_dodge(0.9), alpha = 0.3, show.legend = F) + 
    geom_text(label = paste(sum(obj$treatment.status == &quot;vehicle&quot;), &quot;cells&quot;), x = &quot;vehicle&quot;, y = min(obj$UNUPDATED.OXPHOS37.centered) -0.03) + 
    geom_text(label = paste(sum(obj$treatment.status == &quot;MRD&quot;), &quot;cells&quot;), x = &quot;MRD&quot;, y = min(obj$UNUPDATED.OXPHOS37.centered) - 0.03) + 
    geom_text(label = paste(sum(obj$treatment.status == &quot;relapse&quot;), &quot;cells&quot;), x = &quot;relapse&quot;, y = min(obj$UNUPDATED.OXPHOS37.centered) - 0.03) + 
    stat_compare_means(comparisons = my_comparisons, method = &quot;wilcox.test&quot;, label = &quot;p.format&quot;, step.increase = 0.06) + 
    stat_compare_means(comparisons = my_comparisons, method = &quot;wilcox.test&quot;, label = &quot;p.signif&quot;, step.increase = 0.06, bracket.size = 0, vjust = 1.8)
  
  go.found &lt;- sum(hallmark.list[[&quot;GO.OXPHOS&quot;]] %in% rownames(obj))
  go.length &lt;- length(hallmark.list[[&quot;GO.OXPHOS&quot;]])
  go.pFound &lt;- round((go.found / go.length)*100, 2)
  p[[2]] &lt;- p[[2]] + labs(title = glue(&quot;{name} GO_OXPHOS scores across treatment&quot;), x = name, subtitle = glue(&quot;{numCells} Malignant Cells, {go.found} out of {go.length} OXPHOS genes found ({go.pFound}%)&quot;)) +
    theme(plot.title = element_text(size = 12), plot.caption = element_text(size = 10)) + 
    geom_boxplot(width = 0.15, position = position_dodge(0.9), alpha = 0.3, show.legend = F) + 
    geom_text(label = paste(sum(obj$treatment.status == &quot;vehicle&quot;), &quot;cells&quot;), x = &quot;vehicle&quot;, y = min(obj$GO.OXPHOS35.centered) -0.03) + 
    geom_text(label = paste(sum(obj$treatment.status == &quot;MRD&quot;), &quot;cells&quot;), x = &quot;MRD&quot;, y = min(obj$GO.OXPHOS35.centered) - 0.03) + 
    geom_text(label = paste(sum(obj$treatment.status == &quot;relapse&quot;), &quot;cells&quot;), x = &quot;relapse&quot;, y = min(obj$GO.OXPHOS35.centered) - 0.03) +
    stat_compare_means(comparisons = my_comparisons, method = &quot;wilcox.test&quot;, label = &quot;p.format&quot;, step.increase = 0.06) + 
    stat_compare_means(comparisons = my_comparisons, method = &quot;wilcox.test&quot;, label = &quot;p.signif&quot;, step.increase = 0.06, bracket.size = 0, vjust = 1.8)
  
  kegg.found &lt;- sum(hallmark.list[[&quot;KEGG.OXPHOS&quot;]] %in% rownames(obj))
  kegg.length &lt;- length(hallmark.list[[&quot;KEGG.OXPHOS&quot;]])
  kegg.pFound &lt;- round((kegg.found / kegg.length)*100, 2)
  p[[3]] &lt;- p[[3]] + labs(title = glue(&quot;{name} KEGG_OXPHOS scores across treatment&quot;), x = name, subtitle = glue(&quot;{numCells} Malignant Cells, {kegg.found} out of {kegg.length} OXPHOS genes found ({kegg.pFound}%)&quot;)) +
    theme(plot.title = element_text(size = 12), plot.caption = element_text(size = 10)) + 
    geom_boxplot(width = 0.15, position = position_dodge(0.9), alpha = 0.3, show.legend = F) + 
    geom_text(label = paste(sum(obj$treatment.status == &quot;vehicle&quot;), &quot;cells&quot;), x = &quot;vehicle&quot;, y = min(obj$KEGG.OXPHOS36.centered) -0.03) + 
    geom_text(label = paste(sum(obj$treatment.status == &quot;MRD&quot;), &quot;cells&quot;), x = &quot;MRD&quot;, y = min(obj$KEGG.OXPHOS36.centered) - 0.03) + 
    geom_text(label = paste(sum(obj$treatment.status == &quot;relapse&quot;), &quot;cells&quot;), x = &quot;relapse&quot;, y = min(obj$KEGG.OXPHOS36.centered) - 0.03) +
    stat_compare_means(comparisons = my_comparisons, method = &quot;wilcox.test&quot;, label = &quot;p.format&quot;, step.increase = 0.06) + 
    stat_compare_means(comparisons = my_comparisons, method = &quot;wilcox.test&quot;, label = &quot;p.signif&quot;, step.increase = 0.06, bracket.size = 0, vjust = 1.8)
  
  p &lt;- p[[1]] + p[[2]] + p[[3]] + plot_layout(guides= &#39;collect&#39;)
  
  Oxphos.Vln.plots[[name]] &lt;- p
}

Oxphos.Vln.plots[[&quot;DF20&quot;]]</code></pre>
<p><img src="figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/oxphosGS-1.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-oxphosGS-1">
Past versions of oxphosGS-1.png
</button>
</p>
<div id="fig-oxphosGS-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/26f64a3c5a9ca82d37db30b427f88739847f3db5/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/oxphosGS-1.png" target="_blank">26f64a3</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/2dc1bee7caab8ef67312b7d286e864b74e8f3028/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/oxphosGS-1.png" target="_blank">2dc1bee</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-31
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/c8bb9fc065ed0d52d9fff97708768a8502747ff6/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/oxphosGS-1.png" target="_blank">c8bb9fc</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/8ca1e011ab3510091358c41f362229573a9741f9/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/oxphosGS-1.png" target="_blank">8ca1e01</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Oxphos.Vln.plots[[&quot;DF101&quot;]]</code></pre>
<p><img src="figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/oxphosGS-2.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-oxphosGS-2">
Past versions of oxphosGS-2.png
</button>
</p>
<div id="fig-oxphosGS-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/26f64a3c5a9ca82d37db30b427f88739847f3db5/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/oxphosGS-2.png" target="_blank">26f64a3</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/2dc1bee7caab8ef67312b7d286e864b74e8f3028/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/oxphosGS-2.png" target="_blank">2dc1bee</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-31
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/c8bb9fc065ed0d52d9fff97708768a8502747ff6/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/oxphosGS-2.png" target="_blank">c8bb9fc</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/8ca1e011ab3510091358c41f362229573a9741f9/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/oxphosGS-2.png" target="_blank">8ca1e01</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Oxphos.Vln.plots[[&quot;DF68&quot;]]</code></pre>
<p><img src="figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/oxphosGS-3.png" width="1728" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-oxphosGS-3">
Past versions of oxphosGS-3.png
</button>
</p>
<div id="fig-oxphosGS-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/26f64a3c5a9ca82d37db30b427f88739847f3db5/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/oxphosGS-3.png" target="_blank">26f64a3</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/2dc1bee7caab8ef67312b7d286e864b74e8f3028/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/oxphosGS-3.png" target="_blank">2dc1bee</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-31
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/c8bb9fc065ed0d52d9fff97708768a8502747ff6/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/oxphosGS-3.png" target="_blank">c8bb9fc</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/8ca1e011ab3510091358c41f362229573a9741f9/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/oxphosGS-3.png" target="_blank">8ca1e01</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ul>
<li><strong>OBSERVATIONS</strong>
<ul>
<li>The overall trend we see between treatment groups within each model is the same across all three oxphos genesets. However, it seems like the difference in scores is most obvious/drastic when using the <strong>KEGG</strong> geneset.</li>
<li>More comparisons are statistically significant when using the <strong>GO</strong> or <strong>KEGG</strong> genesets in comparison to using the <strong>HALLMARK</strong> geneset.
<ul>
<li>All comparisons within DF101 are significant
<ul>
<li><strong>Vehicle</strong> &gt; <strong>MRD</strong> &gt; <strong>Relapse</strong> (does not support our hypothesis)</li>
</ul></li>
<li>DF68 <strong>MRD</strong> &gt; <strong>Relapse</strong> is significant (supports our hypothesis)</li>
</ul></li>
</ul></li>
<li><strong>SUMMARY STATISTICS</strong></li>
</ul>
<pre class="r"><code>#DF20 ---------------------------------
DF20.vehicle &lt;- subset(PDX_DF20, subset = (treatment.status == &quot;vehicle&quot;))
DF20.MRD &lt;- subset(PDX_DF20, subset = (treatment.status == &quot;MRD&quot;))
DF20.relapse &lt;- subset(PDX_DF20, subset = (treatment.status == &quot;relapse&quot;))

DF20.hm.oxphos.df &lt;- data.frame(
  &quot;MRDvsVehicle&quot; = wilcox.test(DF20.MRD$UNUPDATED.OXPHOS37.centered, DF20.vehicle$UNUPDATED.OXPHOS37.centered)$p.value, 
  &quot;MRDvsRelapse&quot; = wilcox.test(DF20.MRD$UNUPDATED.OXPHOS37.centered, DF20.relapse$UNUPDATED.OXPHOS37.centered)$p.value, 
  &quot;VehiclevsRelapse&quot; = wilcox.test(DF20.vehicle$UNUPDATED.OXPHOS37.centered, DF20.relapse$UNUPDATED.OXPHOS37.centered)$p.value
)

DF20.go.oxphos.df &lt;- 
  data.frame(
  &quot;MRDvsVehicle&quot; = wilcox.test(DF20.MRD$GO.OXPHOS35.centered, DF20.vehicle$GO.OXPHOS35.centered)$p.value, 
  &quot;MRDvsRelapse&quot; = wilcox.test(DF20.MRD$GO.OXPHOS35.centered, DF20.relapse$GO.OXPHOS35.centered)$p.value, 
  &quot;VehiclevsRelapse&quot; = wilcox.test(DF20.vehicle$GO.OXPHOS35.centered, DF20.relapse$GO.OXPHOS35.centered)$p.value
)

DF20.kegg.oxphos.df &lt;- 
  data.frame(
  &quot;MRDvsVehicle&quot; = wilcox.test(DF20.MRD$KEGG.OXPHOS36.centered, DF20.vehicle$KEGG.OXPHOS36.centered)$p.value,
  &quot;MRDvsRelapse&quot; = wilcox.test(DF20.MRD$KEGG.OXPHOS36.centered, DF20.relapse$KEGG.OXPHOS36.centered)$p.value,
  &quot;VehiclevsRelapse&quot; = wilcox.test(DF20.vehicle$KEGG.OXPHOS36.centered, DF20.relapse$KEGG.OXPHOS36.centered)$p.value
)

#DF101 ---------------------------------
DF101.vehicle &lt;- subset(PDX_DF101, subset = (treatment.status == &quot;vehicle&quot;))
DF101.MRD &lt;- subset(PDX_DF101, subset = (treatment.status == &quot;MRD&quot;))
DF101.relapse &lt;- subset(PDX_DF101, subset = (treatment.status == &quot;relapse&quot;))

DF101.hm.oxphos.df &lt;- data.frame(
  &quot;MRDvsVehicle&quot; = wilcox.test(DF101.MRD$UNUPDATED.OXPHOS37.centered, DF101.vehicle$UNUPDATED.OXPHOS37.centered)$p.value, 
  &quot;MRDvsRelapse&quot; = wilcox.test(DF101.MRD$UNUPDATED.OXPHOS37.centered, DF101.relapse$UNUPDATED.OXPHOS37.centered)$p.value, 
  &quot;VehiclevsRelapse&quot; = wilcox.test(DF101.vehicle$UNUPDATED.OXPHOS37.centered, DF101.relapse$UNUPDATED.OXPHOS37.centered)$p.value
)

DF101.go.oxphos.df &lt;- 
  data.frame(
  &quot;MRDvsVehicle&quot; = wilcox.test(DF101.MRD$GO.OXPHOS35.centered, DF101.vehicle$GO.OXPHOS35.centered)$p.value, 
  &quot;MRDvsRelapse&quot; = wilcox.test(DF101.MRD$GO.OXPHOS35.centered, DF101.relapse$GO.OXPHOS35.centered)$p.value, 
  &quot;VehiclevsRelapse&quot; = wilcox.test(DF101.vehicle$GO.OXPHOS35.centered, DF101.relapse$GO.OXPHOS35.centered)$p.value
)

DF101.kegg.oxphos.df &lt;- 
  data.frame(
  &quot;MRDvsVehicle&quot; = wilcox.test(DF101.MRD$KEGG.OXPHOS36.centered, DF101.vehicle$KEGG.OXPHOS36.centered)$p.value,
  &quot;MRDvsRelapse&quot; = wilcox.test(DF101.MRD$KEGG.OXPHOS36.centered, DF101.relapse$KEGG.OXPHOS36.centered)$p.value,
  &quot;VehiclevsRelapse&quot; = wilcox.test(DF101.vehicle$KEGG.OXPHOS36.centered, DF101.relapse$KEGG.OXPHOS36.centered)$p.value
)

#DF68 ---------------------------------
DF68.vehicle &lt;- subset(PDX_DF68, subset = (treatment.status == &quot;vehicle&quot;))
DF68.MRD &lt;- subset(PDX_DF68, subset = (treatment.status == &quot;MRD&quot;))
DF68.relapse &lt;- subset(PDX_DF68, subset = (treatment.status == &quot;relapse&quot;))

DF68.hm.oxphos.df &lt;- data.frame(
  &quot;MRDvsVehicle&quot; = wilcox.test(DF68.MRD$UNUPDATED.OXPHOS37.centered, DF68.vehicle$UNUPDATED.OXPHOS37.centered)$p.value, 
  &quot;MRDvsRelapse&quot; = wilcox.test(DF68.MRD$UNUPDATED.OXPHOS37.centered, DF68.relapse$UNUPDATED.OXPHOS37.centered)$p.value, 
  &quot;VehiclevsRelapse&quot; = wilcox.test(DF68.vehicle$UNUPDATED.OXPHOS37.centered, DF68.relapse$UNUPDATED.OXPHOS37.centered)$p.value
)

DF68.go.oxphos.df &lt;- 
  data.frame(
  &quot;MRDvsVehicle&quot; = wilcox.test(DF68.MRD$GO.OXPHOS35.centered, DF68.vehicle$GO.OXPHOS35.centered)$p.value, 
  &quot;MRDvsRelapse&quot; = wilcox.test(DF68.MRD$GO.OXPHOS35.centered, DF68.relapse$GO.OXPHOS35.centered)$p.value, 
  &quot;VehiclevsRelapse&quot; = wilcox.test(DF68.vehicle$GO.OXPHOS35.centered, DF68.relapse$GO.OXPHOS35.centered)$p.value
)

DF68.kegg.oxphos.df &lt;- 
  data.frame(
  &quot;MRDvsVehicle&quot; = wilcox.test(DF68.MRD$KEGG.OXPHOS36.centered, DF68.vehicle$KEGG.OXPHOS36.centered)$p.value,
  &quot;MRDvsRelapse&quot; = wilcox.test(DF68.MRD$KEGG.OXPHOS36.centered, DF68.relapse$KEGG.OXPHOS36.centered)$p.value,
  &quot;VehiclevsRelapse&quot; = wilcox.test(DF68.vehicle$KEGG.OXPHOS36.centered, DF68.relapse$KEGG.OXPHOS36.centered)$p.value
)

#combine ------------------------------
hm.oxphos.DF &lt;- rbind(DF20.hm.oxphos.df, DF101.hm.oxphos.df, DF68.hm.oxphos.df)
rownames(hm.oxphos.DF) &lt;- c(&quot;HM.OXPHOS.DF20&quot;, &quot;HM.OXPHOS.DF101&quot;, &quot;HM.OXPHOS.DF68&quot;)

go.oxphos.DF &lt;- rbind(DF20.go.oxphos.df, DF101.go.oxphos.df, DF68.go.oxphos.df)
rownames(go.oxphos.DF) &lt;- c(&quot;GO.OXPHOS.DF20&quot;, &quot;GO.OXPHOS.DF101&quot;, &quot;GO.OXPHOS.DF68&quot;)

kegg.oxphos.DF &lt;- rbind(DF20.kegg.oxphos.df, DF101.kegg.oxphos.df, DF68.kegg.oxphos.df)
rownames(kegg.oxphos.DF) &lt;- c(&quot;KEGG.OXPHOS.DF20&quot;, &quot;KEGG.OXPHOS.DF101&quot;, &quot;KEGG.OXPHOS.DF68&quot;)

all.oxphos.DF &lt;- rbind(hm.oxphos.DF, go.oxphos.DF, kegg.oxphos.DF)
DT::datatable(all.oxphos.DF) %&gt;% 
  DT::formatSignif(names(all.oxphos.DF), digits = 2) %&gt;% 
  DT::formatStyle(names(all.oxphos.DF), color = DT::styleInterval(0.05, c(&#39;red&#39;, &#39;black&#39;)))</code></pre>
<div id="htmlwidget-ce5c3853d8cf75d7e03c" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-ce5c3853d8cf75d7e03c">{"x":{"filter":"none","data":[["HM.OXPHOS.DF20","HM.OXPHOS.DF101","HM.OXPHOS.DF68","GO.OXPHOS.DF20","GO.OXPHOS.DF101","GO.OXPHOS.DF68","KEGG.OXPHOS.DF20","KEGG.OXPHOS.DF101","KEGG.OXPHOS.DF68"],[0.713582333120361,2.93398926074337e-05,0.23354658226192,0.539910629357917,1.05912304507862e-06,0.262203696434015,0.460698666301977,1.81961942711733e-05,0.220082024126051],[0.239306460964626,0.00582504483054365,0.0824469174562721,0.167136317291695,0.000190204318100418,0.0143820901123957,0.177053988013909,2.11507138839815e-05,0.0111176507629096],[0.447916118434429,1.04913548727899e-07,0.843493154318674,0.567722174320975,1.03732209313145e-10,0.82339642391274,0.546075091342282,1.47738702776111e-10,0.897549974026351]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>MRDvsVehicle<\/th>\n      <th>MRDvsRelapse<\/th>\n      <th>VehiclevsRelapse<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"targets":1,"render":"function(data, type, row, meta) { return DTWidget.formatSignif(data, 2, 3, \",\", \".\"); }"},{"targets":2,"render":"function(data, type, row, meta) { return DTWidget.formatSignif(data, 2, 3, \",\", \".\"); }"},{"targets":3,"render":"function(data, type, row, meta) { return DTWidget.formatSignif(data, 2, 3, \",\", \".\"); }"},{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data) {\nvar value=data[1]; $(this.api().cell(row, 1).node()).css({'color':isNaN(parseFloat(value)) ? '' : value <= 0.05 ? \"red\" : \"black\"});\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'color':isNaN(parseFloat(value)) ? '' : value <= 0.05 ? \"red\" : \"black\"});\nvar value=data[3]; $(this.api().cell(row, 3).node()).css({'color':isNaN(parseFloat(value)) ? '' : value <= 0.05 ? \"red\" : \"black\"});\n}"}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.columnDefs.2.render","options.rowCallback"],"jsHooks":[]}</script>
<p>We also tested how the <strong>UNUPDATED</strong> version of <strong>HALLMARK_UNFOLDED_PROTEIN_RESPONSE</strong> differs from the updated version.</p>
<pre class="r"><code>upr.length.df &lt;- data.frame(
  &quot;UNUPDATED.HM.UPR&quot; = length(hallmark.list[[&quot;UNUPDATED.UPR&quot;]]),
  &quot;HALLMARK.UPR&quot; = length(hallmark.list[[&quot;HALLMARK_UNFOLDED_PROTEIN_RESPONSE&quot;]])
)

Found.upr.df &lt;- data.frame(
  &quot;UNUPDATED.HM.UPR&quot; = sum((hallmark.list[[&quot;UNUPDATED.UPR&quot;]] %in% rownames(PDX_All))),
  &quot;HALLMARK.UPR&quot; = sum((hallmark.list[[&quot;HALLMARK_UNFOLDED_PROTEIN_RESPONSE&quot;]] %in% rownames(PDX_All)))
)

PFound.upr.df &lt;- data.frame(
  &quot;UNUPDATED.HM.UPR&quot; = (sum((hallmark.list[[&quot;UNUPDATED.UPR&quot;]] %in% rownames(PDX_All)))/length(hallmark.list[[&quot;UNUPDATED.UPR&quot;]]))*100,
  &quot;HALLMARK.UPR&quot; = (sum((hallmark.list[[&quot;HALLMARK_UNFOLDED_PROTEIN_RESPONSE&quot;]] %in% rownames(PDX_All)))/length(hallmark.list[[&quot;HALLMARK_UNFOLDED_PROTEIN_RESPONSE&quot;]])) * 100
)

upr.df &lt;- rbind(upr.length.df, Found.upr.df, PFound.upr.df)
upr.df[,&quot;UNUPDATED.HM.UPR&quot;] &lt;- round(upr.df[,&quot;UNUPDATED.HM.UPR&quot;])
upr.df[,&quot;HALLMARK.UPR&quot;] &lt;- round(upr.df[,&quot;HALLMARK.UPR&quot;])
rownames(upr.df) &lt;- c(&quot;Num UPR Genes&quot;, &quot;Num Found&quot;, &quot;% Found&quot;)
upr.df</code></pre>
<pre><code>              UNUPDATED.HM.UPR HALLMARK.UPR
Num UPR Genes              113          113
Num Found                  107          105
% Found                     95           93</code></pre>
<ul>
<li>Since the <strong>UNUPDATED</strong> version identifies two more genes than the updated version, we decide to proceed with the <strong>UNUPDATED.UPR</strong> geneset.</li>
</ul>
<p>After deciding the geneset to use for our data, we can now analyze the differential expression of OXPHOS and UPR genes across treatment condition within each model.</p>
</div>
<div id="step-3-de-analysis-1.-visualizing-and-quantifying-de-hallmark-genesets" class="section level3">
<h3>STEP 3 DE ANALYSIS #1. VISUALIZING AND QUANTIFYING DE HALLMARK GENESETS</h3>
<ul>
<li><strong>QUESTION</strong> Are modules (<strong>OXPHOS and UPR</strong>) differentially expressed across treatment conditions within each model?</li>
<li><strong>HYPOTHESIS</strong> We hypothesize that the <strong>MRD</strong> treatment condition will express enriched levels of OXPHOS and UPR hallmarks relative to cells in the <strong>vehicle</strong> and <strong>relapse</strong> treatment conditions (MRD &gt; vehicle, MRD &gt; relapse).
<ul>
<li><strong>APPROACH #1</strong> Violin Plots and UMAP
<ul>
<li>Center <strong>OXPHOS and UPR</strong> module scores at 0 and reassign to metadata</li>
<li>Visualize differences in hallmark score across treatment conditions with Violin Plot (center module scores at 0)</li>
<li>Statistical test (<strong>wilcoxon rank sum test</strong>): is the difference in hallmark score across treatment conditions statistically significant?</li>
</ul></li>
</ul></li>
</ul>
<pre class="r"><code>hms.centered &lt;- c(&quot;UNUPDATED.OXPHOS37.centered&quot;, &quot;UNUPDATED.UPR38.centered&quot;)
#VlnPlot
PDX.names &lt;- c(&quot;DF20&quot;, &quot;DF101&quot;, &quot;DF68&quot;)
PDXs &lt;- c(PDX_DF20, PDX_DF101, PDX_DF68)
Vln.plots &lt;- vector(&quot;list&quot;, length(PDXs))
names(Vln.plots) &lt;- PDX.names

for (i in 1:length(PDXs)){
  obj &lt;- PDX.names[[i]]
  numCells &lt;- nrow(PDXs[[i]]@meta.data)
  
  my_comparisons &lt;- list(
    c(&quot;MRD&quot;, &quot;vehicle&quot;),
    c(&quot;MRD&quot;, &quot;relapse&quot;), 
    c(&quot;vehicle&quot;, &quot;relapse&quot;)
  )
  
  if(obj == &quot;DF101&quot;){
    p &lt;- VlnPlot(PDXs[[i]], features = hms.centered, group.by = &quot;treatment.status&quot;, pt.size = 0, c(&quot;#00AFBB&quot;, &quot;#E7B800&quot;, &quot;#FC4E07&quot;), combine = F, y.max = 1.3)
  }
  else if(obj == &quot;DF68&quot;){
    p &lt;- VlnPlot(PDXs[[i]], features = hms.centered, group.by = &quot;treatment.status&quot;, pt.size = 0, c(&quot;#00AFBB&quot;, &quot;#E7B800&quot;, &quot;#FC4E07&quot;), combine = F, y.max = 1.4)
  }
  else{
  p &lt;- VlnPlot(PDXs[[i]], features = hms.centered, group.by = &quot;treatment.status&quot;, pt.size = 0, c(&quot;#00AFBB&quot;, &quot;#E7B800&quot;, &quot;#FC4E07&quot;), combine = F, y.max = 1.7)
  }
  
  unupdated.found &lt;- sum(hallmark.list[[&quot;UNUPDATED.OXPHOS&quot;]] %in% rownames(PDXs[[i]]))
  unupdated.length &lt;- length(hallmark.list[[&quot;UNUPDATED.OXPHOS&quot;]])
  unupdated.pFound &lt;- round((unupdated.found / unupdated.length)*100, 2)
  p[[1]] &lt;- p[[1]] + labs(title = glue(&quot;OXPHOS scores across treatment in {obj}&quot;), x = obj, subtitle = glue(&quot;{numCells} Malignant Cells, {unupdated.found} out of {unupdated.length} OXPHOS genes found ({unupdated.pFound}%)&quot;), caption = &quot;www.gsea-msigdb.org: HALLMARK_OXIDATIVE_PHOSPHORYLATION&quot;) +
    theme(plot.caption = element_text(size = 10)) + 
    geom_boxplot(width = 0.15, position = position_dodge(0.9), alpha = 0.3, show.legend = F) + 
    geom_text(label = paste(sum(PDXs[[i]]$treatment.status == &quot;vehicle&quot;), &quot;cells&quot;), x = &quot;vehicle&quot;, y = min(PDXs[[i]]$UNUPDATED.OXPHOS37.centered) -0.03) + 
    geom_text(label = paste(sum(PDXs[[i]]$treatment.status == &quot;MRD&quot;), &quot;cells&quot;), x = &quot;MRD&quot;, y = min(PDXs[[i]]$UNUPDATED.OXPHOS37.centered) - 0.03) + 
    geom_text(label = paste(sum(PDXs[[i]]$treatment.status == &quot;relapse&quot;), &quot;cells&quot;), x = &quot;relapse&quot;, y = min(PDXs[[i]]$UNUPDATED.OXPHOS37.centered) - 0.03) + 
    stat_compare_means(comparisons = my_comparisons, method = &quot;wilcox.test&quot;, label = &quot;p.format&quot;, step.increase = 0.05) + 
    stat_compare_means(comparisons = my_comparisons, method = &quot;wilcox.test&quot;, label = &quot;p.signif&quot;, step.increase = 0.05, bracket.size = 0, vjust = 1.8)
  
  UPR.found &lt;- sum(hallmark.list[[&quot;UNUPDATED.UPR&quot;]] %in% rownames(PDXs[[i]]))
  UPR.length &lt;- length(hallmark.list[[&quot;UNUPDATED.UPR&quot;]])
  UPR.pFound &lt;- round((UPR.found / UPR.length)*100, 2)
  p[[2]] &lt;- p[[2]] + labs(title = glue(&quot;UPR scores across treatment in {obj}&quot;), x = obj, caption = &quot;www.gsea-msigdb.org: HALLMARK_UNFOLDED_PROTEIN_RESPONSE&quot;, subtitle= glue(&quot;{numCells} Malignant Cells, {UPR.found} out of {UPR.length} UPR genes found ({UPR.pFound}%)&quot;)) +
    theme(plot.caption = element_text(size = 10)) + 
    geom_boxplot(width = 0.15, position = position_dodge(0.9), alpha = 0.3, show.legend = F) + 
    geom_text(label = paste(sum(PDXs[[i]]$treatment.status == &quot;vehicle&quot;), &quot;cells&quot;), x = &quot;vehicle&quot;, y = min(PDXs[[i]]$UNUPDATED.UPR38.centered) -0.03) + 
    geom_text(label = paste(sum(PDXs[[i]]$treatment.status == &quot;MRD&quot;), &quot;cells&quot;), x = &quot;MRD&quot;, y = min(PDXs[[i]]$UNUPDATED.UPR38.centered) - 0.03) + 
    geom_text(label = paste(sum(PDXs[[i]]$treatment.status == &quot;relapse&quot;), &quot;cells&quot;), x = &quot;relapse&quot;, y = min(PDXs[[i]]$UNUPDATED.UPR38.centered) - 0.03) +
    stat_compare_means(comparisons = my_comparisons, method = &quot;wilcox.test&quot;, label = &quot;p.format&quot;, step.increase = 0.1) + 
    stat_compare_means(comparisons = my_comparisons, method = &quot;wilcox.test&quot;, label = &quot;p.signif&quot;, step.increase = 0.1, bracket.size = 0, vjust = 1.8)
  
  p &lt;- p[[1]] + p[[2]] + plot_layout(guides= &#39;collect&#39;)
  
  Vln.plots[[obj]] &lt;- p
}

Vln.plots[[&quot;DF20&quot;]]</code></pre>
<p><img src="figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/DEAnalysis1%20VlnPlot-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-DEAnalysis1-VlnPlot-1">
Past versions of “DEAnalysis1 VlnPlot-1.png”
</button>
</p>
<div id="fig-no-spaces-DEAnalysis1-VlnPlot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/26f64a3c5a9ca82d37db30b427f88739847f3db5/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/DEAnalysis1 VlnPlot-1.png" target="_blank">26f64a3</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/2dc1bee7caab8ef67312b7d286e864b74e8f3028/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/DEAnalysis1 VlnPlot-1.png" target="_blank">2dc1bee</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-31
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/c8bb9fc065ed0d52d9fff97708768a8502747ff6/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/DEAnalysis1 VlnPlot-1.png" target="_blank">c8bb9fc</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/8ca1e011ab3510091358c41f362229573a9741f9/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/DEAnalysis1 VlnPlot-1.png" target="_blank">8ca1e01</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-27
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/bc21d3a4bbfeaaab3c3701621e489ba430369973/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/DEAnalysis1 VlnPlot-1.png" target="_blank">bc21d3a</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Vln.plots[[&quot;DF101&quot;]]</code></pre>
<p><img src="figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/DEAnalysis1%20VlnPlot-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-DEAnalysis1-VlnPlot-2">
Past versions of “DEAnalysis1 VlnPlot-2.png”
</button>
</p>
<div id="fig-no-spaces-DEAnalysis1-VlnPlot-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/26f64a3c5a9ca82d37db30b427f88739847f3db5/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/DEAnalysis1 VlnPlot-2.png" target="_blank">26f64a3</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/2dc1bee7caab8ef67312b7d286e864b74e8f3028/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/DEAnalysis1 VlnPlot-2.png" target="_blank">2dc1bee</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-31
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/c8bb9fc065ed0d52d9fff97708768a8502747ff6/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/DEAnalysis1 VlnPlot-2.png" target="_blank">c8bb9fc</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/8ca1e011ab3510091358c41f362229573a9741f9/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/DEAnalysis1 VlnPlot-2.png" target="_blank">8ca1e01</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-27
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/bc21d3a4bbfeaaab3c3701621e489ba430369973/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/DEAnalysis1 VlnPlot-2.png" target="_blank">bc21d3a</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Vln.plots[[&quot;DF68&quot;]]</code></pre>
<p><img src="figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/DEAnalysis1%20VlnPlot-3.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-DEAnalysis1-VlnPlot-3">
Past versions of “DEAnalysis1 VlnPlot-3.png”
</button>
</p>
<div id="fig-no-spaces-DEAnalysis1-VlnPlot-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/26f64a3c5a9ca82d37db30b427f88739847f3db5/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/DEAnalysis1 VlnPlot-3.png" target="_blank">26f64a3</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/2dc1bee7caab8ef67312b7d286e864b74e8f3028/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/DEAnalysis1 VlnPlot-3.png" target="_blank">2dc1bee</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-31
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/c8bb9fc065ed0d52d9fff97708768a8502747ff6/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/DEAnalysis1 VlnPlot-3.png" target="_blank">c8bb9fc</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/8ca1e011ab3510091358c41f362229573a9741f9/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/DEAnalysis1 VlnPlot-3.png" target="_blank">8ca1e01</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-27
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/bc21d3a4bbfeaaab3c3701621e489ba430369973/docs/figure/03.1_Izar2020_PDX_DEAnalysis.Rmd/DEAnalysis1 VlnPlot-3.png" target="_blank">bc21d3a</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ul>
<li><strong>CONCLUSIONS</strong>
<ul>
<li>The only statistically significant difference (p &lt; 0.05) in hallmark scores is between treatment conditions within model <strong>DF101</strong> for <strong>OXPHOS</strong></li>
<li>DF101 OXPHOS: <strong>Vehicle &gt; MRD &gt; Relapse</strong></li>
<li>However, the trends found in DF101 do not agree with our hypothesis that cells in <strong>MRD</strong> differentially enrich OXPHOS genes)</li>
</ul></li>
</ul>
<p>Considering how our approach with analyzing and comparing <strong>module score</strong> did not support our hypothesis, we try to confirm our results again with our second appraoch: <strong>GSEA</strong></p>
<ul>
<li><strong>STATISTICAL SUMMARY</strong></li>
</ul>
<pre class="r"><code>#DF20 ---------------------------------

DF20.oxphos.df &lt;- data.frame(
  &quot;MRDvsVehicle&quot; = wilcox.test(DF20.MRD$UNUPDATED.OXPHOS37.centered, DF20.vehicle$UNUPDATED.OXPHOS37.centered)$p.value, 
  &quot;MRDvsRelapse&quot; = wilcox.test(DF20.MRD$UNUPDATED.OXPHOS37.centered, DF20.relapse$UNUPDATED.OXPHOS37.centered)$p.value, 
  &quot;VehiclevsRelapse&quot; = wilcox.test(DF20.vehicle$UNUPDATED.OXPHOS37.centered, DF20.relapse$UNUPDATED.OXPHOS37.centered)$p.value
)
DF20.UPR.df &lt;- 
  data.frame(
  &quot;MRDvsVehicle&quot; = wilcox.test(DF20.MRD$UNUPDATED.UPR38.centered, DF20.vehicle$UNUPDATED.UPR38.centered)$p.value, 
  &quot;MRDvsRelapse&quot; = wilcox.test(DF20.MRD$UNUPDATED.UPR38.centered, DF20.relapse$UNUPDATED.UPR38.centered)$p.value, 
  &quot;VehiclevsRelapse&quot; = wilcox.test(DF20.vehicle$UNUPDATED.UPR38.centered, DF20.relapse$UNUPDATED.UPR38.centered)$p.value
)
#DF101 ---------------------------------

DF101.oxphos.df &lt;- data.frame(
  &quot;MRDvsVehicle&quot; = wilcox.test(DF101.MRD$UNUPDATED.OXPHOS37.centered, DF101.vehicle$UNUPDATED.OXPHOS37.centered)$p.value, 
  &quot;MRDvsRelapse&quot; = wilcox.test(DF101.MRD$UNUPDATED.OXPHOS37.centered, DF101.relapse$UNUPDATED.OXPHOS37.centered)$p.value, 
  &quot;VehiclevsRelapse&quot; = wilcox.test(DF101.vehicle$UNUPDATED.OXPHOS37.centered, DF101.relapse$UNUPDATED.OXPHOS37.centered)$p.value
)
DF101.UPR.df &lt;- 
  data.frame(
  &quot;MRDvsVehicle&quot; = wilcox.test(DF101.MRD$UNUPDATED.UPR38.centered, DF101.vehicle$UNUPDATED.UPR38.centered)$p.value, 
  &quot;MRDvsRelapse&quot; = wilcox.test(DF101.MRD$UNUPDATED.UPR38.centered, DF101.relapse$UNUPDATED.UPR38.centered)$p.value, 
  &quot;VehiclevsRelapse&quot; = wilcox.test(DF101.vehicle$UNUPDATED.UPR38.centered, DF101.relapse$UNUPDATED.UPR38.centered)$p.value
)
#DF68 ---------------------------------

DF68.oxphos.df &lt;- data.frame(
  &quot;MRDvsVehicle&quot; = wilcox.test(DF68.MRD$UNUPDATED.OXPHOS37.centered, DF68.vehicle$UNUPDATED.OXPHOS37.centered)$p.value, 
  &quot;MRDvsRelapse&quot; = wilcox.test(DF68.MRD$UNUPDATED.OXPHOS37.centered, DF68.relapse$UNUPDATED.OXPHOS37.centered)$p.value, 
  &quot;VehiclevsRelapse&quot; = wilcox.test(DF68.vehicle$UNUPDATED.OXPHOS37.centered, DF68.relapse$UNUPDATED.OXPHOS37.centered)$p.value
)
DF68.UPR.df &lt;- 
  data.frame(
  &quot;MRDvsVehicle&quot; = wilcox.test(DF68.MRD$UNUPDATED.UPR38.centered, DF68.vehicle$UNUPDATED.UPR38.centered)$p.value, 
  &quot;MRDvsRelapse&quot; = wilcox.test(DF68.MRD$UNUPDATED.UPR38.centered, DF68.relapse$UNUPDATED.UPR38.centered)$p.value, 
  &quot;VehiclevsRelapse&quot; = wilcox.test(DF68.vehicle$UNUPDATED.UPR38.centered, DF68.relapse$UNUPDATED.UPR38.centered)$p.value
)
#combine ------------------------------
oxphos.DF &lt;- rbind(DF20.oxphos.df, DF101.oxphos.df, DF68.oxphos.df)
rownames(oxphos.DF) &lt;- c(&quot;OXPHOS.DF20&quot;, &quot;OXPHOS.DF101&quot;, &quot;OXPHOS.DF68&quot;)
UPR.DF &lt;- rbind(DF20.UPR.df, DF101.UPR.df, DF68.UPR.df)
rownames(UPR.DF) &lt;- c(&quot;UPR.DF20&quot;, &quot;UPR.DF101&quot;, &quot;UPR.DF68&quot;)
all.DF &lt;- rbind(oxphos.DF, UPR.DF)
DT::datatable(all.DF) %&gt;% 
  DT::formatSignif(names(all.DF), digits = 2) %&gt;% 
  DT::formatStyle(names(all.DF), color = DT::styleInterval(0.05, c(&#39;red&#39;, &#39;black&#39;)))</code></pre>
<div id="htmlwidget-65aae89599b535d33635" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-65aae89599b535d33635">{"x":{"filter":"none","data":[["OXPHOS.DF20","OXPHOS.DF101","OXPHOS.DF68","UPR.DF20","UPR.DF101","UPR.DF68"],[0.713582333120361,2.93398926074337e-05,0.23354658226192,0.262058430419473,0.201155986562501,0.194842911474621],[0.239306460964626,0.00582504483054365,0.0824469174562721,0.375647254681424,0.632146304836974,0.0957990505421039],[0.447916118434429,1.04913548727899e-07,0.843493154318674,0.964281631269759,0.524168950789418,0.850215460872883]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>MRDvsVehicle<\/th>\n      <th>MRDvsRelapse<\/th>\n      <th>VehiclevsRelapse<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"targets":1,"render":"function(data, type, row, meta) { return DTWidget.formatSignif(data, 2, 3, \",\", \".\"); }"},{"targets":2,"render":"function(data, type, row, meta) { return DTWidget.formatSignif(data, 2, 3, \",\", \".\"); }"},{"targets":3,"render":"function(data, type, row, meta) { return DTWidget.formatSignif(data, 2, 3, \",\", \".\"); }"},{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data) {\nvar value=data[1]; $(this.api().cell(row, 1).node()).css({'color':isNaN(parseFloat(value)) ? '' : value <= 0.05 ? \"red\" : \"black\"});\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'color':isNaN(parseFloat(value)) ? '' : value <= 0.05 ? \"red\" : \"black\"});\nvar value=data[3]; $(this.api().cell(row, 3).node()).css({'color':isNaN(parseFloat(value)) ? '' : value <= 0.05 ? \"red\" : \"black\"});\n}"}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.columnDefs.2.render","options.rowCallback"],"jsHooks":[]}</script>
<ul>
<li><strong>APPROACH #2</strong> Geneset Enrichment Analysis (GSEA) - GSEA enrichment plots for hallmarks of interest between condition.1 vs. condition.2 - Rank genes and compute GSEA enrichment scores - Statistical test: how significant are the enrichment scores?</li>
</ul>
<p><strong>have not finalized the statistical test and ranking method to use for GSEA</strong></p>
</div>
<div id="step-4-de-analysis-2.-identifying-individual-de-genes" class="section level3">
<h3>STEP 4 DE ANALYSIS #2. IDENTIFYING INDIVIDUAL DE GENES</h3>
<p><strong>have not finalized the statistical test to use for volcano plot (need to match what we use for GSEA)</strong></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Mojave 10.14.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
 [1] ggpubr_0.4.0          GGally_2.0.0          gt_0.2.1              reshape2_1.4.4        tidyselect_1.1.0     
 [6] presto_1.0.0          data.table_1.12.8     Rcpp_1.0.5            glue_1.4.1            patchwork_1.0.1      
[11] EnhancedVolcano_1.6.0 ggrepel_0.8.2         here_0.1              readxl_1.3.1          forcats_0.5.0        
[16] stringr_1.4.0         dplyr_1.0.0           purrr_0.3.4           readr_1.3.1           tidyr_1.1.0          
[21] tibble_3.0.3          ggplot2_3.3.2         tidyverse_1.3.0       cowplot_1.0.0         Seurat_3.1.5         
[26] BiocManager_1.30.10   renv_0.11.0-4        

loaded via a namespace (and not attached):
  [1] Rtsne_0.15         colorspace_1.4-1   ggsignif_0.6.0     rio_0.5.16         ellipsis_0.3.1     ggridges_0.5.2    
  [7] rprojroot_1.3-2    fs_1.4.2           rstudioapi_0.11    farver_2.0.3       leiden_0.3.3       listenv_0.8.0     
 [13] DT_0.14            fansi_0.4.1        lubridate_1.7.9    xml2_1.3.2         codetools_0.2-16   splines_4.0.2     
 [19] knitr_1.29         jsonlite_1.7.0     workflowr_1.6.2    broom_0.7.0        ica_1.0-2          cluster_2.1.0     
 [25] dbplyr_1.4.4       png_0.1-7          uwot_0.1.8         sctransform_0.2.1  compiler_4.0.2     httr_1.4.1        
 [31] backports_1.1.8    assertthat_0.2.1   Matrix_1.2-18      lazyeval_0.2.2     limma_3.44.3       cli_2.0.2         
 [37] later_1.1.0.1      htmltools_0.5.0    tools_4.0.2        rsvd_1.0.3         igraph_1.2.5       gtable_0.3.0      
 [43] RANN_2.6.1         carData_3.0-4      cellranger_1.1.0   vctrs_0.3.2        ape_5.4            nlme_3.1-148      
 [49] crosstalk_1.1.0.1  lmtest_0.9-37      xfun_0.15          globals_0.12.5     openxlsx_4.1.5     rvest_0.3.5       
 [55] lifecycle_0.2.0    irlba_2.3.3        rstatix_0.6.0      future_1.18.0      MASS_7.3-51.6      zoo_1.8-8         
 [61] scales_1.1.1       hms_0.5.3          promises_1.1.1     parallel_4.0.2     RColorBrewer_1.1-2 curl_4.3          
 [67] yaml_2.2.1         reticulate_1.16    pbapply_1.4-2      gridExtra_2.3      reshape_0.8.8      stringi_1.4.6     
 [73] zip_2.0.4          rlang_0.4.7        pkgconfig_2.0.3    evaluate_0.14      lattice_0.20-41    ROCR_1.0-11       
 [79] labeling_0.3       htmlwidgets_1.5.1  RcppAnnoy_0.0.16   plyr_1.8.6         magrittr_1.5       R6_2.4.1          
 [85] generics_0.0.2     DBI_1.1.0          foreign_0.8-80     pillar_1.4.6       haven_2.3.1        whisker_0.4       
 [91] withr_2.2.0        fitdistrplus_1.1-1 abind_1.4-5        survival_3.2-3     future.apply_1.6.0 tsne_0.1-3        
 [97] car_3.0-8          modelr_0.1.8       crayon_1.3.4       KernSmooth_2.23-17 plotly_4.9.2.1     rmarkdown_2.3     
[103] grid_4.0.2         blob_1.2.1         git2r_0.27.1       reprex_0.3.0       digest_0.6.25      httpuv_1.5.4      
[109] munsell_0.5.0      viridisLite_0.3.0 </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
