<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jesslyn Goh and Mike Cuoco" />


<title>Izar 2020 PDX (Cohort 3) Cell Cycle Analysis</title>

<script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.14/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">jesslyn_ovca</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="coding.html">Coding convention</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/mikecuoco/jesslyn_ovca">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Izar 2020 PDX (Cohort 3) Cell Cycle Analysis</h1>
<h4 class="author">Jesslyn Goh and Mike Cuoco</h4>
<h4 class="date">7/29/2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-08-03
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>jesslyn_ovca/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200713code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200713)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200713code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200713)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjgoh2jesslynovcatreeeaa900ebf0881bea5dc0b4a7395e3b2b0da3a17etargetblankeaa900ea"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/jgoh2/jesslyn_ovca/tree/eaa900ebf0881bea5dc0b4a7395e3b2b0da3a17e" target="_blank">eaa900e</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomjgoh2jesslynovcatreeeaa900ebf0881bea5dc0b4a7395e3b2b0da3a17etargetblankeaa900ea" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/jgoh2/jesslyn_ovca/tree/eaa900ebf0881bea5dc0b4a7395e3b2b0da3a17e" target="_blank">eaa900e</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/HTAPP/
    Ignored:    data/Izar_2020/
    Ignored:    data/gene_lists/.DS_Store
    Ignored:    data/gene_lists/extra/.DS_Store
    Ignored:    jesslyn_plots/
    Ignored:    mike_plots/
    Ignored:    old/.DS_Store
    Ignored:    old/edited/.DS_Store
    Ignored:    renv/.DS_Store
    Ignored:    renv/library/
    Ignored:    renv/python/
    Ignored:    renv/staging/
    Ignored:    vignettes/

Unstaged changes:
    Modified:   old/edited/02_Izar2020_SS2_Load_Plots.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/03.2_Izar2020_PDX_CellCycleAnalysis.Rmd</code>) and HTML (<code>docs/03.2_Izar2020_PDX_CellCycleAnalysis.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/eaa900ebf0881bea5dc0b4a7395e3b2b0da3a17e/analysis/03.2_Izar2020_PDX_CellCycleAnalysis.Rmd" target="_blank">eaa900e</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-03
</td>
<td>
workflowr::wflow_publish(files = files)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgoh2/jesslyn_ovca/2dc1bee7caab8ef67312b7d286e864b74e8f3028/docs/03.2_Izar2020_PDX_CellCycleAnalysis.html" target="_blank">2dc1bee</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/284aad41a1a7e82fdd119034dd23aeb385fa2f03/analysis/03.2_Izar2020_PDX_CellCycleAnalysis.Rmd" target="_blank">284aad4</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-31
</td>
<td>
workflowr::wflow_publish(files = files)
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/c8bb9fc065ed0d52d9fff97708768a8502747ff6/analysis/03.2_Izar2020_PDX_CellCycleAnalysis.Rmd" target="_blank">c8bb9fc</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-30
</td>
<td>
PDX Exploratory + DE + Cell Cycle Analyses
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgoh2/jesslyn_ovca/c8bb9fc065ed0d52d9fff97708768a8502747ff6/docs/03.2_Izar2020_PDX_CellCycleAnalysis.html" target="_blank">c8bb9fc</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-30
</td>
<td>
PDX Exploratory + DE + Cell Cycle Analyses
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="izar-2020-pdx-cohort-3-data-differantial-expresison-analysis" class="section level1">
<h1>IZAR 2020 PDX (COHORT 3) DATA DIFFERANTIAL EXPRESISON ANALYSIS</h1>
<div id="overview" class="section level3">
<h3>OVERVIEW</h3>
<ul>
<li>The PDX data from Izar2020 consists of <strong>only Malignant cells</strong> from three HGSOC PDX models derived from patients with different treatment histories were selected for implantation:
<ul>
<li>DF20 (BRCA WT treatment-naive, clinically platinum sensitive)</li>
<li>DF101 (BRCA1 mutant, 2 lines of prior therapy, clinically platinum resistant)</li>
<li>DF68 (BRCA1 mutant, 6 lines of prior therapy, clinically platinum resistant)</li>
</ul></li>
<li>After tumors were established, animals were divided into two groups per model:
<ul>
<li>Vehicle (treated with DMSO)</li>
<li>Carboplatin (treated with IP carboplatin)
<ul>
<li>Carboplatin-treated mice for minimal residual disease (<strong>MRD</strong>) group were harvested for scRNA-seq</li>
<li>The remaining carboplatin-treated mice were harvested at endpoint (<strong>vehicle</strong>)</li>
</ul></li>
</ul></li>
<li>In our 5-part analysis of the Izar 2020 PDX data, we are interested in identifying differentially expressed genes and hallmark genesets between treatment statuses <strong>within</strong> each model.</li>
<li>We split our PDX analysis into three parts:
<ol style="list-style-type: decimal">
<li><strong>Load Data and Create PDX Seurat Object</strong>
<ol style="list-style-type: lower-alpha">
<li>The code to this part of our analysis is in the <strong>read_Izar_2020.R</strong> file in the <strong>code</strong> folder. During this part of our analysis we:
<ol style="list-style-type: lower-roman">
<li>Load in PDX count matrix and Create Seurat Object</li>
<li>Assign Metadata identities including:
<ul>
<li>Mouse ID</li>
<li>Model ID</li>
<li>Treatment Status</li>
</ul></li>
<li>Score cells for <strong>cell cycle</strong> and <strong>hallmark genesets</strong> - <strong>Note</strong>: It does not matter whether we call AddModuleScore before or after subsetting and scaling each model because AddModuleScore uses the <strong>data</strong> slot.</li>
<li>Save Seurat Object</li>
</ol></li>
</ol></li>
<li><strong>Process Data and Exploratory Data Analysis</strong>
<ol style="list-style-type: lower-alpha">
<li>The code to this part of our analysis can be found in the <strong>03_Izar2020_PDX_Load</strong> file in the <strong>old/edited</strong> folder. During this part of our analysis we:
<ol style="list-style-type: lower-roman">
<li>Load in PDX Seurat Object from Part 1 and <strong>subset by model</strong>. Continue analysis separately for each model.</li>
<li>Scale and FindVariableFeatures (prepares data for dimensionality reduction)</li>
<li>Dimensionality Reduction (PCA + UMAP)</li>
<li>Save Seurat Objects</li>
</ol></li>
</ol></li>
<li><strong>Exploratory Data Analysis</strong>
<ol style="list-style-type: lower-alpha">
<li>The code to this part of our analysis can be found in the <strong>03.0_Izar2020_PDX_Exploratory Analysis</strong> file in the <strong>analysis</strong> folder. During this part of our analysis we:
<ol style="list-style-type: lower-roman">
<li>Load in PDX Seurat Object from Part 2. Analyze separately for each model.</li>
<li>Compute summary metrics for PDX data such as:
<ul>
<li>Number of cells per model per treatment</li>
<li>Number of cells per treatment per cell cycle phase</li>
</ul></li>
<li>Visualize how cells separate based on metadata identities via UMAP - Intermodel heterogeneity: How do cells separate by model? - Intramodel heterogeneity:
<ul>
<li>How do cells separate by treament status?</li>
<li>How do cells separate by cell cycle phase?</li>
</ul></li>
</ol></li>
</ol></li>
<li><strong>DE Analysis</strong>
<ol style="list-style-type: lower-alpha">
<li><strong>TYPE #1 DE ANALYSIS</strong>: Visualizing and Quantifying Differentially Expression on <strong>Predefined</strong> GO Genesets
<ol style="list-style-type: lower-roman">
<li>Violin Plots and UMAP</li>
<li>Gene Set Enrichment Analysis (GSEA)</li>
</ol></li>
<li><strong>TYPE #2 DE ANALYSIS</strong>: Finding DE Genes from scratch
<ol style="list-style-type: lower-roman">
<li>Volcano Plots</li>
</ol></li>
</ol></li>
<li><strong>CELL CYCLE ANALYSIS</strong>
<ol style="list-style-type: lower-alpha">
<li><strong>TYPE #1 CELL CYCLE ANALYSIS</strong>: Examine correlation between treatment condition and cell cycle phase</li>
<li>Evaluate the idea that cell cycle might influence expression of signatures</li>
</ol></li>
</ol></li>
</ul>
<p>This is the fifth part of our 5-part analysis of the Izar 2020 PDX (Cohort 3) data.</p>
</div>
<div id="cell-cycle-analysis-in-depth-explanation" class="section level3">
<h3>CELL CYCLE ANALYSIS IN-DEPTH EXPLANATION</h3>
<p>We are interested in answering a few questions for our Cell Cycle Analysis:</p>
<p><strong>CELL CYCLE ANALYSIS</strong></p>
<ul>
<li><strong>QUESTION #1</strong> Are more cells at different cell cycle phases between treatments?
<ul>
<li><strong>APPROACH</strong> Violin Plots and UMAP
<ul>
<li>Visualize differences in cell cycle score across treatment conditions with:
<ul>
<li>Stacked Bar Plot</li>
<li>Violin Plot</li>
<li>UMAP by treatment vs. UMAP by cell cycle phase vs. UMAP by S and G2M Score</li>
</ul></li>
<li>Statistical test: is the difference in cell cycle scores across treatment conditions statistically</li>
</ul></li>
</ul></li>
<li><strong>QUESTION #2</strong> Is there a correlation between cell cycle phase and the expression of signatures?
<ul>
<li><strong>APPROACH</strong> Violin Plots
<ul>
<li>Create Violin Plots for each hallmark of interest like **DE Analysis #1/APPROACH #1.</li>
<li>Stratify the Violin Plots by cell cycle phase</li>
</ul></li>
</ul></li>
</ul>
</div>
<div id="step-1-load-in-seurat-objects-and-genesets" class="section level3">
<h3>STEP 1 LOAD IN SEURAT OBJECTS AND GENESETS</h3>
<pre class="r"><code># Load packages
source(here::here(&#39;packages.R&#39;))

#Read in PDX RDS object
PDX_All = readRDS(&quot;data/Izar_2020/test/jesslyn_PDX_All_processed.RDS&quot;)
PDX_DF20 = readRDS(&quot;data/Izar_2020/test/jesslyn_PDX_DF20_processed.RDS&quot;)
PDX_DF101 = readRDS(&quot;data/Izar_2020/test/jesslyn_PDX_DF101_processed.RDS&quot;)
PDX_DF68 = readRDS(&quot;data/Izar_2020/test/jesslyn_PDX_DF68_processed.RDS&quot;)

#center module and cell cycle scores and reassign to the metadata of each Seurat object
hm.names &lt;- names(PDX_All@meta.data)[9:48]

for(i in hm.names){
    DF20.hm.centered &lt;- scale(PDX_DF20[[i]], center = TRUE, scale = FALSE)
    PDX_DF20 &lt;- AddMetaData(PDX_DF20, DF20.hm.centered, col.name = glue(&quot;{i}.centered&quot;))
    
    DF101.hm.centered &lt;- scale(PDX_DF101[[i]], center = TRUE, scale = FALSE)
    PDX_DF101 &lt;- AddMetaData(PDX_DF101, DF101.hm.centered, col.name = glue(&quot;{i}.centered&quot;))
    
    DF68.hm.centered &lt;- scale(PDX_DF68[[i]], center = TRUE, scale = FALSE)
    PDX_DF68 &lt;- AddMetaData(PDX_DF68, DF68.hm.centered, col.name = glue(&quot;{i}.centered&quot;))
}</code></pre>
</div>
<div id="step-2-cell-cycle-analysis" class="section level3">
<h3>STEP 2 CELL CYCLE ANALYSIS</h3>
<p>In our <strong>Exploratory Analysis</strong> section, we observed that PC_2 captures Cell Cycle Phase, S.Score and G2M scores, as cells in the same phase or have high S or G2M scores are found to be near each other on the PC_2 Axis. However, cells did not seem to cluster by treatment status. We now use a more quantitative approach to investigate the correlation between treatment and cell cycle phase.</p>
<ul>
<li><strong>QUESTION #1</strong> Are more cells at different cell cycle phases between treatments?</li>
<li><strong>HYPOTHESIS</strong> There should be less cycling cells in the MRD treatment conditions relative to the other two treatment conditions.
<ul>
<li><strong>APPROACH</strong> Violin Plots and UMAP
<ul>
<li>Visualize differences in cell cycle score across treatment conditions with:
<ul>
<li>Stacked Bar Plot</li>
<li>Violin Plot</li>
</ul></li>
<li>Statistical test: is the difference in cell cycle scores across treatment conditions statistically significant.</li>
</ul></li>
</ul></li>
</ul>
<pre class="r"><code>#Stacked Bar Plot --------------------------
hms.centered &lt;- c(&quot;UNUPDATED.OXPHOS37.centered&quot;, &quot;UNUPDATED.UPR38.centered&quot;)
PDXs &lt;- c(PDX_DF20, PDX_DF101, PDX_DF68)
PDX.names &lt;- c(&quot;DF20&quot;, &quot;DF101&quot;, &quot;DF68&quot;)
bar.plots &lt;- vector(&quot;list&quot;, length = 3)
names(bar.plots) &lt;- PDX.names

for(i in 1:length(PDXs)){
  obj = PDXs[[i]]
  name = PDX.names[[i]]
  numCells = nrow(obj@meta.data)
  
  obj$Phase &lt;- factor(obj$Phase, levels = c(&quot;G1&quot;, &quot;S&quot;, &quot;G2M&quot;))
  t = table(obj$Phase, obj$treatment.status) %&gt;% 
    as.data.frame() %&gt;% 
    rename(Phase = Var1, Treatment = Var2, numCells = Freq)

sum = c()
z = 1
while(z &lt; nrow(t)){
    n = t$numCells[z] + t$numCells[z+1] + t$numCells[z+2]
    sum = c(sum, rep(n,3))
    z = z + 3
  }
  
t &lt;- t %&gt;% mutate(total = sum) %&gt;% mutate(percent = (numCells/total)*100)
  
  p = t %&gt;% 
    ggplot(aes(x=Treatment, y=percent, fill=Phase)) + 
    geom_bar(stat=&quot;identity&quot;) + 
    labs(title = glue(&quot;{name} % of Malignant Cells in Each Cell Cycle Phase/ Treatment&quot;), subtitle = glue(&quot;{numCells} Malignant Cells&quot;), caption = &quot;Labeled by # cells in each phase per treatment&quot;) +
    theme(plot.title = element_text(size = 10), plot.subtitle = element_text(size = 10), plot.caption = element_text(size = 8)) +
    geom_text(aes(label = numCells), position = &quot;stack&quot;, hjust = 0.5, vjust = 2, size = 3, color = &quot;white&quot;)
  bar.plots[[i]] &lt;- p
}

bar.plots[[&quot;DF20&quot;]] + bar.plots[[&quot;DF101&quot;]] + bar.plots[[&quot;DF68&quot;]] + plot_layout(guides= &#39;collect&#39;)</code></pre>
<p><img src="figure/03.2_Izar2020_PDX_CellCycleAnalysis.Rmd/cellcycle1-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cellcycle1-1">
Past versions of cellcycle1-1.png
</button>
</p>
<div id="fig-cellcycle1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/c8bb9fc065ed0d52d9fff97708768a8502747ff6/docs/figure/03.2_Izar2020_PDX_CellCycleAnalysis.Rmd/cellcycle1-1.png" target="_blank">c8bb9fc</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ul>
<li><strong>OBSERVATIONS</strong>
<ul>
<li>Different fractions of cells do seem to be in different cell cycle phases depending on the treatment condition. However, the trends we see do not necessary agree with our hypothesis that the <strong>MRD</strong> treatment condition will have the highest % of noncycling cells:
<ul>
<li>DF20: % Noncycling - Vehicle &gt; Relapse &gt; MRD (does not support our hypothesis)</li>
<li>DF101: % Noncycling - Vehicle &gt; MRD &gt; Relapse (does not support our hypothesis)</li>
<li>DF68: % Noncycling - Relapse &gt; Vehicle &gt; MRD (does not support our hypothesis)</li>
</ul></li>
<li>In all cases, cells in <strong>MRD</strong> do not seem to have the highest fraction of noncycling G1 cells.</li>
</ul></li>
</ul>
<p>To investigate this further, we build violin plots of cell cycle scores separated by treatment status.</p>
<pre class="r"><code>#VlnPlot -------------------------
cc.Vln.plots &lt;- vector(&quot;list&quot;, length = 3)
names(cc.Vln.plots) &lt;- PDX.names

for (i in 1:length(PDXs)){
  obj &lt;- PDX.names[[i]]
  numCells = nrow(PDXs[[i]]@meta.data)
  
  my_comparisons &lt;- list(
    c(&quot;MRD&quot;, &quot;vehicle&quot;),
    c(&quot;MRD&quot;, &quot;relapse&quot;), 
    c(&quot;vehicle&quot;, &quot;relapse&quot;)
  )
  
if(obj == &quot;DF101&quot;){
  p &lt;- VlnPlot(PDXs[[i]], features = c(&quot;S.Score.centered&quot;, &quot;G2M.Score.centered&quot;), group.by = &quot;treatment.status&quot;, pt.size = 0, combine = F, y.max = 4.3, c(&quot;#00AFBB&quot;, &quot;#E7B800&quot;, &quot;#FC4E07&quot;))
}
else{
  p &lt;- VlnPlot(PDXs[[i]], features = c(&quot;S.Score.centered&quot;, &quot;G2M.Score.centered&quot;), group.by = &quot;treatment.status&quot;, pt.size = 0, combine = F, y.max = 4, c(&quot;#00AFBB&quot;, &quot;#E7B800&quot;, &quot;#FC4E07&quot;))
}
  p[[1]] &lt;- p[[1]] + labs(title = glue(&quot;{obj} S.Score across treatment&quot;), x = obj, subtitle = glue(&quot;{numCells} Malignant Cells&quot;)) + 
    geom_boxplot(width = 0.15, position = position_dodge(0.9), alpha = 0.3, show.legend = F) + 
    geom_text(label = paste(sum(PDXs[[i]]$treatment.status == &quot;vehicle&quot;), &quot;cells&quot;), x = &quot;vehicle&quot;, y = min(PDXs[[i]]$S.Score.centered) -0.05) + 
    geom_text(label = paste(sum(PDXs[[i]]$treatment.status == &quot;MRD&quot;), &quot;cells&quot;), x = &quot;MRD&quot;, y = min(PDXs[[i]]$S.Score.centered) - 0.05) + 
    geom_text(label = paste(sum(PDXs[[i]]$treatment.status == &quot;relapse&quot;), &quot;cells&quot;), x = &quot;relapse&quot;, y = min(PDXs[[i]]$S.Score.centered) - 0.05) + 
    stat_compare_means(comparisons = my_comparisons, method = &quot;wilcox.test&quot;, label = &quot;p.format&quot;, step.increase = 0.06) + 
    stat_compare_means(comparisons = my_comparisons, method = &quot;wilcox.test&quot;, label = &quot;p.signif&quot;, step.increase = 0.06, bracket.size = 0, vjust = 1.8)
  
  p[[2]] &lt;- p[[2]] + labs(title = glue(&quot;{obj} G2M.Score across treatment&quot;), x = obj, subtitle = glue(&quot;{numCells} Malignant Cells&quot;)) +
    geom_boxplot(width = 0.15, position = position_dodge(0.9), alpha = 0.3, show.legend = F) + 
    geom_text(label = paste(sum(PDXs[[i]]$treatment.status == &quot;vehicle&quot;), &quot;cells&quot;), x = &quot;vehicle&quot;, y = min(PDXs[[i]]$G2M.Score.centered) -0.05) + 
    geom_text(label = paste(sum(PDXs[[i]]$treatment.status == &quot;MRD&quot;), &quot;cells&quot;), x = &quot;MRD&quot;, y = min(PDXs[[i]]$G2M.Score.centered) - 0.05) + 
    geom_text(label = paste(sum(PDXs[[i]]$treatment.status == &quot;relapse&quot;), &quot;cells&quot;), x = &quot;relapse&quot;, y = min(PDXs[[i]]$G2M.Score.centered) - 0.05) + 
    stat_compare_means(comparisons = my_comparisons, method = &quot;wilcox.test&quot;, label = &quot;p.format&quot;, step.increase = 0.06) + 
    stat_compare_means(comparisons = my_comparisons, method = &quot;wilcox.test&quot;, label = &quot;p.signif&quot;, step.increase = 0.06, bracket.size = 0, vjust = 1.8)

  cc.Vln.plots[[obj]] &lt;- p[[1]] + p[[2]] + plot_layout(guides= &#39;collect&#39;)
}

cc.Vln.plots[[&quot;DF20&quot;]]</code></pre>
<p><img src="figure/03.2_Izar2020_PDX_CellCycleAnalysis.Rmd/cellcycleVln-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cellcycleVln-1">
Past versions of cellcycleVln-1.png
</button>
</p>
<div id="fig-cellcycleVln-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/2dc1bee7caab8ef67312b7d286e864b74e8f3028/docs/figure/03.2_Izar2020_PDX_CellCycleAnalysis.Rmd/cellcycleVln-1.png" target="_blank">2dc1bee</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-31
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/c8bb9fc065ed0d52d9fff97708768a8502747ff6/docs/figure/03.2_Izar2020_PDX_CellCycleAnalysis.Rmd/cellcycleVln-1.png" target="_blank">c8bb9fc</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cc.Vln.plots[[&quot;DF101&quot;]]</code></pre>
<p><img src="figure/03.2_Izar2020_PDX_CellCycleAnalysis.Rmd/cellcycleVln-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cellcycleVln-2">
Past versions of cellcycleVln-2.png
</button>
</p>
<div id="fig-cellcycleVln-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/2dc1bee7caab8ef67312b7d286e864b74e8f3028/docs/figure/03.2_Izar2020_PDX_CellCycleAnalysis.Rmd/cellcycleVln-2.png" target="_blank">2dc1bee</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-31
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/c8bb9fc065ed0d52d9fff97708768a8502747ff6/docs/figure/03.2_Izar2020_PDX_CellCycleAnalysis.Rmd/cellcycleVln-2.png" target="_blank">c8bb9fc</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cc.Vln.plots[[&quot;DF68&quot;]]</code></pre>
<p><img src="figure/03.2_Izar2020_PDX_CellCycleAnalysis.Rmd/cellcycleVln-3.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cellcycleVln-3">
Past versions of cellcycleVln-3.png
</button>
</p>
<div id="fig-cellcycleVln-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/2dc1bee7caab8ef67312b7d286e864b74e8f3028/docs/figure/03.2_Izar2020_PDX_CellCycleAnalysis.Rmd/cellcycleVln-3.png" target="_blank">2dc1bee</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-31
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/c8bb9fc065ed0d52d9fff97708768a8502747ff6/docs/figure/03.2_Izar2020_PDX_CellCycleAnalysis.Rmd/cellcycleVln-3.png" target="_blank">c8bb9fc</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ul>
<li><strong>CONCLUSIONS</strong>
<ul>
<li>Only comparisons within model <strong>DF20</strong> yielded statistically significant results (p &lt; 0.05) for both <strong>S and G2M scores</strong>:
<ul>
<li>DF20: For both S and G2M scores - MRD &gt;&gt; Vehicle; Relapse &gt;&gt; Vehicle</li>
</ul></li>
<li>These results do not support our hypothesis that there should be less cycling cells in the <strong>MRD</strong> treatment condition than the other two treatment conditions.</li>
</ul></li>
<li><strong>STATISTICAL SUMMARY</strong></li>
</ul>
<pre class="r"><code>#DF20 ---------------------------------
DF20.vehicle &lt;- subset(PDX_DF20, subset = (treatment.status == &quot;vehicle&quot;))
DF20.MRD &lt;- subset(PDX_DF20, subset = (treatment.status == &quot;MRD&quot;))
DF20.relapse &lt;- subset(PDX_DF20, subset = (treatment.status == &quot;relapse&quot;))

DF20.sscore.df &lt;- data.frame(
  &quot;MRDvsVehicle&quot; = wilcox.test(DF20.MRD$S.Score.centered, DF20.vehicle$S.Score.centered)$p.value, 
  &quot;MRDvsRelapse&quot; = wilcox.test(DF20.MRD$S.Score.centered, DF20.relapse$S.Score.centered)$p.value, 
  &quot;VehiclevsRelapse&quot; = wilcox.test(DF20.vehicle$S.Score.centered, DF20.relapse$S.Score.centered)$p.value
)
DF20.g2m.df &lt;- 
  data.frame(
  &quot;MRDvsVehicle&quot; = wilcox.test(DF20.MRD$G2M.Score.centered, DF20.vehicle$G2M.Score.centered)$p.value, 
  &quot;MRDvsRelapse&quot; = wilcox.test(DF20.MRD$G2M.Score.centered, DF20.relapse$G2M.Score.centered)$p.value, 
  &quot;VehiclevsRelapse&quot; = wilcox.test(DF20.vehicle$G2M.Score.centered, DF20.relapse$G2M.Score.centered)$p.value
)
#DF101 ---------------------------------
DF101.vehicle &lt;- subset(PDX_DF101, subset = (treatment.status == &quot;vehicle&quot;))
DF101.MRD &lt;- subset(PDX_DF101, subset = (treatment.status == &quot;MRD&quot;))
DF101.relapse &lt;- subset(PDX_DF101, subset = (treatment.status == &quot;relapse&quot;))

DF101.sscore.df &lt;- data.frame(
  &quot;MRDvsVehicle&quot; = wilcox.test(DF101.MRD$S.Score.centered, DF101.vehicle$S.Score.centered)$p.value, 
  &quot;MRDvsRelapse&quot; = wilcox.test(DF101.MRD$S.Score.centered, DF101.relapse$S.Score.centered)$p.value, 
  &quot;VehiclevsRelapse&quot; = wilcox.test(DF101.vehicle$S.Score.centered, DF101.relapse$S.Score.centered)$p.value
)
DF101.g2m.df &lt;- 
  data.frame(
  &quot;MRDvsVehicle&quot; = wilcox.test(DF101.MRD$G2M.Score.centered, DF101.vehicle$G2M.Score.centered)$p.value, 
  &quot;MRDvsRelapse&quot; = wilcox.test(DF101.MRD$G2M.Score.centered, DF101.relapse$G2M.Score.centered)$p.value, 
  &quot;VehiclevsRelapse&quot; = wilcox.test(DF101.vehicle$G2M.Score.centered, DF101.relapse$G2M.Score.centered)$p.value
)
#DF68 ---------------------------------
DF68.vehicle &lt;- subset(PDX_DF68, subset = (treatment.status == &quot;vehicle&quot;))
DF68.MRD &lt;- subset(PDX_DF68, subset = (treatment.status == &quot;MRD&quot;))
DF68.relapse &lt;- subset(PDX_DF68, subset = (treatment.status == &quot;relapse&quot;))

DF68.sscore.df &lt;- data.frame(
  &quot;MRDvsVehicle&quot; = wilcox.test(DF68.MRD$S.Score.centered, DF68.vehicle$S.Score.centered)$p.value, 
  &quot;MRDvsRelapse&quot; = wilcox.test(DF68.MRD$S.Score.centered, DF68.relapse$S.Score.centered)$p.value, 
  &quot;VehiclevsRelapse&quot; = wilcox.test(DF68.vehicle$S.Score.centered, DF68.relapse$S.Score.centered)$p.value
)
DF68.g2m.df &lt;- 
  data.frame(
  &quot;MRDvsVehicle&quot; = wilcox.test(DF68.MRD$G2M.Score.centered, DF68.vehicle$G2M.Score.centered)$p.value, 
  &quot;MRDvsRelapse&quot; = wilcox.test(DF68.MRD$G2M.Score.centered, DF68.relapse$G2M.Score.centered)$p.value, 
  &quot;VehiclevsRelapse&quot; = wilcox.test(DF68.vehicle$G2M.Score.centered, DF68.relapse$G2M.Score.centered)$p.value
)
#combine ------------------------------
sscore.DF &lt;- rbind(DF20.sscore.df, DF101.sscore.df, DF68.sscore.df)
rownames(sscore.DF) &lt;- c(&quot;S.DF20&quot;, &quot;S.DF101&quot;, &quot;S.DF68&quot;)
g2m.DF &lt;- rbind(DF20.g2m.df, DF101.g2m.df, DF68.g2m.df)
rownames(g2m.DF) &lt;- c(&quot;g2m.DF20&quot;, &quot;g2m.DF101&quot;, &quot;g2m.DF68&quot;)
both.DF &lt;- rbind(sscore.DF, g2m.DF)
DT::datatable(both.DF) %&gt;% 
  DT::formatSignif(names(both.DF), digits = 2) %&gt;% 
  DT::formatStyle(names(both.DF), color = DT::styleInterval(0.05, c(&#39;red&#39;, &#39;black&#39;)))</code></pre>
<div id="htmlwidget-ce5c3853d8cf75d7e03c" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-ce5c3853d8cf75d7e03c">{"x":{"filter":"none","data":[["S.DF20","S.DF101","S.DF68","g2m.DF20","g2m.DF101","g2m.DF68"],[0.00271073794553325,0.888942538114057,0.770303000465612,0.0184812739341011,0.540353699001157,0.770303000465612],[0.454680305623842,0.81626199280517,0.140856820535078,0.680187954269424,0.345466689376595,0.190157561978004],[0.0383322303842036,0.607147010679134,0.302983921375116,0.0498299437004385,0.75304320436794,0.409914299566131]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>MRDvsVehicle<\/th>\n      <th>MRDvsRelapse<\/th>\n      <th>VehiclevsRelapse<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"targets":1,"render":"function(data, type, row, meta) { return DTWidget.formatSignif(data, 2, 3, \",\", \".\"); }"},{"targets":2,"render":"function(data, type, row, meta) { return DTWidget.formatSignif(data, 2, 3, \",\", \".\"); }"},{"targets":3,"render":"function(data, type, row, meta) { return DTWidget.formatSignif(data, 2, 3, \",\", \".\"); }"},{"className":"dt-right","targets":[1,2,3]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"rowCallback":"function(row, data) {\nvar value=data[1]; $(this.api().cell(row, 1).node()).css({'color':isNaN(parseFloat(value)) ? '' : value <= 0.05 ? \"red\" : \"black\"});\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'color':isNaN(parseFloat(value)) ? '' : value <= 0.05 ? \"red\" : \"black\"});\nvar value=data[3]; $(this.api().cell(row, 3).node()).css({'color':isNaN(parseFloat(value)) ? '' : value <= 0.05 ? \"red\" : \"black\"});\n}"}},"evals":["options.columnDefs.0.render","options.columnDefs.1.render","options.columnDefs.2.render","options.rowCallback"],"jsHooks":[]}</script>
<p>Our results from above collectively suggest that <strong>OXPHOS and UPR</strong> expression, and cell cycle phase, do not significantly correlate with the treatment condition a cell is in. However, considering the idea that cell cycle might influence the expression of signatures, we are now interested in examining whether there is a correlation between the expression of <strong>OXPHOS and UPR</strong> and <strong>cell cycle phase</strong>.</p>
<ul>
<li><strong>QUESTION #2</strong> Is there a correlation between cell cycle phase and the expression of signatures?</li>
<li><strong>HYPOTHESIS</strong> We hypothesize that low cycling cells (G1) should express higher levels of <strong>OXPHOS and UPR</strong> genes.
<ul>
<li><strong>APPROACH</strong> Violin Plots
<ul>
<li>Create Violin Plots for each hallmark of interest like **DE Analysis #1/APPROACH #1.</li>
<li>Stratify the Violin Plots by cell cycle phase</li>
</ul></li>
</ul></li>
</ul>
<pre class="r"><code>cc.exp.plot &lt;- vector(&quot;list&quot;, length = 3)
names(cc.exp.plot) &lt;- PDX.names

for(i in 1:length(PDXs)){
  obj = PDXs[[i]]
  name = PDX.names[[i]]
  numCells = nrow(obj@meta.data)
  obj$Phase &lt;- factor(obj$Phase, levels = c(&quot;G1&quot;, &quot;S&quot;, &quot;G2M&quot;))
  
  my_comparisons &lt;- list(
    c(glue(&quot;G1&quot;), glue(&quot;S&quot;)), 
    c(glue(&quot;S&quot;), glue(&quot;G2M&quot;)), 
    c(glue(&quot;G1&quot;), glue(&quot;G2M&quot;))
  )
  
  if(name == &quot;DF20&quot;){
    p1 &lt;- VlnPlot(obj, features = hms.centered, group.by = &quot;Phase&quot;, pt.size = 0, combine = FALSE, c(&quot;#00AFBB&quot;, &quot;#E7B800&quot;, &quot;#FC4E07&quot;), y.max = 2.1)
  }
  else{
    p1 &lt;- VlnPlot(obj, features = hms.centered, group.by = &quot;Phase&quot;, pt.size = 0, combine = FALSE, c(&quot;#00AFBB&quot;, &quot;#E7B800&quot;, &quot;#FC4E07&quot;), y.max = 1.8)
  }
  p1[[1]] &lt;- p1[[1]] + labs(title = glue(&quot;{name} OXPHOS score by Cell Cycle Phase&quot;), subtitle = glue(&quot;{numCells} Malignant Cells&quot;)) +
    geom_boxplot(width = 0.15, position = position_dodge(0.9), alpha = 0.3, show.legend = F)+  
    geom_text(label = paste(sum(obj$Phase == &quot;G1&quot;), &quot;cells&quot;), x = &quot;G1&quot;, y = min(obj$HALLMARK_OXIDATIVE_PHOSPHORYLATION25.centered) -0.03) + 
    geom_text(label = paste(sum(obj$Phase == &quot;G2M&quot;), &quot;cells&quot;), x = &quot;G2M&quot;, y = min(obj$HALLMARK_OXIDATIVE_PHOSPHORYLATION25.centered) - 0.03) + 
    geom_text(label = paste(sum(obj$Phase == &quot;S&quot;), &quot;cells&quot;), x = &quot;S&quot;, y = min(obj$HALLMARK_OXIDATIVE_PHOSPHORYLATION25.centered) - 0.03) +
    stat_compare_means(comparisons = my_comparisons, method = &quot;wilcox.test&quot;, label = &quot;p.format&quot;, step.increase = 0.12) + 
    stat_compare_means(comparisons = my_comparisons, method = &quot;wilcox.test&quot;, label = &quot;p.signif&quot;, step.increase = 0.12, bracket.size = 0, vjust = 1.8) + 
    theme(plot.title = element_text(size = 10))
    
  
  p1[[2]] &lt;- p1[[2]] + labs(title = glue(&quot;{name} UPR score by Cell Cycle Phase&quot;), subtitle = glue(&quot;{numCells} Malignant Cells&quot;)) +
    geom_boxplot(width = 0.15, position = position_dodge(0.9), alpha = 0.3, show.legend = F) + 
    geom_text(label = paste(sum(obj$Phase == &quot;G1&quot;), &quot;cells&quot;), x = &quot;G1&quot;, y = min(obj$UNUPDATED.UPR38.centered) -0.03) + 
    geom_text(label = paste(sum(obj$Phase == &quot;G2M&quot;), &quot;cells&quot;), x = &quot;G2M&quot;, y = min(obj$UNUPDATED.UPR38.centered) - 0.03) + 
    geom_text(label = paste(sum(obj$Phase == &quot;S&quot;), &quot;cells&quot;), x = &quot;S&quot;, y = min(obj$UNUPDATED.UPR38.centered) - 0.03) + 
    stat_compare_means(comparisons = my_comparisons, method = &quot;wilcox.test&quot;, label = &quot;p.format&quot;, step.increase = 0.18) + 
    stat_compare_means(comparisons = my_comparisons, method = &quot;wilcox.test&quot;, label = &quot;p.signif&quot;, step.increase = 0.18, bracket.size = 0, vjust = 1.8) + 
    theme(plot.title = element_text(size = 10))
  
  p2 &lt;- VlnPlot(obj, features = hms.centered, group.by = &quot;treatment.status&quot;, split.by = &quot;Phase&quot;, pt.size = 0, combine = FALSE, c(&quot;#00AFBB&quot;, &quot;#FC4E07&quot;, &quot;#E7B800&quot;))
  p2[[1]] &lt;- p2[[1]] + labs(title = glue(&quot;{name} OXPHOS score by Cell Cycle Phase / Treatment&quot;)) +
    geom_boxplot(width = 0.15, position = position_dodge(0.9), alpha = 0.3, show.legend = F) + 
    theme(plot.title = element_text(size = 10))
  p2[[2]] &lt;- p2[[2]] + labs(title = glue(&quot;{name} UPR score by Cell Cycle Phase / Treatment&quot;)) +
    geom_boxplot(width = 0.15, position = position_dodge(0.9), alpha = 0.3, show.legend = F) + 
    theme(plot.title = element_text(size = 10))
  
  cc.exp.plot[[i]] &lt;- p1[[1]] + p1[[2]] + p2[[1]]+ p2[[2]] + plot_layout(guides= &#39;collect&#39;)
}

cc.exp.plot[[&quot;DF20&quot;]]</code></pre>
<p><img src="figure/03.2_Izar2020_PDX_CellCycleAnalysis.Rmd/cellcycle2-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cellcycle2-1">
Past versions of cellcycle2-1.png
</button>
</p>
<div id="fig-cellcycle2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/2dc1bee7caab8ef67312b7d286e864b74e8f3028/docs/figure/03.2_Izar2020_PDX_CellCycleAnalysis.Rmd/cellcycle2-1.png" target="_blank">2dc1bee</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-31
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/c8bb9fc065ed0d52d9fff97708768a8502747ff6/docs/figure/03.2_Izar2020_PDX_CellCycleAnalysis.Rmd/cellcycle2-1.png" target="_blank">c8bb9fc</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cc.exp.plot[[&quot;DF101&quot;]]</code></pre>
<p><img src="figure/03.2_Izar2020_PDX_CellCycleAnalysis.Rmd/cellcycle2-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cellcycle2-2">
Past versions of cellcycle2-2.png
</button>
</p>
<div id="fig-cellcycle2-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/2dc1bee7caab8ef67312b7d286e864b74e8f3028/docs/figure/03.2_Izar2020_PDX_CellCycleAnalysis.Rmd/cellcycle2-2.png" target="_blank">2dc1bee</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-31
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/c8bb9fc065ed0d52d9fff97708768a8502747ff6/docs/figure/03.2_Izar2020_PDX_CellCycleAnalysis.Rmd/cellcycle2-2.png" target="_blank">c8bb9fc</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cc.exp.plot[[&quot;DF68&quot;]]</code></pre>
<p><img src="figure/03.2_Izar2020_PDX_CellCycleAnalysis.Rmd/cellcycle2-3.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-cellcycle2-3">
Past versions of cellcycle2-3.png
</button>
</p>
<div id="fig-cellcycle2-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/2dc1bee7caab8ef67312b7d286e864b74e8f3028/docs/figure/03.2_Izar2020_PDX_CellCycleAnalysis.Rmd/cellcycle2-3.png" target="_blank">2dc1bee</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-31
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/c8bb9fc065ed0d52d9fff97708768a8502747ff6/docs/figure/03.2_Izar2020_PDX_CellCycleAnalysis.Rmd/cellcycle2-3.png" target="_blank">c8bb9fc</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ul>
<li><strong>OBSERVATIONS</strong>
<ul>
<li><strong>OXPHOS and UPR</strong> scores comparison across cell cycle phase within each model: S &gt; G1 and G2M (in general) <strong>especially true within DF20</strong></li>
<li><strong>OXPHOS and UPR</strong> scores comparison across cell cycle phase within each treatment condition per model: S &gt; G1 and G2M especially withtin <strong>MRD</strong> treatment condition for <strong>OXPHOS</strong></li>
<li>We hypothesized that low / cycling cells (G1) should express higher levels of <strong>OXPHOS and/or UPR genes</strong> based on findings from previous studies that low cycling cells seem to survive cancer treatment. We try to rationalize our findings that cells in S phase actually express higher levels of our hallmarks:
<ul>
<li>It makes sense that cells in S phase express higher levels of OXPHOS genes because DNA is being replicated in S phase, which requires a lot of energy.</li>
<li>Even if a cell is found to be in the S phase, it might not mean that it is cycling. It is possible that the cell entered the S phase but does not continue through the cell cycle upon treatment. Furthermore, since DNA is being replicated during S phase and mistakes are most commonly made during DNA replication, it is possible that DNA damage repair programs are also most active during this phase, and may subsequently promote survival upon cancer treatment.</li>
<li>This rationale would also agree with our results from comparing S and G2M scores across treatment conditions, where we found that cells in <strong>MRD and Relapse</strong> have statistically significantly higher S and G2M scores than cells in <strong>Vehicle</strong>.</li>
<li>However, this rationale does not explain why we do not see enriched expression of OXPHOS and UPR genes in the MRD treatment condition.</li>
</ul></li>
</ul></li>
</ul>
<p><strong>note: our sample size is extremely small for each condition, which may mean that we have really low power</strong></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Mojave 10.14.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
 [1] ggpubr_0.4.0          GGally_2.0.0          gt_0.2.1              reshape2_1.4.4        tidyselect_1.1.0     
 [6] presto_1.0.0          data.table_1.12.8     Rcpp_1.0.5            glue_1.4.1            patchwork_1.0.1      
[11] EnhancedVolcano_1.6.0 ggrepel_0.8.2         here_0.1              readxl_1.3.1          forcats_0.5.0        
[16] stringr_1.4.0         dplyr_1.0.0           purrr_0.3.4           readr_1.3.1           tidyr_1.1.0          
[21] tibble_3.0.3          ggplot2_3.3.2         tidyverse_1.3.0       cowplot_1.0.0         Seurat_3.1.5         
[26] BiocManager_1.30.10   renv_0.11.0-4        

loaded via a namespace (and not attached):
  [1] Rtsne_0.15         colorspace_1.4-1   ggsignif_0.6.0     rio_0.5.16         ellipsis_0.3.1     ggridges_0.5.2    
  [7] rprojroot_1.3-2    fs_1.4.2           rstudioapi_0.11    farver_2.0.3       leiden_0.3.3       listenv_0.8.0     
 [13] DT_0.14            fansi_0.4.1        lubridate_1.7.9    xml2_1.3.2         codetools_0.2-16   splines_4.0.2     
 [19] knitr_1.29         jsonlite_1.7.0     workflowr_1.6.2    broom_0.7.0        ica_1.0-2          cluster_2.1.0     
 [25] dbplyr_1.4.4       png_0.1-7          uwot_0.1.8         sctransform_0.2.1  compiler_4.0.2     httr_1.4.1        
 [31] backports_1.1.8    assertthat_0.2.1   Matrix_1.2-18      lazyeval_0.2.2     cli_2.0.2          later_1.1.0.1     
 [37] htmltools_0.5.0    tools_4.0.2        rsvd_1.0.3         igraph_1.2.5       gtable_0.3.0       RANN_2.6.1        
 [43] carData_3.0-4      cellranger_1.1.0   vctrs_0.3.2        ape_5.4            nlme_3.1-148       crosstalk_1.1.0.1 
 [49] lmtest_0.9-37      xfun_0.15          globals_0.12.5     openxlsx_4.1.5     rvest_0.3.5        lifecycle_0.2.0   
 [55] irlba_2.3.3        rstatix_0.6.0      future_1.18.0      MASS_7.3-51.6      zoo_1.8-8          scales_1.1.1      
 [61] hms_0.5.3          promises_1.1.1     parallel_4.0.2     RColorBrewer_1.1-2 curl_4.3           yaml_2.2.1        
 [67] reticulate_1.16    pbapply_1.4-2      gridExtra_2.3      reshape_0.8.8      stringi_1.4.6      zip_2.0.4         
 [73] rlang_0.4.7        pkgconfig_2.0.3    evaluate_0.14      lattice_0.20-41    ROCR_1.0-11        labeling_0.3      
 [79] htmlwidgets_1.5.1  RcppAnnoy_0.0.16   plyr_1.8.6         magrittr_1.5       R6_2.4.1           generics_0.0.2    
 [85] DBI_1.1.0          foreign_0.8-80     pillar_1.4.6       haven_2.3.1        whisker_0.4        withr_2.2.0       
 [91] fitdistrplus_1.1-1 abind_1.4-5        survival_3.2-3     future.apply_1.6.0 tsne_0.1-3         car_3.0-8         
 [97] modelr_0.1.8       crayon_1.3.4       KernSmooth_2.23-17 plotly_4.9.2.1     rmarkdown_2.3      grid_4.0.2        
[103] blob_1.2.1         git2r_0.27.1       reprex_0.3.0       digest_0.6.25      httpuv_1.5.4       munsell_0.5.0     
[109] viridisLite_0.3.0 </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
