<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jesslyn Goh and Mike Cuoco" />


<title>GSEA Choices</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">jesslyn_ovca</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="coding.html">Coding convention</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/mikecuoco/jesslyn_ovca">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">GSEA Choices</h1>
<h4 class="author">Jesslyn Goh and Mike Cuoco</h4>
<h4 class="date">7/20/2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-08-29
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>jesslyn_ovca/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200713code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200713)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200713code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200713)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjgoh2jesslynovcatree5ffd81e8aa34f2a2247aef5a4d52dd8e1ca13e6dtargetblank5ffd81ea"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/jgoh2/jesslyn_ovca/tree/5ffd81e8aa34f2a2247aef5a4d52dd8e1ca13e6d" target="_blank">5ffd81e</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomjgoh2jesslynovcatree5ffd81e8aa34f2a2247aef5a4d52dd8e1ca13e6dtargetblank5ffd81ea" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/jgoh2/jesslyn_ovca/tree/5ffd81e8aa34f2a2247aef5a4d52dd8e1ca13e6d" target="_blank">5ffd81e</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    GO_results/.DS_Store
    Ignored:    GO_results/GO_PCA/.DS_Store
    Ignored:    GO_results/GO_Vln/.DS_Store
    Ignored:    GO_results/GO_Vln/PDX/.DS_Store
    Ignored:    GO_results/GO_plots/.DS_Store
    Ignored:    GO_results/GO_plots/PDX/.DS_Store
    Ignored:    GO_results/GO_plots/PDX/F/
    Ignored:    GO_results/GO_plots/SS2/.DS_Store
    Ignored:    GO_results/Tables/.DS_Store
    Ignored:    GO_results/Tables/PDX/.DS_Store
    Ignored:    GO_results/Tables/PDX/F/
    Ignored:    GO_results/Tables/SS2/.DS_Store
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/HTAPP/
    Ignored:    data/Izar_2020/
    Ignored:    data/gene_lists/.DS_Store
    Ignored:    data/gene_lists/GO_PDX/.DS_Store
    Ignored:    data/gene_lists/GO_SS2/.DS_Store
    Ignored:    data/gene_lists/extra/.DS_Store
    Ignored:    jesslyn_plots/
    Ignored:    mike_plots/
    Ignored:    old/.DS_Store
    Ignored:    old/edited/.DS_Store
    Ignored:    renv/.DS_Store
    Ignored:    renv/library/
    Ignored:    renv/python/
    Ignored:    renv/staging/
    Ignored:    vignettes/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/GSEA_choices.Rmd</code>) and HTML (<code>docs/GSEA_choices.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/5ffd81e8aa34f2a2247aef5a4d52dd8e1ca13e6d/analysis/GSEA_choices.Rmd" target="_blank">5ffd81e</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-29
</td>
<td>
PDX choices publish
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgoh2/jesslyn_ovca/2c1accf8a5c21b85e3658190c11441d341d0d635/docs/GSEA_choices.html" target="_blank">2c1accf</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgoh2/jesslyn_ovca/92568b87a1f3b0cfe5715dab09817ed8f3fee774/docs/GSEA_choices.html" target="_blank">92568b8</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/d60aeff9be61fbef75b246307d4c93f2b192aa7f/analysis/GSEA_choices.Rmd" target="_blank">d60aeff</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-17
</td>
<td>
More DE Analysis of other hallmarks
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/a27da95336780edd598ac7adc54c051d3837d254/analysis/GSEA_choices.Rmd" target="_blank">a27da95</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-16
</td>
<td>
GSEA decisions
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgoh2/jesslyn_ovca/a27da95336780edd598ac7adc54c051d3837d254/docs/GSEA_choices.html" target="_blank">a27da95</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-16
</td>
<td>
GSEA decisions
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="gsea-choices" class="section level1">
<h1>GSEA CHOICES</h1>
<ul>
<li>This is a continuation of our GSEA choices exploration within the PDX_choices.Rmd file.</li>
<li>We have previously decided to use the <strong>wilcoxon rank sum test</strong>, and attempted to determine the best ranking method for GSEA by plotting genes on Volcano Plots and labeling them with their gene ranks. We compared the following ranking methods:
<ul>
<li>logFC descending, then padj ascending</li>
<li>padj ascending, then logFC descending</li>
<li>-log10padj x logFC descending</li>
<li>-log10padj x FC descending</li>
</ul></li>
<li><p>Here is the comparison between each ranking method: [<img src="jesslyn_plots/PDX_test/ranktest3.png" alt="Comparison of Ranking Methods" />]</p></li>
<li>While the volcano plots show us that -log10padj x logFC is the best ranking metric because there is a great separation between highly upregulated and downregulated genes, and the false positives are in the middle of the list, the GSEA plot says otherwise.
<ul>
<li>We graphed the GSEA plot for DF20, Patient 8. However, while the -log10padj x logFC metric gave us a negative enrichment score, the -log10padj x FC metric gave us a positive enrichment score (which agrees with results from our VlnPlot).</li>
</ul></li>
<li><p>Here, we would like to have a more in-depth analysis of how GSEA results change depending on our ranking methods, and hopefully be able to detect any problems in our code that might have resulted in the discrepancy in GSEA results based on our ranks.</p></li>
</ul>
<div id="step-1-load-seurat-object" class="section level3">
<h3>STEP 1 LOAD SEURAT OBJECT</h3>
<pre class="r"><code># Load packages
source(here::here(&#39;packages.R&#39;))

#Read in PDX RDS object
PDX_DF20 = readRDS(&quot;data/Izar_2020/test/jesslyn_PDX_DF20_processed.RDS&quot;)
PDX_DF101 = readRDS(&quot;data/Izar_2020/test/jesslyn_PDX_DF101_processed.RDS&quot;)
PDX_DF68 = readRDS(&quot;data/Izar_2020/test/jesslyn_PDX_DF68_processed.RDS&quot;)

#Read in SS2 RDS object
SS2Malignant.8 = readRDS(file = &quot;data/Izar_2020/jesslyn_SS2Malignant8_processed.RDS&quot;)
SS2Malignant.9 = readRDS(file = &quot;data/Izar_2020/jesslyn_SS2Malignant9_processed.RDS&quot;)

# Read in ccgenes
ccgenes = read_lines(&quot;data/gene_lists/regev_lab_cell_cycle_genes.txt&quot;)
s.genes &lt;- ccgenes[1:43]
g2m.genes &lt;- ccgenes[44:97]

#Read in hallmarks of interest
hallmark_names = read_lines(&quot;data/gene_lists/hallmarks.txt&quot;)
hallmark.list &lt;- vector(mode = &quot;list&quot;, length = length(hallmark_names))
names(hallmark.list) &lt;- hallmark_names

for(hm in hallmark_names){
  if(file.exists(glue(&quot;data/gene_lists/hallmarks/{hm}_updated.txt&quot;))){
    file &lt;- read_lines(glue(&quot;data/gene_lists/hallmarks/{hm}_updated.txt&quot;), skip = 1)
    hallmark.list[[hm]] &lt;- file
  }
  else{
    file &lt;- read_lines(glue(&quot;data/gene_lists/extra/{hm}.txt&quot;), skip =2)
    hallmark.list[[hm]] &lt;- file
  }
}

#center module and cell cycle scores and reassign to the metadata of each Seurat object
hm.names &lt;- names(SS2Malignant.8@meta.data)[14:61]

for(i in hm.names){
    SS2Malignant8.hm.centered &lt;- scale(SS2Malignant.8[[i]], center = TRUE, scale = FALSE)
    SS2Malignant.8 &lt;- AddMetaData(SS2Malignant.8, SS2Malignant8.hm.centered, col.name = glue(&quot;{i}.centered&quot;))
    
    SS2Malignant9.hm.centered &lt;- scale(SS2Malignant.9[[i]], center = TRUE, scale = FALSE)
    SS2Malignant.9 &lt;- AddMetaData(SS2Malignant.9, SS2Malignant9.hm.centered, col.name = glue(&quot;{i}.centered&quot;))

    DF20.hm.centered &lt;- scale(PDX_DF20[[i]], center = TRUE, scale = FALSE)
    PDX_DF20 &lt;- AddMetaData(PDX_DF20, DF20.hm.centered, col.name = glue(&quot;{i}.centered&quot;))
    
    DF101.hm.centered &lt;- scale(PDX_DF101[[i]], center = TRUE, scale = FALSE)
    PDX_DF101 &lt;- AddMetaData(PDX_DF101, DF101.hm.centered, col.name = glue(&quot;{i}.centered&quot;))
    
    DF68.hm.centered &lt;- scale(PDX_DF68[[i]], center = TRUE, scale = FALSE)
    PDX_DF68 &lt;- AddMetaData(PDX_DF68, DF68.hm.centered, col.name = glue(&quot;{i}.centered&quot;))
    
}</code></pre>
</div>
<div id="step-2-call-findmarkers" class="section level3">
<h3>STEP 2 CALL FINDMARKERS</h3>
<pre class="r"><code>markers &lt;- vector(&quot;list&quot;, length = 5)
obj.names &lt;- c(&quot;DF20&quot;, &quot;DF101&quot;, &quot;DF68&quot;, &quot;SS2.8&quot;, &quot;SS2.9&quot;)
names(markers) &lt;- obj.names
objs &lt;- c(PDX_DF20, PDX_DF101, PDX_DF68, SS2Malignant.8, SS2Malignant.9)

for(i in 1:5){
  obj.name &lt;- obj.names[[i]]
  obj &lt;- objs[[i]]
  
  comparisons &lt;- c(&quot;MV&quot;, &quot;MR&quot;, &quot;VR&quot;)
  obj.markers &lt;- vector(&quot;list&quot;, length = 3)
  names(obj.markers) &lt;- comparisons
  
  if(str_detect(obj.name, &quot;[DF]&quot;)){
    obj.markers[[&quot;MV&quot;]] &lt;- FindMarkers(obj, group.by = &quot;treatment.status&quot;, ident.1 = &quot;MRD&quot;, ident.2 = &quot;vehicle&quot;, logfc.threshold = 0)
    obj.markers[[&quot;MR&quot;]] &lt;- FindMarkers(obj, group.by = &quot;treatment.status&quot;, ident.1 = &quot;MRD&quot;, ident.2 = &quot;relapse&quot;, logfc.threshold = 0)
    obj.markers[[&quot;VR&quot;]] &lt;- FindMarkers(obj, group.by = &quot;treatment.status&quot;, ident.1 = &quot;vehicle&quot;, ident.2 = &quot;relapse&quot;, logfc.threshold = 0)
  }
  
  else{
    p &lt;- str_split(obj.name, &quot;SS2.&quot;)[[1]][2]
    obj.markers[[&quot;MV&quot;]] &lt;- FindMarkers(obj, group.by = &quot;sample&quot;, ident.1 = glue(&quot;{p}.1&quot;), ident.2 = glue(&quot;{p}.0&quot;), logfc.threshold = 0)
    obj.markers[[&quot;MR&quot;]] &lt;- FindMarkers(obj, group.by = &quot;sample&quot;, ident.1 = glue(&quot;{p}.1&quot;), ident.2 = glue(&quot;{p}.2&quot;), logfc.threshold = 0)
    obj.markers[[&quot;VR&quot;]] &lt;- FindMarkers(obj, group.by = &quot;sample&quot;, ident.1 = glue(&quot;{p}.0&quot;), ident.2 = glue(&quot;{p}.2&quot;), logfc.threshold = 0)
  }
  
  markers[[obj.name]] &lt;- obj.markers
}</code></pre>
</div>
<div id="step-3.-rank-genes" class="section level3">
<h3>STEP 3. RANK GENES</h3>
<pre class="r"><code># logFC descending, padj ascending ------------
logFC.padj.rank &lt;- vector(&quot;list&quot;, length = 5)
names(logFC.padj.rank) &lt;- obj.names

for(name in obj.names){
  comparisons &lt;- c(&quot;MV&quot;, &quot;MR&quot;, &quot;VR&quot;)
  obj.markers &lt;- vector(&quot;list&quot;, length = 3)
  names(obj.markers) &lt;- comparisons
  
  for(comparison in comparisons){
    obj.markers[[comparison]] &lt;- markers[[name]][[comparison]] %&gt;% 
      rownames_to_column() %&gt;% 
      select(rowname, avg_logFC, p_val_adj) %&gt;%
      arrange(-avg_logFC, p_val_adj) %&gt;% 
      select(rowname, avg_logFC)
  }
  
  logFC.padj.rank[[name]] &lt;- obj.markers
}


# padj ascending, logFC descending -------------
padj.logFC.rank &lt;- vector(&quot;list&quot;, length = 5)
names(padj.logFC.rank) &lt;- obj.names

for(name in obj.names){
  comparisons &lt;- c(&quot;MV&quot;, &quot;MR&quot;, &quot;VR&quot;)
  obj.markers &lt;- vector(&quot;list&quot;, length = 3)
  names(obj.markers) &lt;- comparisons
  
  for(comparison in comparisons){
    obj.markers[[comparison]] &lt;- markers[[name]][[comparison]] %&gt;% 
      rownames_to_column() %&gt;% 
      select(rowname, avg_logFC, p_val_adj) %&gt;%
      arrange(p_val_adj, -avg_logFC) %&gt;% 
      select(rowname, p_val_adj)
  }
  
  padj.logFC.rank[[name]] &lt;- obj.markers
}

# -log10padj x logFC descending -------------
neglog10padj.logFC.rank &lt;- vector(&quot;list&quot;, length = 5)
names(neglog10padj.logFC.rank) &lt;- obj.names

for(name in obj.names){
  comparisons &lt;- c(&quot;MV&quot;, &quot;MR&quot;, &quot;VR&quot;)
  obj.markers &lt;- vector(&quot;list&quot;, length = 3)
  names(obj.markers) &lt;- comparisons
  
  for(comparison in comparisons){
    p &lt;- markers[[name]][[comparison]] %&gt;% 
      rownames_to_column() %&gt;% 
      select(rowname, avg_logFC, p_val_adj) %&gt;%
      mutate(&quot;neglog10padj&quot; = -(log10(p_val_adj)))
    p &lt;- p %&gt;% mutate(&quot;rank.metric&quot; = neglog10padj * avg_logFC)
    obj.markers[[comparison]] &lt;- p %&gt;% 
      arrange(-rank.metric) %&gt;% 
      select(rowname, rank.metric)
  }
  
  neglog10padj.logFC.rank[[name]] &lt;- obj.markers
}

# -log10padj x FC descending -------------
neglog10padj.FC.rank &lt;- vector(&quot;list&quot;, length = 5)
names(neglog10padj.FC.rank) &lt;- obj.names

for(name in obj.names){
  comparisons &lt;- c(&quot;MV&quot;, &quot;MR&quot;, &quot;VR&quot;)
  obj.markers &lt;- vector(&quot;list&quot;, length = 3)
  names(obj.markers) &lt;- comparisons
  
  for(comparison in comparisons){
    p &lt;- markers[[name]][[comparison]] %&gt;% 
      rownames_to_column() %&gt;% 
      select(rowname, avg_logFC, p_val_adj) %&gt;%
      mutate(&quot;neglog10padj&quot; = -(log10(p_val_adj))) %&gt;% 
      mutate(&quot;FC&quot; = exp(avg_logFC))
    p &lt;- p %&gt;% mutate(&quot;rank.metric&quot; = neglog10padj * FC)
    obj.markers[[comparison]] &lt;- p %&gt;% arrange(-rank.metric) %&gt;% 
      select(rowname, rank.metric)
  }
  
  neglog10padj.FC.rank[[name]] &lt;- obj.markers
}</code></pre>
</div>
<div id="step-4.-run-gsea" class="section level3">
<h3>STEP 4. RUN GSEA</h3>
<ul>
<li>We will run GSEA with each ranking method and compare results</li>
</ul>
<pre class="r"><code># logFC descending, padj ascending -----------
logFC.padj.GSEA.results &lt;- vector(&quot;list&quot;, length = 5)
logFC.padj.GSEA.plots &lt;- vector(&quot;list&quot;, length = 5)
names(logFC.padj.GSEA.plots) &lt;- obj.names
names(logFC.padj.GSEA.results) &lt;- obj.names

for(name in obj.names){
  comparisons &lt;- c(&quot;MV&quot;, &quot;MR&quot;, &quot;VR&quot;)
  obj.results &lt;- vector(&quot;list&quot;, length = 3)
  names(obj.results) &lt;- comparisons
  obj.plots &lt;- vector(&quot;list&quot;, length = 3)
  names(obj.plots) &lt;- comparisons 
  
  for(comparison in comparisons){
    ranks &lt;- logFC.padj.rank[[name]][[comparison]] %&gt;% deframe()
    obj.results[[comparison]] &lt;- fgsea(hallmark.list, stats = ranks, minSize = 15, maxSize = 500, nperm = 1000) %&gt;% 
  as_tibble() %&gt;% 
  select(-leadingEdge, -ES, -nMoreExtreme) %&gt;% 
  arrange(desc(NES), padj)
    
    padj &lt;- obj.results[[comparison]] %&gt;% filter(pathway == &quot;UNUPDATED.OXPHOS&quot;) %&gt;% select(padj) %&gt;% deframe() %&gt;% round(digits = 3)
NES = obj.results[[comparison]] %&gt;% filter(pathway == &quot;UNUPDATED.OXPHOS&quot;) %&gt;% select(NES) %&gt;% deframe() %&gt;% round(digits = 3)
    
    obj.plots[[comparison]] &lt;- plotEnrichment(pathway = hallmark.list[[&quot;UNUPDATED.OXPHOS&quot;]], stats = ranks) + labs(title = glue(&quot;{name} {comparison} OXPHOS GSEA results&quot;), subtitle = glue(&quot;(NES = {NES}, padj = {padj})&quot;), caption = &quot;logFC descending, padj ascending&quot;)
  }
  
  logFC.padj.GSEA.results[[name]] &lt;- obj.results 
  logFC.padj.GSEA.plots[[name]] &lt;- obj.plots
}

# padj ascending, logFC descending -----------
padj.logFC.GSEA.results &lt;- vector(&quot;list&quot;, length = 5)
padj.logFC.GSEA.plots &lt;- vector(&quot;list&quot;, length = 5)
names(padj.logFC.GSEA.plots) &lt;- obj.names
names(padj.logFC.GSEA.results) &lt;- obj.names

for(name in obj.names){
  comparisons &lt;- c(&quot;MV&quot;, &quot;MR&quot;, &quot;VR&quot;)
  obj.results &lt;- vector(&quot;list&quot;, length = 3)
  names(obj.results) &lt;- comparisons
  obj.plots &lt;- vector(&quot;list&quot;, length = 3)
  names(obj.plots) &lt;- comparisons 
  
  for(comparison in comparisons){
    ranks &lt;- padj.logFC.rank[[name]][[comparison]] %&gt;% deframe()
    obj.results[[comparison]] &lt;- fgsea(hallmark.list, stats = ranks, minSize = 15, maxSize = 500, nperm = 1000) %&gt;% 
  as_tibble() %&gt;% 
  select(-leadingEdge, -ES, -nMoreExtreme) %&gt;% 
  arrange(desc(NES), padj)
    
    padj &lt;- obj.results[[comparison]] %&gt;% filter(pathway == &quot;UNUPDATED.OXPHOS&quot;) %&gt;% select(padj) %&gt;% deframe() %&gt;% round(digits = 3)
NES = obj.results[[comparison]] %&gt;% filter(pathway == &quot;UNUPDATED.OXPHOS&quot;) %&gt;% select(NES) %&gt;% deframe() %&gt;% round(digits = 3)
    
    obj.plots[[comparison]] &lt;- plotEnrichment(pathway = hallmark.list[[&quot;UNUPDATED.OXPHOS&quot;]], stats = ranks) + labs(title = glue(&quot;{name} {comparison} OXPHOS GSEA results&quot;), subtitle = glue(&quot;(NES = {NES}, padj = {padj})&quot;), caption = &quot;padj asecdning, logFC descending&quot;)
  }
  
  padj.logFC.GSEA.results[[name]] &lt;- obj.results 
  padj.logFC.GSEA.plots[[name]] &lt;- obj.plots
}

# neglog10padj x logFC -----------
neglogpadj.logFC.GSEA.results &lt;- vector(&quot;list&quot;, length = 5)
neglogpadj.logFC.GSEA.plots &lt;- vector(&quot;list&quot;, length = 5)
names(neglogpadj.logFC.GSEA.plots) &lt;- obj.names
names(neglogpadj.logFC.GSEA.results) &lt;- obj.names

for(name in obj.names){
  comparisons &lt;- c(&quot;MV&quot;, &quot;MR&quot;, &quot;VR&quot;)
  obj.results &lt;- vector(&quot;list&quot;, length = 3)
  names(obj.results) &lt;- comparisons
  obj.plots &lt;- vector(&quot;list&quot;, length = 3)
  names(obj.plots) &lt;- comparisons 
  
  for(comparison in comparisons){
    ranks &lt;- neglog10padj.logFC.rank[[name]][[comparison]] %&gt;% deframe()
    obj.results[[comparison]] &lt;- fgsea(hallmark.list, stats = ranks, minSize = 15, maxSize = 500, nperm = 1000) %&gt;% 
  as_tibble() %&gt;% 
  select(-leadingEdge, -ES, -nMoreExtreme) %&gt;% 
  arrange(desc(NES), padj)
    
    padj &lt;- obj.results[[comparison]] %&gt;% filter(pathway == &quot;UNUPDATED.OXPHOS&quot;) %&gt;% select(padj) %&gt;% deframe() %&gt;% round(digits = 3)
NES = obj.results[[comparison]] %&gt;% filter(pathway == &quot;UNUPDATED.OXPHOS&quot;) %&gt;% select(NES) %&gt;% deframe() %&gt;% round(digits = 3)
    
    obj.plots[[comparison]] &lt;- plotEnrichment(pathway = hallmark.list[[&quot;UNUPDATED.OXPHOS&quot;]], stats = ranks) + labs(title = glue(&quot;{name} {comparison} OXPHOS GSEA results&quot;), subtitle = glue(&quot;(NES = {NES}, padj = {padj})&quot;), caption = &quot;-log10padj x logFC descending&quot;)
  }
  
  neglogpadj.logFC.GSEA.results[[name]] &lt;- obj.results 
  neglogpadj.logFC.GSEA.plots[[name]] &lt;- obj.plots
}

# neglog10padj x FC -----------
neglogpadj.FC.GSEA.results &lt;- vector(&quot;list&quot;, length = 5)
neglogpadj.FC.GSEA.plots &lt;- vector(&quot;list&quot;, length = 5)
names(neglogpadj.FC.GSEA.plots) &lt;- obj.names
names(neglogpadj.FC.GSEA.results) &lt;- obj.names

for(name in obj.names){
  comparisons &lt;- c(&quot;MV&quot;, &quot;MR&quot;, &quot;VR&quot;)
  obj.results &lt;- vector(&quot;list&quot;, length = 3)
  names(obj.results) &lt;- comparisons
  obj.plots &lt;- vector(&quot;list&quot;, length = 3)
  names(obj.plots) &lt;- comparisons 
  
  for(comparison in comparisons){
    ranks &lt;- neglog10padj.FC.rank[[name]][[comparison]] %&gt;% deframe()
    obj.results[[comparison]] &lt;- fgsea(hallmark.list, stats = ranks, minSize = 15, maxSize = 500, nperm = 1000) %&gt;% 
  as_tibble() %&gt;% 
  select(-leadingEdge, -ES, -nMoreExtreme) %&gt;% 
  arrange(desc(NES), padj)
    
    padj &lt;- obj.results[[comparison]] %&gt;% filter(pathway == &quot;UNUPDATED.OXPHOS&quot;) %&gt;% select(padj) %&gt;% deframe() %&gt;% round(digits = 3)
NES = obj.results[[comparison]] %&gt;% filter(pathway == &quot;UNUPDATED.OXPHOS&quot;) %&gt;% select(NES) %&gt;% deframe() %&gt;% round(digits = 3)
    
    obj.plots[[comparison]] &lt;- plotEnrichment(pathway = hallmark.list[[&quot;UNUPDATED.OXPHOS&quot;]], stats = ranks) + labs(title = glue(&quot;{name} {comparison} OXPHOS GSEA results&quot;), subtitle = glue(&quot;(NES = {NES}, padj = {padj})&quot;), caption = &quot;-log10padj x FC descending&quot;)
  }
  
  neglogpadj.FC.GSEA.results[[name]] &lt;- obj.results 
  neglogpadj.FC.GSEA.plots[[name]] &lt;- obj.plots
}

# show plots ----------
GSEA.plots &lt;- vector(&quot;list&quot;, length = 5)
names(GSEA.plots) &lt;- obj.names

for(name in obj.names){
  comparisons &lt;- c(&quot;MV&quot;, &quot;MR&quot;, &quot;VR&quot;)
  obj.plots &lt;- vector(&quot;list&quot;, length =3)
  names(obj.plots) &lt;- comparisons
  
  for(comparison in comparisons){
    p &lt;- logFC.padj.GSEA.plots[[name]][[comparison]] + padj.logFC.GSEA.plots[[name]][[comparison]] + neglogpadj.logFC.GSEA.plots[[name]][[comparison]] + neglogpadj.FC.GSEA.plots[[name]][[comparison]] + plot_layout(ncol = 2, nrow = 2)
    obj.plots[[comparison]] &lt;- p
    ggsave(plot = p, filename = glue(&quot;GSEA.{name}{comparison}.png&quot;), path = &quot;jesslyn_plots/PDX_test/GSEA_paired/test&quot;, width = 10, height = 10)
  }
  GSEA.plots[[name]] &lt;- obj.plots
}

GSEA.plots[[&quot;DF20&quot;]][[&quot;MV&quot;]]</code></pre>
<p><img src="figure/GSEA_choices.Rmd/GSEA-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-GSEA-1">
Past versions of GSEA-1.png
</button>
</p>
<div id="fig-GSEA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/92568b87a1f3b0cfe5715dab09817ed8f3fee774/docs/figure/GSEA_choices.Rmd/GSEA-1.png" target="_blank">92568b8</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/a27da95336780edd598ac7adc54c051d3837d254/docs/figure/GSEA_choices.Rmd/GSEA-1.png" target="_blank">a27da95</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>GSEA.plots[[&quot;DF20&quot;]][[&quot;MR&quot;]]</code></pre>
<p><img src="figure/GSEA_choices.Rmd/GSEA-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-GSEA-2">
Past versions of GSEA-2.png
</button>
</p>
<div id="fig-GSEA-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/92568b87a1f3b0cfe5715dab09817ed8f3fee774/docs/figure/GSEA_choices.Rmd/GSEA-2.png" target="_blank">92568b8</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/a27da95336780edd598ac7adc54c051d3837d254/docs/figure/GSEA_choices.Rmd/GSEA-2.png" target="_blank">a27da95</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>GSEA.plots[[&quot;DF20&quot;]][[&quot;VR&quot;]]</code></pre>
<p><img src="figure/GSEA_choices.Rmd/GSEA-3.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-GSEA-3">
Past versions of GSEA-3.png
</button>
</p>
<div id="fig-GSEA-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/92568b87a1f3b0cfe5715dab09817ed8f3fee774/docs/figure/GSEA_choices.Rmd/GSEA-3.png" target="_blank">92568b8</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/a27da95336780edd598ac7adc54c051d3837d254/docs/figure/GSEA_choices.Rmd/GSEA-3.png" target="_blank">a27da95</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>GSEA.plots[[&quot;DF101&quot;]][[&quot;MV&quot;]]</code></pre>
<p><img src="figure/GSEA_choices.Rmd/GSEA-4.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-GSEA-4">
Past versions of GSEA-4.png
</button>
</p>
<div id="fig-GSEA-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/92568b87a1f3b0cfe5715dab09817ed8f3fee774/docs/figure/GSEA_choices.Rmd/GSEA-4.png" target="_blank">92568b8</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/a27da95336780edd598ac7adc54c051d3837d254/docs/figure/GSEA_choices.Rmd/GSEA-4.png" target="_blank">a27da95</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>GSEA.plots[[&quot;DF101&quot;]][[&quot;MR&quot;]]</code></pre>
<p><img src="figure/GSEA_choices.Rmd/GSEA-5.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-GSEA-5">
Past versions of GSEA-5.png
</button>
</p>
<div id="fig-GSEA-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/92568b87a1f3b0cfe5715dab09817ed8f3fee774/docs/figure/GSEA_choices.Rmd/GSEA-5.png" target="_blank">92568b8</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/a27da95336780edd598ac7adc54c051d3837d254/docs/figure/GSEA_choices.Rmd/GSEA-5.png" target="_blank">a27da95</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>GSEA.plots[[&quot;DF101&quot;]][[&quot;VR&quot;]]</code></pre>
<p><img src="figure/GSEA_choices.Rmd/GSEA-6.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-GSEA-6">
Past versions of GSEA-6.png
</button>
</p>
<div id="fig-GSEA-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/92568b87a1f3b0cfe5715dab09817ed8f3fee774/docs/figure/GSEA_choices.Rmd/GSEA-6.png" target="_blank">92568b8</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/a27da95336780edd598ac7adc54c051d3837d254/docs/figure/GSEA_choices.Rmd/GSEA-6.png" target="_blank">a27da95</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>GSEA.plots[[&quot;DF68&quot;]][[&quot;MV&quot;]]</code></pre>
<p><img src="figure/GSEA_choices.Rmd/GSEA-7.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-GSEA-7">
Past versions of GSEA-7.png
</button>
</p>
<div id="fig-GSEA-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/92568b87a1f3b0cfe5715dab09817ed8f3fee774/docs/figure/GSEA_choices.Rmd/GSEA-7.png" target="_blank">92568b8</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/a27da95336780edd598ac7adc54c051d3837d254/docs/figure/GSEA_choices.Rmd/GSEA-7.png" target="_blank">a27da95</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>GSEA.plots[[&quot;DF68&quot;]][[&quot;MR&quot;]]</code></pre>
<p><img src="figure/GSEA_choices.Rmd/GSEA-8.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-GSEA-8">
Past versions of GSEA-8.png
</button>
</p>
<div id="fig-GSEA-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/92568b87a1f3b0cfe5715dab09817ed8f3fee774/docs/figure/GSEA_choices.Rmd/GSEA-8.png" target="_blank">92568b8</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/a27da95336780edd598ac7adc54c051d3837d254/docs/figure/GSEA_choices.Rmd/GSEA-8.png" target="_blank">a27da95</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>GSEA.plots[[&quot;DF68&quot;]][[&quot;VR&quot;]]</code></pre>
<p><img src="figure/GSEA_choices.Rmd/GSEA-9.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-GSEA-9">
Past versions of GSEA-9.png
</button>
</p>
<div id="fig-GSEA-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/92568b87a1f3b0cfe5715dab09817ed8f3fee774/docs/figure/GSEA_choices.Rmd/GSEA-9.png" target="_blank">92568b8</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/a27da95336780edd598ac7adc54c051d3837d254/docs/figure/GSEA_choices.Rmd/GSEA-9.png" target="_blank">a27da95</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>GSEA.plots[[&quot;SS2.8&quot;]][[&quot;MV&quot;]]</code></pre>
<p><img src="figure/GSEA_choices.Rmd/GSEA-10.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-GSEA-10">
Past versions of GSEA-10.png
</button>
</p>
<div id="fig-GSEA-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/92568b87a1f3b0cfe5715dab09817ed8f3fee774/docs/figure/GSEA_choices.Rmd/GSEA-10.png" target="_blank">92568b8</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/a27da95336780edd598ac7adc54c051d3837d254/docs/figure/GSEA_choices.Rmd/GSEA-10.png" target="_blank">a27da95</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>GSEA.plots[[&quot;SS2.8&quot;]][[&quot;MR&quot;]]</code></pre>
<p><img src="figure/GSEA_choices.Rmd/GSEA-11.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-GSEA-11">
Past versions of GSEA-11.png
</button>
</p>
<div id="fig-GSEA-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/92568b87a1f3b0cfe5715dab09817ed8f3fee774/docs/figure/GSEA_choices.Rmd/GSEA-11.png" target="_blank">92568b8</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/a27da95336780edd598ac7adc54c051d3837d254/docs/figure/GSEA_choices.Rmd/GSEA-11.png" target="_blank">a27da95</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>GSEA.plots[[&quot;SS2.8&quot;]][[&quot;VR&quot;]]</code></pre>
<p><img src="figure/GSEA_choices.Rmd/GSEA-12.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-GSEA-12">
Past versions of GSEA-12.png
</button>
</p>
<div id="fig-GSEA-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/92568b87a1f3b0cfe5715dab09817ed8f3fee774/docs/figure/GSEA_choices.Rmd/GSEA-12.png" target="_blank">92568b8</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/a27da95336780edd598ac7adc54c051d3837d254/docs/figure/GSEA_choices.Rmd/GSEA-12.png" target="_blank">a27da95</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>GSEA.plots[[&quot;SS2.9&quot;]][[&quot;MV&quot;]]</code></pre>
<p><img src="figure/GSEA_choices.Rmd/GSEA-13.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-GSEA-13">
Past versions of GSEA-13.png
</button>
</p>
<div id="fig-GSEA-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/92568b87a1f3b0cfe5715dab09817ed8f3fee774/docs/figure/GSEA_choices.Rmd/GSEA-13.png" target="_blank">92568b8</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/a27da95336780edd598ac7adc54c051d3837d254/docs/figure/GSEA_choices.Rmd/GSEA-13.png" target="_blank">a27da95</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>GSEA.plots[[&quot;SS2.9&quot;]][[&quot;MR&quot;]]</code></pre>
<p><img src="figure/GSEA_choices.Rmd/GSEA-14.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-GSEA-14">
Past versions of GSEA-14.png
</button>
</p>
<div id="fig-GSEA-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/92568b87a1f3b0cfe5715dab09817ed8f3fee774/docs/figure/GSEA_choices.Rmd/GSEA-14.png" target="_blank">92568b8</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/a27da95336780edd598ac7adc54c051d3837d254/docs/figure/GSEA_choices.Rmd/GSEA-14.png" target="_blank">a27da95</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>GSEA.plots[[&quot;SS2.9&quot;]][[&quot;VR&quot;]]</code></pre>
<p><img src="figure/GSEA_choices.Rmd/GSEA-15.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-GSEA-15">
Past versions of GSEA-15.png
</button>
</p>
<div id="fig-GSEA-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/92568b87a1f3b0cfe5715dab09817ed8f3fee774/docs/figure/GSEA_choices.Rmd/GSEA-15.png" target="_blank">92568b8</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/a27da95336780edd598ac7adc54c051d3837d254/docs/figure/GSEA_choices.Rmd/GSEA-15.png" target="_blank">a27da95</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ul>
<li>We get very different results depending on the ranking method we use. -log10padj x logFC and -log10padj x FC seem to give us opposite NES values most of the time. Which of the two metrics is better is not consistent, with x FC sometimes giving us GSEA results that agree more with our other analyses, and x logFC at other times.</li>
<li>Interesting to note that GSEA plots using -log10padj x logFC or FC metrics give us plots with straight lines, while the plots using only logFC or padj give us curvier plots that look more like what GSEA plots usually look like.</li>
<li>Since GSEA is usually done on RNA-seq data, it is possible that our single-cell RNAseq data is not suitable for GSEA. We will therefore not proceed with GSEA for our analysis.</li>
</ul>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Mojave 10.14.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
 [1] ggbeeswarm_0.6.0      ggpubr_0.4.0          GGally_2.0.0          gt_0.2.1              reshape2_1.4.4       
 [6] tidyselect_1.1.0      fgsea_1.14.0          presto_1.0.0          data.table_1.12.8     Rcpp_1.0.5           
[11] glue_1.4.1            patchwork_1.0.1       EnhancedVolcano_1.6.0 ggrepel_0.8.2         here_0.1             
[16] readxl_1.3.1          forcats_0.5.0         stringr_1.4.0         dplyr_1.0.0           purrr_0.3.4          
[21] readr_1.3.1           tidyr_1.1.0           tibble_3.0.3          ggplot2_3.3.2         tidyverse_1.3.0      
[26] cowplot_1.0.0         Seurat_3.1.5          BiocManager_1.30.10   renv_0.11.0-4        

loaded via a namespace (and not attached):
  [1] backports_1.1.8     fastmatch_1.1-0     workflowr_1.6.2     plyr_1.8.6          igraph_1.2.5       
  [6] lazyeval_0.2.2      splines_4.0.2       BiocParallel_1.22.0 listenv_0.8.0       digest_0.6.25      
 [11] htmltools_0.5.0     fansi_0.4.1         magrittr_1.5        cluster_2.1.0       ROCR_1.0-11        
 [16] openxlsx_4.1.5      limma_3.44.3        globals_0.12.5      modelr_0.1.8        colorspace_1.4-1   
 [21] blob_1.2.1          rvest_0.3.5         haven_2.3.1         xfun_0.15           crayon_1.3.4       
 [26] jsonlite_1.7.0      survival_3.2-3      zoo_1.8-8           ape_5.4             gtable_0.3.0       
 [31] leiden_0.3.3        car_3.0-8           future.apply_1.6.0  abind_1.4-5         scales_1.1.1       
 [36] DBI_1.1.0           rstatix_0.6.0       viridisLite_0.3.0   reticulate_1.16     foreign_0.8-80     
 [41] rsvd_1.0.3          tsne_0.1-3          htmlwidgets_1.5.1   httr_1.4.1          RColorBrewer_1.1-2 
 [46] ellipsis_0.3.1      ica_1.0-2           farver_2.0.3        pkgconfig_2.0.3     reshape_0.8.8      
 [51] uwot_0.1.8          dbplyr_1.4.4        labeling_0.3        rlang_0.4.7         later_1.1.0.1      
 [56] munsell_0.5.0       cellranger_1.1.0    tools_4.0.2         cli_2.0.2           generics_0.0.2     
 [61] broom_0.7.0         ggridges_0.5.2      evaluate_0.14       yaml_2.2.1          knitr_1.29         
 [66] fs_1.4.2            fitdistrplus_1.1-1  zip_2.0.4           RANN_2.6.1          pbapply_1.4-2      
 [71] future_1.18.0       nlme_3.1-148        whisker_0.4         xml2_1.3.2          compiler_4.0.2     
 [76] rstudioapi_0.11     beeswarm_0.2.3      plotly_4.9.2.1      curl_4.3            png_0.1-7          
 [81] ggsignif_0.6.0      reprex_0.3.0        stringi_1.4.6       lattice_0.20-41     Matrix_1.2-18      
 [86] vctrs_0.3.2         pillar_1.4.6        lifecycle_0.2.0     lmtest_0.9-37       RcppAnnoy_0.0.16   
 [91] irlba_2.3.3         httpuv_1.5.4        R6_2.4.1            promises_1.1.1      KernSmooth_2.23-17 
 [96] gridExtra_2.3       rio_0.5.16          vipor_0.4.5         codetools_0.2-16    MASS_7.3-51.6      
[101] assertthat_0.2.1    rprojroot_1.3-2     withr_2.2.0         sctransform_0.2.1   parallel_4.0.2     
[106] hms_0.5.3           grid_4.0.2          rmarkdown_2.3       carData_3.0-4       Rtsne_0.15         
[111] git2r_0.27.1        lubridate_1.7.9    </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
