<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Home</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">jesslyn_ovca</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="coding.html">Coding convention</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/mikecuoco/jesslyn_ovca">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Home</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-08-29
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>jesslyn_ovca/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjgoh2jesslynovcatree5ffd81e8aa34f2a2247aef5a4d52dd8e1ca13e6dtargetblank5ffd81ea"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/jgoh2/jesslyn_ovca/tree/5ffd81e8aa34f2a2247aef5a4d52dd8e1ca13e6d" target="_blank">5ffd81e</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomjgoh2jesslynovcatree5ffd81e8aa34f2a2247aef5a4d52dd8e1ca13e6dtargetblank5ffd81ea" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/jgoh2/jesslyn_ovca/tree/5ffd81e8aa34f2a2247aef5a4d52dd8e1ca13e6d" target="_blank">5ffd81e</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    GO_results/.DS_Store
    Ignored:    GO_results/GO_PCA/.DS_Store
    Ignored:    GO_results/GO_Vln/.DS_Store
    Ignored:    GO_results/GO_Vln/PDX/.DS_Store
    Ignored:    GO_results/GO_plots/.DS_Store
    Ignored:    GO_results/GO_plots/PDX/.DS_Store
    Ignored:    GO_results/GO_plots/PDX/F/
    Ignored:    GO_results/GO_plots/SS2/.DS_Store
    Ignored:    GO_results/Tables/.DS_Store
    Ignored:    GO_results/Tables/PDX/.DS_Store
    Ignored:    GO_results/Tables/PDX/F/
    Ignored:    GO_results/Tables/SS2/.DS_Store
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/HTAPP/
    Ignored:    data/Izar_2020/
    Ignored:    data/gene_lists/.DS_Store
    Ignored:    data/gene_lists/GO_PDX/.DS_Store
    Ignored:    data/gene_lists/GO_SS2/.DS_Store
    Ignored:    data/gene_lists/extra/.DS_Store
    Ignored:    jesslyn_plots/
    Ignored:    mike_plots/
    Ignored:    old/.DS_Store
    Ignored:    old/edited/.DS_Store
    Ignored:    renv/.DS_Store
    Ignored:    renv/library/
    Ignored:    renv/python/
    Ignored:    renv/staging/
    Ignored:    vignettes/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/index.Rmd</code>) and HTML (<code>docs/index.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/5ffd81e8aa34f2a2247aef5a4d52dd8e1ca13e6d/analysis/index.Rmd" target="_blank">5ffd81e</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-29
</td>
<td>
PDX choices publish
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgoh2/jesslyn_ovca/2ceded0f46c54594f6d7aed778ff5acf629ddde2/docs/index.html" target="_blank">2ceded0</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/e7a34101870d963b01786971f797e4cabc269258/analysis/index.Rmd" target="_blank">e7a3410</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-28
</td>
<td>
PDX GO html render
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgoh2/jesslyn_ovca/e7a34101870d963b01786971f797e4cabc269258/docs/index.html" target="_blank">e7a3410</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-28
</td>
<td>
PDX GO html render
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgoh2/jesslyn_ovca/b7634b5550c6e7c65470f800f820a74f229dfa51/docs/index.html" target="_blank">b7634b5</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgoh2/jesslyn_ovca/2c1accf8a5c21b85e3658190c11441d341d0d635/docs/index.html" target="_blank">2c1accf</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/729ebf46e71604f91406a12b0a6c92ad2a832af9/analysis/index.Rmd" target="_blank">729ebf4</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-19
</td>
<td>
workflowr::wflow_publish(files = files)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgoh2/jesslyn_ovca/92568b87a1f3b0cfe5715dab09817ed8f3fee774/docs/index.html" target="_blank">92568b8</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/d60aeff9be61fbef75b246307d4c93f2b192aa7f/analysis/index.Rmd" target="_blank">d60aeff</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-17
</td>
<td>
More DE Analysis of other hallmarks
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgoh2/jesslyn_ovca/80345fe4bf9bd7015617c6c1500cd15d07991f5a/docs/index.html" target="_blank">80345fe</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgoh2/jesslyn_ovca/789982d87a20d30b6ae656007dda319809fe3e6e/docs/index.html" target="_blank">789982d</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgoh2/jesslyn_ovca/e05c3281876449d3de1475d4b6c0c687986bd466/docs/index.html" target="_blank">e05c328</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgoh2/jesslyn_ovca/84edf85071d5ecdaced41f7357a865d5d5b458b8/docs/index.html" target="_blank">84edf85</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgoh2/jesslyn_ovca/a3ddf5426a00200791376a7aca6df663ea19f7ef/docs/index.html" target="_blank">a3ddf54</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgoh2/jesslyn_ovca/d801a7a707759fdd07c5918342259d636a0e5b22/docs/index.html" target="_blank">d801a7a</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgoh2/jesslyn_ovca/26f64a3c5a9ca82d37db30b427f88739847f3db5/docs/index.html" target="_blank">26f64a3</a>
</td>
<td>
jgoh2
</td>
<td>
2020-08-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgoh2/jesslyn_ovca/2dc1bee7caab8ef67312b7d286e864b74e8f3028/docs/index.html" target="_blank">2dc1bee</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/284aad41a1a7e82fdd119034dd23aeb385fa2f03/analysis/index.Rmd" target="_blank">284aad4</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-31
</td>
<td>
workflowr::wflow_publish(files = files)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgoh2/jesslyn_ovca/5c5305bc4eacccf4cb747f6983527abec81a5f22/docs/index.html" target="_blank">5c5305b</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/c8bb9fc065ed0d52d9fff97708768a8502747ff6/analysis/index.Rmd" target="_blank">c8bb9fc</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-30
</td>
<td>
PDX Exploratory + DE + Cell Cycle Analyses
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgoh2/jesslyn_ovca/6be6c853e8ff1ecf2ad96e905cf37ee9b0412dda/docs/index.html" target="_blank">6be6c85</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/cdd10f957d932d708833a7b496e26c575b9e65fd/analysis/index.Rmd" target="_blank">cdd10f9</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-28
</td>
<td>
SS2 DE Analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgoh2/jesslyn_ovca/cdd10f957d932d708833a7b496e26c575b9e65fd/docs/index.html" target="_blank">cdd10f9</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-28
</td>
<td>
SS2 DE Analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgoh2/jesslyn_ovca/35c794768b06ac80182d293ea94c867226d20d49/docs/index.html" target="_blank">35c7947</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-27
</td>
<td>
SS2 Analysis Part 1 and 2
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgoh2/jesslyn_ovca/f1acd7b63f7d61d83ee574a9d77c21cffeeb5cb9/docs/index.html" target="_blank">f1acd7b</a>
</td>
<td>
jgoh2
</td>
<td>
2020-07-24
</td>
<td>
Move PDX_choices.Rmd to old
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/2b6116dfa34b8b875fa6fe6e26c9a9279af32e0f/analysis/index.Rmd" target="_blank">2b6116d</a>
</td>
<td>
Mike Cuoco
</td>
<td>
2020-07-24
</td>
<td>
added 03.1_Izar2020_PDX_DEAnalysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgoh2/jesslyn_ovca/00588156e0b552404a4f77f65893827f117c78da/docs/index.html" target="_blank">0058815</a>
</td>
<td>
Mike Cuoco
</td>
<td>
2020-07-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jgoh2/jesslyn_ovca/a526dc68d0800c3983b778b005e8e7438c6a6efd/docs/index.html" target="_blank">a526dc6</a>
</td>
<td>
Mike Cuoco
</td>
<td>
2020-07-15
</td>
<td>
add workflowr website
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/086db99deebc85ed0d22c8c20a6e83dc188f5674/analysis/index.Rmd" target="_blank">086db99</a>
</td>
<td>
Mike Cuoco
</td>
<td>
2020-07-15
</td>
<td>
add project description
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jgoh2/jesslyn_ovca/blob/e407c2d3b2e6d6b813d229328937a929a78e9431/analysis/index.Rmd" target="_blank">e407c2d</a>
</td>
<td>
Mike Cuoco
</td>
<td>
2020-07-14
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="characterization-of-metabolic-dynamics-of-ovarian-persister-cells-following-carboplatin-treatment" class="section level1">
<h1>CHARACTERIZATION OF METABOLIC DYNAMICS OF OVARIAN PERSISTER CELLS FOLLOWING CARBOPLATIN TREATMENT</h1>
<p>The purpose of this project is to identify gene expression signatures from <strong>High-grade Serous Ovarian Cancer single-cell RNA-seq</strong> data that are associated with a drug tolerant state that is characteristic of cancer persister cells.</p>
<div id="ovarian-cancer" class="section level3">
<h3>OVARIAN CANCER</h3>
<p>Ovarian cancer (OVCA) is one of the deadliest cancers in women (Matulonis et al., 2016). OVCA is a disease with multiple origins and subtypes that share a common anatomical site upon presentation. 90% of the cases are of epithelial origin, the majority of which are High-grade serous tumors, which are known to have the worst prognosis. Despite not being listed as one of the top 10 leading sites of (new) cancer cases, OVCA is among the top 5 leading sites of estimated cancer deaths, with an estimate of 13,940 new deaths in 2020, taking up 5% of all cancer deaths according to the American Cancer Society. Unfortunately, HGSOC is typically detected at late stage due to the lack of symptoms, and the low specificity and sensitivity of the common biomarker CA 125, and in combination with the lack of druggable target molecules, makes the prognosis of HGSOC poor.</p>
<center>
<div class="figure">
<img src="http://img.asiancancer.com/uploads/allimg/2018/08/18/1-15150046.jpg" alt="Ovarian Cancer Stages" style="width:50.0%;height:50.0%" />
<p class="caption">Ovarian Cancer Stages</p>
</div>
</center>
<p><strong>TREATMENT</strong> Late stage OVCA detection is frequently associated with metastasis, and along with the lack of druggable target molecules in ovarian cancer, treatment of ovarian cancer is fairly difficult. Current treatment for ovarian cancer often includes surgery and chemotherapy. The goal of the surgery is to stage the cancer and to debulk the tumor (remove as much as possible). The treatment for ovarian cancer is different at different stages.</p>
<ul>
<li>Stage 1: limited to one or both ovaries, and can be found on the surface of the ovary.</li>
<li>Stage 2: tumor has invaded one or both ovaries and extended to the pelvic region but not to the abdomen</li>
<li>Stage 3: tumor has extended to the abdominal organs</li>
<li>Stage 4: cancer has spread to the lungs, liver, or lymph nodes in the neck</li>
</ul>
<p>The lack of druggable target molecules makes cytotoxic platinum-based chemotherapy, such as cisplatin or carboplatin, the standard chemotherapy drugs for ovarian cancer.</p>
<p><strong>RESISTANCE</strong> Despite improved clinical response and survival in recent years, numerous patients succumb to the disease as a result of chemotherapy resistance. Although 70% of ovarian cancer patients initially respond to platinum-based chemotherapy, relapse occurs in 25% of patients with early stage ovarian cancer, and more than 80% in patients with advanced disease. The majority of these patients experience relapse within 2 years of initial treatment.</p>
<center>
<div class="figure">
<img src="https://www.intechopen.com/media/chapter/43348/media/image1.png" alt="Tapia and Dias-Padilla 2013" style="width:50.0%;height:50.0%" />
<p class="caption">Tapia and Dias-Padilla 2013</p>
</div>
</center>
<p>Being able to understand the molecular mechanisms of chemotherapy resistance would therefore have important clinical relevance as researchers design new therapeutic strategies to overcome resistance and prevent cancer relapse.</p>
</div>
<div id="nongenetic-drug-resistance" class="section level3">
<h3>NONGENETIC DRUG RESISTANCE</h3>
<p>Drug resistance is associated with the heterogeneity within the tumor cell population, where a subpopulation of tumor cells acquires the ability to survive treatment. These cells, termed minimal residual disease cells (MRD) or drug-tolerant persister cells (DTPs) re-proliferate once the environment becomes appropriate to do so, leading to cancer relapse. Earlier studies focused on genetic mechanisms of drug resistance, where pro-survival genetic mutations can either be present in cells of the initial tumor cell population or be acquired during therapy. These mutations allow cancer cells to survive treatment while also passing down this capability permanently to their progeny.</p>
<p>However, non-genetic mechanisms have also been found to play an important role in drug resistance and cancer relapse, one of which was found in the PC9 cell line (epidermal growth factor receptor EFGR mutant non-small-cell lung carcinoma NSCLC), where a subpopulation of cells acquires a slow-cycling, transiently drug-tolerant cell state following treatment (EGFR tyrosine kinase inhibitor) (Sharma et al. 2010). Interestingly, the progeny of these drug-tolerant persisters (drug-tolerant expanded persisters) reacquire cancer drug sensitivity following 30+ passages in drug-free medium, which demonstrates a nongenetic mechanism of drug resistance in the original drug-tolerant persisters.</p>
<p>Although the actual mechanisms of non-genetic drug resistance remain to be fully understood, several studies suggest that chemotherapy resistance may be linked to cancer stem-like cell characteristics such as cancer quiescence or metabolic reprogramming.</p>
<p><strong>CANCER STEM CELLS AND DRUG RESISTANT PERSISTER CELLS</strong> Chemotherapy resistant cells are found to have overlapping biological features with those of cancer stem cells, which includes plasticity, tumor-initiating capacity, and slow proliferation. Cancer stem cells are slow-proliferating cells at the top of the cancer hierarchy, and are involved in tumor-initiation as they give rise to intermediate progenitors and to differentiated progeny. However, the boundaries between stem and non-stem cells have been found to be relatively weak in cancer, and studies report phenomena of therapy-induced enrichment of cancer stem-like cells, where differentiated cells are found to transition to a stem-like cell state that confer resistance.For example, Sharma et al observed that the putative CSC marker CD133 is expressed in all DTPs (but only in approximately 2% of the parental PC9 population), while another CSC marker CD24 is also enriched in some settings.</p>
<center>
<div class="figure">
<img src="https://blog.crownbio.com/hs-fs/hubfs/assets/cancer-stem-cell-hypothesis.jpg?width=1000&amp;name=cancer-stem-cell-hypothesis.jpg" alt="Cancer Stem Cells vs. Normal Stem Cells" style="width:50.0%;height:50.0%" />
<p class="caption">Cancer Stem Cells vs. Normal Stem Cells</p>
</div>
</center>
<p>So how does cancer stem-like characteristics confer drug resistance?</p>
<p><strong>CANCER DORMANCY AND DRUG RESISTANCE</strong> One of the many ways that a stem-like cell state may confer resistance is quiescent. Drug resistance is commonly characterized by a suppression of proliferation.Cancer cells can escape from the mitotic cell cycle and stay arrested at the non or low cycling G0-G1 phase to wait for an appropriate environment. Cancer cells can therefore make use of dormancy to overcome hostile microenvironments and re-enter the cell cycle to proliferate once unfavourable conditions are ruled out.</p>
<center>
<div class="figure">
<img src="https://d3i71xaburhd42.cloudfront.net/344e2110539a24b048fa3c777a27c9395323d71e/2-Figure1-1.png" alt="Takeishi and Nakayama, 2016" style="width:50.0%;height:50.0%" />
<p class="caption">Takeishi and Nakayama, 2016</p>
</div>
</center>
<p><strong>METABOLIC REPROGRAMMING AND DRUG RESISTANCE</strong> Another hypothesized way that a stem-like cell state may confer resistance is plasticity, specifically metabolic plasticity. Several studies report that drug resistant persister cells are transcriptionally and/or metabolically different from their drug-sensitive proliferating counterparts.</p>
<p>A recent study reports a metabolic shift from aerobic glycolysis to oxidative phosphorylation (OXPHOS) in cisplatin-resistant (SKOV3/DPP) ovarian cancer cells (Xu et al., 2018). This is a crucial finding because cancer cells are known to favor glycolysis over OXPHOS even under aerobic conditions (the Warburg effect).</p>
<center>
<div class="figure">
<img src="https://www.researchgate.net/profile/Tom_Thomas14/publication/318647071/figure/fig2/AS:662896304467969@1535058203439/Cancer-stem-cell-metabolism-may-demonstrate-another-element-of-cellular-plasticity-that.png" alt="Thomas and Yu, 2017" style="width:50.0%;height:50.0%" />
<p class="caption">Thomas and Yu, 2017</p>
</div>
</center>
<p>Other studies report a correlation between the activation of the unfolded protein response (UPR) and tumor progression (Madden et al., 2019; Sisinni et al., 2019). UPR is an ER stress response mechanism that can help to restore ER homeostasis or induce apoptosis under overloaded ER stress. ER stress can be induced by adverse conditions such as the accumulation of ROS, nutrient deprivation, or mutations that lead to misfolded proteins in the ER, and UPR helps relieve this stress by fixing misfolded and unfolded proteins through three main pathways.</p>
<ul>
<li>PERK (Protein Kinase R-like Endoplasmic Reticulum Kinase):
<ul>
<li>Global inhibition of protein translation by phosphorylating and inactivating the eukaryotic translation initiation factor 2 (eIF2)</li>
</ul></li>
<li>IRE1a (Inositol-requiring enzyme 1)
<ul>
<li>Dimeric kinase and endoribonuclease splices the mRNA of the X-box binding protein-1 (XBP1), and the changed ORF is translated into XBP1s, which serves as a transcription factor for genes involved in folding, glycosylation, and ER associated degradation.</li>
</ul></li>
<li>ATF6 (Activating Transcription Factor 6)
<ul>
<li>The transmembrane ATF6 protein gets transported to the Golgi, where proteases cleaves of off the transmembrane domain and allows cATF6 to enter into the nucleus and act as a transcription factor of UPR target genes</li>
</ul></li>
</ul>
<center>
<div class="figure">
<img src="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6125421/bin/molce-41-8-705f2.jpg" alt="So 2018" style="width:50.0%;height:50.0%" />
<p class="caption">So 2018</p>
</div>
</center>
<p>In addition to being a survival mechanism in normal cells, the dysregulation of ER stress response is reported to be one of the hallmarks of cancer, contributing to tumor progression and survival. For instance, studies report that UPR can ensure the survival of cancer cells and induce drug resistance through dormancy, angiogenesis, and the upregulation of antioxidant responses.</p>
</div>
<div id="our-purpose" class="section level3">
<h3>OUR PURPOSE</h3>
<p>Despite these findings, the actual mechanisms behind drug resistance remain to be elucidated, and many of the hypotheses are contradicted by other evidence as well.</p>
<ul>
<li>For instance, although the expression of the putative CSC marker CD133 in all DTPs from the Sharma et al paper suggests a link between stemness and resistance, they also observed that CSC-specific markers are lost during the transition of DTPs to DTEPs despite the ability of both cell populations to be equally drug insensitive.</li>
<li>Another study contradicts the hypothesis on the link between OXPHOS metabolism and drug resistance, where very different metabolic patterns are found in drug resistant cells (Morandi &amp; Indraccolo, 2017). In fact, a recent study in 2019 showed that cancer cells are able to shift between OXPHOS and glycolysis in response to the inhibition of either process (Elgendy et al. 2019).</li>
</ul>
<p>Together, these findings reveal the complex mechanisms that confer drug resistance in cancer cells, which are in all likelihood different in different cancer types, tumor microenvironment, and the drug used. The eradication of DTPs is the first step to overcoming resistance and preventing cancer relapse. It is therefore important for us to identify and characterize the underlying biological mechanisms that make them different from their drug sensitive counterparts and allow them to survive and proliferate. We aim to confirm the results found in previous papers and identify epigenetic differences in specific cell cycle and metabolic gene-sets by analyzing the gene expression profile of high grade serous ovarian cancer cells across different treatment time-points through Gene ontology (GO) and Gene-set enrichment analysis (GSEA) as well as differential expression (DE) analysis of single genes. We are interested in discovering whether the links between cancer stem cell-like quiescent, metabolic plasticity specifically in OXPHOS and UPR expression, and drug resistance reported in other studies are also found in our ovarian cancer data.</p>
</div>
<div id="our-data" class="section level3">
<h3>OUR DATA</h3>
<p>Our single-cell Ovarian cancer data is kindly provided by Izar and Tirosh et al., from their paper <strong>A single-cell landscape of high-grade serous ovarian cancer</strong>.</p>
<p>For their study, they collected scRNA-seq data from three HGSOC cohorts, which includes a set of 22 ascites samples from 11 patients and three patient ascites-derived xenograft models. * 8 ascites samples out of the 22 samples were sequenced using droplet-based 10X scRNAseq, while the remaining 14 ascites samples were sequenced using plate-based single cell RNAseq (SS2). * Considering that only 1 patient (Patient 5) in the 10X cohort had samples from more than 1 treatment condition, and that 99% of the malignant cells from that patient is from the on-treatment condition, with only 8 cells from the treatment-naive condition(8 TN vs. 4885 1CCT cells), we focus our analysis on the SS2 and PDX cohorts, which have more equally distributed cells across all treatment statuses.</p>
<p><strong>OUR FOCUS</strong> * The SS2 data from Izar2020 consists of cells from 14 ascites samples from 6 individuals: - Selected for malignant EPCAM+CD24+ cells by fluorophore-activated cell sorting into 96-well plates - We are only interested in the <strong>malignant</strong> ascites population, specifically samples from <strong>patient 8 and 9</strong> because they are the only patients that have samples from different treatment statuses. * The PDX data from Izar2020 consists of <strong>only Malignant cells</strong> from three HGSOC PDX models derived from patients with different treatment histories were selected for implantation: - DF20 (BRCA WT treatment-naive, clinically platinum sensitive) - DF101 (BRCA1 mutant, 2 lines of prior therapy, clinically platinum resistant) - DF68 (BRCA1 mutant, 6 lines of prior therapy, clinically platinum resistant) * After tumors were established, animals were divided into two groups per model: - Vehicle (treated with DMSO) - Carboplatin (treated with IP carboplatin) * Carboplatin-treated mice for minimal residual disease (<strong>MRD</strong>) group were harvested for scRNA-seq * The remaining carboplatin-treated mice were harvested at endpoint (<strong>vehicle</strong>)</p>
</div>
<div id="our-analysis-and-results" class="section level3">
<h3>OUR ANALYSIS AND RESULTS</h3>
<p>Please read our README documentation on <a href="https://github.com/jgoh2/jesslyn_ovca">our GitHub project repository</a> to understand how each part of our analysis is divided and structured.</p>
<ol style="list-style-type: decimal">
<li><a href="data_summary.html">A summary of the data from Izar 2020 et al.</a></li>
<li><strong>PDX Data Analysis</strong>
<ol style="list-style-type: lower-alpha">
<li><a href="03.0_Izar2020_PDX_ExploratoryAnalysis.html">PDX Exploratory analysis</a></li>
<li><a href="03.1_Izar2020_PDX_DEAnalysis.html">PDX DE analysis</a></li>
<li><a href="03.2_Izar2020_PDX_CellCycleAnalysis.html">PDX Cell Cycle analysis</a></li>
</ol></li>
<li><strong>SS2 Data Analysis</strong>
<ol style="list-style-type: lower-alpha">
<li><a href="02.0_Izar2020_SS2_ExploratoryAnalysis.html">SS2 Exploratory analysis</a></li>
<li><a href="02.1_Izar2020_SS2_DEAnalysis.html">SS2 DE analysis</a></li>
<li><a href="02.2_Izar2020_SS2_CellCycleAnalysis.html">SS2 Cell Cycle analysis</a></li>
</ol></li>
</ol>
<p><strong>Decision Making Process</strong></p>
<ol style="list-style-type: decimal">
<li><a href="PDX_choices.html">Choices along our single-cell RNA-seq Analysis</a></li>
<li><a href="GSEA_choices.html">GSEA Choices</a></li>
</ol>
</div>
<div id="credits" class="section level2">
<h2>Credits</h2>
<p>Developed by <a href="https://github.com/jgoh2">Jesslyn Goh</a> and <a href="https://github.com/mikecuoco">Mike Cuoco</a></p>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
