---
title: "Izar 2020 PDX (Cohort 3) Load + Plot"
author: "Jesslyn Goh and Mike Cuoco"
date: "7/20/2020"
editor_options: 
  chunk_output_type: console
output:
 html_document:
  code_folding: hide
  toc_float: TRUE 
---

```{r setup, include=FALSE}
knitr::opts_knit$set(root.dir = here::here())
knitr::opts_chunk$set(echo = TRUE)
options(width=120)
```

# IZAR 2020 PDX (COHORT 3) DATA ANALYSIS - PART 2

### OVERVIEW
* The PDX data from Izar2020 consists of **only Malignant cells** from three HGSOC PDX models derived from patients with different treatment histories were selected for implantation: 
    - DF20 (BRCA WT treatment-naive, clinically platinum sensitive) 
    - DF101 (BRCA1 mutant, 2 lines of prior therapy, clinically platinum resistant) 
    - DF68 (BRCA1 mutant, 6 lines of prior therapy, clinically platinum resistant) 
* After tumors were established, animals were divided into two groups per model: 
    - Vehicle (treated with DMSO) 
    - Carboplatin (treated with IP carboplatin) 
        * Carboplatin-treated mice for minimal residual disease (**MRD**) group were harvested for scRNA-seq 
        * The remaining carboplatin-treated mice were harvested at endpoint (**vehicle**)
* In our 3-part analysis of the Izar 2020 PDX data, we are interested in identifying differentially expressed genes and hallmark genesets between treatment statuses **within** each model. 
* We split our PDX analysis into three parts: 
    1. **Load Data and Create PDX Seurat Object**
        a. The code to this part of our analysis is in the **read_Izar_2020.R** file in the **code** folder. During this part of our analysis we: 
            i. Load in PDX count matrix and Create Seurat Object 
            ii. Assign Metadata identities including: 
                - Mouse ID 
                - Model ID 
                - Treatment Status 
            iii. Score cells for **cell cycle** and **hallmark genesets** 
                - **Note**: It does not matter whether we call AddModuleScore before or after subsetting and scaling each model because AddModuleScore uses the **data** slot. 
            iv. Save Seurat Object
          
    2. **Process Data and Exploratory Data Analysis**
        a. The code to this part of our analysis can be found in the **03_Izar2020_PDX_Load** file in the **old/edited** folder. During this part of our analysis we: 
            i. Load in PDX Seurat Object from Part 1 and **subset by model**. Continue analysis separately for each model. 
            ii. Scale and FindVariableFeatures (prepares data for dimensionality reduction)
            iii. Dimensionality Reduction (PCA + UMAP)
            iv. Visualize how cells separate based on metadata identities via UMAP 
                  - Intermodel heterogeneity: How do cells separate by model? 
                  - Intramodel heterogeneity: 
                      * How do cells separate by treament status? 
                      * How do cells separate by cell cycle phase? 
            v. Compute summary metrics for PDX data such as:
                - Number of cells per model per treatment 
                - Number of cells per treatment per cell cycle phase 
            vi. Save Seurat Objects 
            
    3. **DE Analysis**
        a. **TYPE #1 DE ANALYSIS**: Visualizing and Quantifying Differentially Expression on **Predefined** GO Genesets
            i. Violin Plots and UMAP 
            ii. Gene Set Enrichment Analysis (GSEA)
        b. **TYPE #2 DE ANALYSIS**: Finding DE Genes from scratch
            i. Volcano Plots 
        c. **CELL CYCLE ANALYSIS**
            i. Evaluate the idea that cell cycle might influence expression of signatures 

This is the second part of our 3-part analysis of the Izar 2020 PDX (Cohort 3) data. 

### STEP 1 LOAD SEURAT OBJECT AND GENESETS

```{r init, message = FALSE, warning = FALSE}
# Load packages
source(here::here('packages.R'))

#Read in PDX RDS object
PDX_All = readRDS("data/Izar_2020/Izar_2020_PDX.RDS")

# Read in gene lists
ccgenes = read_lines("data/gene_lists/regev_lab_cell_cycle_genes.txt")
s.genes <- ccgenes[1:43]
g2m.genes <- ccgenes[44:97]

#Read in hallmarks of interest
hallmark_names = read_lines("data/gene_lists/hallmarks.txt")
hallmark.list <- vector(mode = "list", length = length(hallmark_names) + 4)
names(hallmark.list) <- c(hallmark_names, "GO.OXPHOS", "KEGG.OXPHOS", "UNUPDATED.OXPHOS", "UNUPDATED.UPR")
for(hm in hallmark_names){
  file <- read_lines(glue("data/gene_lists/hallmarks/{hm}_updated.txt"), skip = 1)
  hallmark.list[[hm]] <- file
}

hallmark.list[["GO.OXPHOS"]] <- read_lines("data/gene_lists/extra/GO.OXPHOS.txt", skip = 1)
hallmark.list[["KEGG.OXPHOS"]] <- read_lines("data/gene_lists/extra/KEGG.OXPHOS.txt", skip = 2)
hallmark.list[["UNUPDATED.OXPHOS"]] <- read_lines("data/gene_lists/oxphos.txt", skip =1)
hallmark.list[["UNUPDATED.UPR"]] <- read_lines("data/gene_lists/upr.txt", skip =1)
```

### STEP 2 SET LEVELS AND EXAMINE METRICS
```{r metrics, message=FALSE, warning=FALSE, fig.align='center'}
PDX_All$treatment.status = factor(PDX_All$treatment.status, levels = c("vehicle", "MRD", "relapse"))
PDX_All$model_ID = factor(PDX_All$model_ID, levels = c("DF20", "DF101", "DF68"))

#number of cells in each treatment level in each model 
numCellsPerTreatmentPerModel = table(PDX_All$treatment.status, PDX_All$model_ID)
numCellsPerTreatmentPerModel %>% 
  as.data.frame() %>% 
  dplyr::rename(treatmentStatus = Var1, model = Var2, numCells = Freq) %>% 
  ggplot(aes(x=model, fill=treatmentStatus, y=numCells)) +
  geom_bar(stat="identity", position="fill") + labs(fill = "treatmentStatus", x = "model")

numCellsPerTreatmentPerModel %>% 
  as.data.frame() %>% 
  dplyr::rename(treatmentStatus = Var1, model = Var2, numCells = Freq) %>% 
  ggplot(aes(fill=model, x=treatmentStatus, y=numCells)) +
  geom_bar(stat="identity", position="fill") + labs(fill="model", x = "treatment")

#create table 
seurat_tab(PDX_All, col_var = "model_ID", row_var = "treatment.status", title = "Number of cells per model per treatment status") %>% 
  gtsave("PDX_numCellsPerTreatmentPerModel.png", path = "jesslyn_plots/PDX")
```

### STEP 3 SUBSET SEURAT OBJECT BY MODEL
```{r subset, message=FALSE, warning=FALSE}
PDX_DF20 <- subset(PDX_All, subset = (model_ID == "DF20"))
PDX_DF101 <- subset(PDX_All, subset = (model_ID == "DF101"))
PDX_DF68 <- subset(PDX_All, subset = (model_ID == "DF68"))
```

### STEP 4 SCALE EACH SEURAT OBJECT + PREPARE FOR PCA
```{r scale, message=FALSE, warning=FALSE}
#scale data
PDX_All <- ScaleData(PDX_All, features = rownames(PDX_All), do.scale = T, do.center = T)
PDX_DF20 <- ScaleData(PDX_DF20, features = rownames(PDX_DF20), do.scale = T, do.center = T)
PDX_DF101 <- ScaleData(PDX_DF101, features = rownames(PDX_DF101), do.scale = T, do.center = T)
PDX_DF68 <- ScaleData(PDX_DF68, features = rownames(PDX_DF68), do.scale = T, do.center = T)

#feature selection (prepare for dimensionality reduction)
PDX_All <- FindVariableFeatures(PDX_All, selection.method = "vst")
PDX_DF20 <- FindVariableFeatures(PDX_DF20, selection.method = "vst")
PDX_DF101 <- FindVariableFeatures(PDX_DF101, selection.method = "vst")
PDX_DF68 <- FindVariableFeatures(PDX_DF68, selection.method = "vst")
```

### STEP 5 DIMENSIONALITY REDUCTION
This step helps reduce all of our features to ones that contribute most to the variability between cells. 
  1) We would first need to find the Variable Features by running FindVariableFeatures 
  2) run PCA 
```{r PCA, message = FALSE}
#PCA
PDX_All <- RunPCA(PDX_All, features = VariableFeatures(PDX_All))
PDX_DF20 <- RunPCA(PDX_DF20, features = VariableFeatures(PDX_DF20))
PDX_DF101 <- RunPCA(PDX_DF101, features = VariableFeatures(PDX_DF101))
PDX_DF68 <- RunPCA(PDX_DF68, features = VariableFeatures(PDX_DF68))

#UMAP 
PDX_All<- RunUMAP(PDX_All, dims=1:15)
PDX_DF20<- RunUMAP(PDX_DF20, dims=1:15)
PDX_DF101<- RunUMAP(PDX_DF101, dims=1:15)
PDX_DF68<- RunUMAP(PDX_DF68, dims=1:15)
```

### STEP 7 SAVE SEURAT OBJECT
```{r save RDS}
saveRDS(PDX_All, file="data/Izar_2020/test/jesslyn_PDX_All_processed.RDS")
saveRDS(PDX_DF20, file="data/Izar_2020/test/jesslyn_PDX_DF20_processed.RDS")
saveRDS(PDX_DF101, file="data/Izar_2020/test/jesslyn_PDX_DF101_processed.RDS")
saveRDS(PDX_DF68, file="data/Izar_2020/test/jesslyn_PDX_DF68_processed.RDS")
```

