---
title: "Izar 2020 SS2 (Cohort 2) Cell Cycle Analysis"
author: "Jesslyn Goh and Mike Cuoco"
date: "7/20/2020"
editor_options: 
  chunk_output_type: console
output:
 html_document:
  code_folding: hide
  toc_float: TRUE 
---

```{r setup, include=FALSE}
knitr::opts_knit$set(root.dir = here::here())
knitr::opts_chunk$set(echo = TRUE)
options(width=120)
```

# IZAR 2020 SS2 (COHORT 2) DATA ANALYSIS - PART 2

### OVERVIEW
* The SS2 data from Izar2020 consists of cells from 14 ascites samples from 6 individuals: 
    - selected for malignant EPCAM+CD24+ cells by fluorophore-activated cell sorting into 96-well plates
    - Includes both **malignant** and **nonmalignant** populations but differs from 10X data - examines malignant ascites more closely. 
    - We are only interested in the **malignant** ascites population, specifically samples from **patient 8 and 9** because they are the only patients that have samples from different treatment statuses. 
    
* In our 5-part analysis of the Izar 2020 SS2 data, we are interested in identifying differentially expressed genes and hallmark genesets between treatment statuses **within** patients 8 and 9 
* We split our SS2 analysis into three parts: 
    1. **Load Data and Create SS2 Seurat Object**
        a. The code to this part of our analysis is in the **read_Izar_2020.R** file in the **code** folder. During this part of our analysis we: 
            i. Load in SS2 count matrix and Create Seurat Object 
            ii. Assign Metadata identities including: 
                - Patient ID
                - Time
                - Sample ID
                - Treatment Status 
            iii. Score cells for **cell cycle** and **hallmark genesets** 
                - **Note**: It does not matter whether we call AddModuleScore before or after subsetting and scaling each model because AddModuleScore uses the **data** slot. 
            iv. Save Seurat Object
          
    2. **Process Data and Exploratory Data Analysis**
        a. The code to this part of our analysis can be found in the **02_Izar2020_SS2_Load_Plots** file in the **old/edited** folder. During this part of our analysis we: 
            i. Load in SS2 Seurat Object from Part 1 and **subset by patients 8 and 9**. Continue analysis separately for each model. 
            ii. Scale and FindVariableFeatures (prepares data for dimensionality reduction)
            iii. Dimensionality Reduction (PCA + UMAP)
            vi. Save Seurat Objects 
      
    3. **Exploratory Data Analysis**
        a. The code to this part of our analysis can be found in the **02.0_Izar2020_SS2_Exploratory Analysis** file in the **analysis** folder. During this part of our analysis we: 
            i. Load in SS2 Seurat Object from Part 2. Analyze separately for patients 8 and 9
            ii. Compute summary metrics for SS2 data such as: 
                - Number of cells per patient per treatment 
                - Number of cells per treatment per cell cycle phase 
            iii. Visualize how cells separate based on metadata identities via UMAP and PCA
                - Intermodel heterogeneity: How do cells separate by sample? 
                - Intramodel heterogeneity: 
                    * How do cells separate by treament status / sample? 
                    * How do cells separate by cell cycle phase?
            
    4. **DE Analysis**
        a. **TYPE #1 DE ANALYSIS**: Visualizing and Quantifying Differentially Expression on **Predefined** GO Genesets
            i. Violin Plots and UMAP 
            ii. Gene Set Enrichment Analysis (GSEA)
        b. **TYPE #2 DE ANALYSIS**: Finding DE Genes from scratch
            i. Volcano Plots 
          
    5. **CELL CYCLE ANALYSIS**
        a. **TYPE #1 CELL CYCLE ANALYSIS**: Examine correlation between treatment condition and cell cycle phase
        b. Evaluate the idea that cell cycle might influence expression of signatures 

This is the third part of our 5-part analysis of the Izar 2020 SS2 (Cohort 2) data. 

### CELL CYCLE ANALYSIS IN-DEPTH EXPLANATION 
We are interested in answering a few questions for our Cell Cycle Analysis:

**CELL CYCLE ANALYSIS** 

* **QUESTION #1** Are more cells at different cell cycle phases between treatments? 
    - **APPROACH** Violin Plots and UMAP 
        * Visualize differences in cell cycle score across treatment conditions with: 
            - Stacked Bar Plot
            - Violin Plot 
        * Statistical test: is the difference in cell cycle scores across treatment conditions statistically
* **QUESTION #2** Is there a correlation between cell cycle phase and the expression of signatures? 
    - **APPROACH** Violin Plots
        * Create Violin Plots for each hallmark of interest like **DE Analysis #1/APPROACH #1. 
        * Stratify the Violin Plots by cell cycle phase

### STEP 1 LOAD SEURAT OBJECT

```{r init, message = FALSE, warning = FALSE}
# Load packages
source(here::here('packages.R'))

#Read in SS2 RDS object
SS2Malignant = readRDS(file = "data/Izar_2020/jesslyn_SS2Malignant_processed.RDS")
SS2Malignant.8 = readRDS(file = "data/Izar_2020/jesslyn_SS2Malignant8_processed.RDS")
SS2Malignant.9 = readRDS(file = "data/Izar_2020/jesslyn_SS2Malignant9_processed.RDS")

# Read in ccgenes
ccgenes = read_lines("data/gene_lists/regev_lab_cell_cycle_genes.txt")
s.genes <- ccgenes[1:43]
g2m.genes <- ccgenes[44:97]

#Read in hallmarks of interest
g0g1.genes <- read_lines("data/gene_lists/cellcycle/G0G1.txt", skip = 2)
g1S.DNAdamage.genes <- read_lines("data/gene_lists/cellcycle/G1SDNAdamage.txt", skip = 2)
g1S.transcription.genes <- read_lines("data/gene_lists/cellcycle/G1Stranscription.txt", skip = 2)
g2m.checkpoint.genes <- read_lines("data/gene_lists/cellcycle/G2Mcheckpoint.txt", skip = 2)
g2m.DNAdamage.genes <- read_lines("data/gene_lists/cellcycle/G2MDNAdamage.txt", skip = 2)
g2m.replicationCP.genes <- read_lines("data/gene_lists/cellcycle/G2MreplicationCheckpoint.txt", skip = 2)

hallmark_names = read_lines("data/gene_lists/hallmarks.txt")
hallmark.list <- vector(mode = "list", length = length(hallmark_names) + 9)
names(hallmark.list) <- c(hallmark_names, "GO.OXPHOS", "KEGG.OXPHOS", "UNUPDATED.OXPHOS", 
                          "UNUPDATED.UPR", "G0G1", "G1S.DNAdamage", "G1S.transcription", "G2M.checkpoint", 
                          "G2M.DNAdamage")
for(hm in hallmark_names){
  file <- read_lines(glue("data/gene_lists/hallmarks/{hm}_updated.txt"), skip = 1)
  hallmark.list[[hm]] <- file
}

hallmark.list[["GO.OXPHOS"]] <- read_lines("data/gene_lists/extra/GO.OXPHOS.txt", skip = 1)
hallmark.list[["KEGG.OXPHOS"]] <- read_lines("data/gene_lists/extra/KEGG.OXPHOS.txt", skip = 2)
hallmark.list[["UNUPDATED.OXPHOS"]] <- read_lines("data/gene_lists/oxphos.txt", skip =1)
hallmark.list[["UNUPDATED.UPR"]] <- read_lines("data/gene_lists/upr.txt", skip =1)
hallmark.list[["G0G1"]] <- g0g1.genes
hallmark.list[["G1S.DNAdamage"]] <- g1S.DNAdamage.genes
hallmark.list[["G1S.transcription"]] <- g1S.transcription.genes
hallmark.list[["G2M.checkpoint"]] <- g2m.checkpoint.genes
hallmark.list[["G2M.DNAdamage"]] <- g2m.DNAdamage.genes

#center module and cell cycle scores and reassign to the metadata of each Seurat object
hm.names <- names(SS2Malignant@meta.data)[14:58]

for(i in hm.names){
    SS2Malignant.hm.centered <- scale(SS2Malignant[[i]], center = TRUE, scale = FALSE)
    SS2Malignant <- AddMetaData(SS2Malignant, SS2Malignant.hm.centered, col.name = glue("{i}.centered"))
    
    SS2Malignant8.hm.centered <- scale(SS2Malignant.8[[i]], center = TRUE, scale = FALSE)
    SS2Malignant.8 <- AddMetaData(SS2Malignant.8, SS2Malignant8.hm.centered, col.name = glue("{i}.centered"))
    
    SS2Malignant9.hm.centered <- scale(SS2Malignant.9[[i]], center = TRUE, scale = FALSE)
    SS2Malignant.9 <- AddMetaData(SS2Malignant.9, SS2Malignant9.hm.centered, col.name = glue("{i}.centered"))
}
```

### STEP 2 CELL CYCLE ANALYSIS
In our **Exploratory Analysis** section, we observed that PC_2 captures Cell Cycle Phase, S.Score and G2M scores, as cells in the same phase or have high S or G2M scores are found to be near each other on the PC_2 Axis. However, cells did not seem to cluster by treatment status. We now use a more quantitative approach to investigate the correlation between treatment and cell cycle phase. 

* **QUESTION #1** Are more cells at different cell cycle phases between treatments? 
* **HYPOTHESIS** There should be less cycling cells in the on-treatment conditions (**p.1 and p.2**) relative to the treatment-naive condition (**p.0**)
    - **APPROACH** Violin Plots and UMAP 
        * Visualize differences in cell cycle score across treatment conditions with: 
            - Stacked Bar Plot
            - Violin Plot 
        * Statistical test: is the difference in cell cycle scores across treatment conditions statistically significant. 
        
```{r cellcycle1, message = FALSE, warning = FALSE, fig.align = 'center', fig.width = 10, fig.height = 5}

#Stacked Bar Plot --------------------------
SS2.names <- c("SS2 Patient 8", "SS2 Patient 9")
SS2s <- c(SS2Malignant.8, SS2Malignant.9)
bar.plots <- vector("list", length = 3)
names(bar.plots) <- SS2.names

for(i in 1:length(SS2s)){
  obj = SS2s[[i]]
  name = SS2.names[[i]]
  numCells = nrow(obj@meta.data)
  
  obj$Phase <- factor(obj$Phase, levels = c("G1", "S", "G2M"))
  t = table(obj$Phase, obj$sample) %>% 
    as.data.frame() %>% 
    rename(Phase = Var1, Sample = Var2, numCells = Freq)

sum = c()
z = 1
while(z < nrow(t)){
    n = t$numCells[z] + t$numCells[z+1] + t$numCells[z+2]
    sum = c(sum, rep(n,3))
    z = z + 3
  }
  
t <- t %>% mutate(total = sum) %>% mutate(percent = (numCells/total)*100)
  
  p = t %>% 
    ggplot(aes(x=Sample, y=percent, fill=Phase)) + 
    geom_bar(stat="identity") + 
    labs(title = glue("{name} % of Malignant Cells in Each Cell Cycle Phase/ Treatment"), subtitle = glue("{numCells} Malignant Cells"), caption = "Labeled by # cells in each phase per treatment") +
    theme(plot.title = element_text(size = 10), plot.subtitle = element_text(size = 10), plot.caption = element_text(size = 8)) +
    geom_text(aes(label = paste(numCells, "cells")), position = "stack", hjust = 0.5, vjust = 2, size = 3, color = "white") + 
    scale_fill_manual(values = c("#00AFBB", "#E7B800", "#FC4E07"))
    
  bar.plots[[i]] <- p
}

bar.plots[["SS2 Patient 8"]] + bar.plots[["SS2 Patient 9"]] + plot_layout(guides= 'collect')
```

* **OBSERVATIONS** 
    - **Patient 8** % Noncyclying cells 8.0 > 8.1 > 8.2 (does not support our hypothesis, on-treatment samples should have more noncycling cells)
    - **Patient 9** % Noncycling cells 9.2 > 9.1 > 9.0 (supports our hypothesis, not sure how to interpret 9.2 > 9.1)
    - Important to note that we have very little cells in each condition, so we have really little power and the results from our sample may not be representative of the true population. 

To investigate this further, we build violin plots of cell cycle scores separated by treatment status: 
    - Noncycling Scores
        - G0 / early G1
        - G1/S DNA Damage Checkpoint 
        - G2M Checkpoint 
        - G2M DNA Damage Checkpoint 
    - Cycling Scores 
        - G1/S Specific Transcription 
        - S Score
        - G2M Score

**1) NONCYCLING SCORES**

```{r noncycling, message=FALSE, warning=FALSE, fig.align='center', fig.width=12, fig.height=8}
nc <- c("G0G139.centered", "G1S.DNAdamage40.centered", "G2M.checkpoint42.centered", "G2M.DNAdamage43.centered")
nc.names <- c("G0/G1", "G1/S DNA Damage CP", "G2M CP", "G2M DNA Damage CP")
patient <- c("8", "9")

# G0G1 -------------
g0g1.swarm <- vector("list", length = 2)
names(g0g1.swarm) <- SS2.names

for(i in 1:2){
  SS2 <- SS2s[[i]]
  SS2.name <- SS2.names[[i]]
  
  my_comparisons <- list(
    c(glue("{patient[[i]]}.0"), glue("{patient[[i]]}.1")), 
    c(glue("{patient[[i]]}.1"), glue("{patient[[i]]}.2")), 
    c(glue("{patient[[i]]}.0"), glue("{patient[[i]]}.2"))
  )
  
  feature.found <- sum(hallmark.list[["G0G1"]] %in% rownames(SS2))
  feature.length <- length(hallmark.list[["G0G1"]])
  feature.pFound <- round((feature.found / feature.length)*100, 2)
  numCells <- nrow(SS2@meta.data)
  
  p <- ggplot(SS2@meta.data, aes(x= sample, y = G0G139.centered, color = sample)) +
  geom_quasirandom(groupOnX =TRUE) + 
  theme_bw() + 
  scale_color_manual(values = c("#00AFBB", "#E7B800", "#FC4E07")) + 
    labs(title = glue("{SS2.name} Malignant Cells G0/G1 expression across sample"), x = SS2.name, subtitle = glue("{numCells} Malignant Cells, {feature.found} out of {feature.length} G0/G1 genes found ({feature.pFound}%)")) + 
    theme(plot.title = element_text(size =10), plot.subtitle = element_text(size = 8)) + 
  geom_boxplot(width = 0.10, position = position_dodge(0.9), alpha = 1, show.legend = F, color = "black", outlier.alpha = 0) + 
    geom_text(label = paste(sum(str_detect(SS2$sample, "[:digit:].0")), "cells"), x = glue("{patient[[i]]}.0"), y = min(SS2$G0G139.centered) -0.03, color = "black", show.legend = FALSE) + 
    geom_text(label = paste(sum(str_detect(SS2$sample, "[:digit:].1")), "cells"), x = glue("{patient[[i]]}.1"), y = min(SS2$G0G139.centered) - 0.03, color = "black", show.legend = FALSE) + 
    geom_text(label = paste(sum(str_detect(SS2$sample, "[:digit:].2")), "cells"), x = glue("{patient[[i]]}.2"), y = min(SS2$G0G139.centered) - 0.03, color = "black", show.legend = FALSE) + 
    stat_compare_means(comparisons = my_comparisons, method = "wilcox.test", label = "p.format", step.increase = 0.06) + 
    stat_compare_means(comparisons = my_comparisons, method = "wilcox.test", label = "p.signif", step.increase = 0.06, bracket.size = 0, vjust = 1.8)
  
  g0g1.swarm[[SS2.name]] <- p 
}

g0g1.swarm[["SS2 Patient 8"]] + g0g1.swarm[["SS2 Patient 9"]]

# G1/S DNA Damage -------------
g1s.swarm <- vector("list", length = 2)
names(g1s.swarm) <- SS2.names

for(i in 1:2){
  SS2 <- SS2s[[i]]
  SS2.name <- SS2.names[[i]]
  
  my_comparisons <- list(
    c(glue("{patient[[i]]}.0"), glue("{patient[[i]]}.1")), 
    c(glue("{patient[[i]]}.1"), glue("{patient[[i]]}.2")), 
    c(glue("{patient[[i]]}.0"), glue("{patient[[i]]}.2"))
  )
  
  feature.found <- sum(hallmark.list[["G1S.DNAdamage"]] %in% rownames(SS2))
  feature.length <- length(hallmark.list[["G1S.DNAdamage"]])
  feature.pFound <- round((feature.found / feature.length)*100, 2)
  numCells <- nrow(SS2@meta.data)
  
  p <- ggplot(SS2@meta.data, aes(x= sample, y = G1S.DNAdamage40.centered, color = sample)) +
  geom_quasirandom(groupOnX =TRUE) + 
  theme_bw() + 
  scale_color_manual(values = c("#00AFBB", "#E7B800", "#FC4E07")) + 
    labs(title = glue("{SS2.name} Malignant Cells G1/S DNA Damage CP expression across sample"), x = SS2.name, subtitle = glue("{numCells} Malignant Cells, {feature.found} out of {feature.length} G1/S DNA Damage CP genes found ({feature.pFound}%)")) + 
    theme(plot.title = element_text(size =10), plot.subtitle = element_text(size = 8)) + 
  geom_boxplot(width = 0.10, position = position_dodge(0.9), alpha = 1, show.legend = F, color = "black", outlier.alpha = 0) + 
    geom_text(label = paste(sum(str_detect(SS2$sample, "[:digit:].0")), "cells"), x = glue("{patient[[i]]}.0"), y = min(SS2$G1S.DNAdamage40.centered) -0.03, color = "black", show.legend = FALSE) + 
    geom_text(label = paste(sum(str_detect(SS2$sample, "[:digit:].1")), "cells"), x = glue("{patient[[i]]}.1"), y = min(SS2$G1S.DNAdamage40.centered) - 0.03, color = "black", show.legend = FALSE) + 
    geom_text(label = paste(sum(str_detect(SS2$sample, "[:digit:].2")), "cells"), x = glue("{patient[[i]]}.2"), y = min(SS2$G1S.DNAdamage40.centered) - 0.03, color = "black", show.legend = FALSE) + 
    stat_compare_means(comparisons = my_comparisons, method = "wilcox.test", label = "p.format", step.increase = 0.06) + 
    stat_compare_means(comparisons = my_comparisons, method = "wilcox.test", label = "p.signif", step.increase = 0.06, bracket.size = 0, vjust = 1.8)
  
  g1s.swarm[[SS2.name]] <- p 
}

g1s.swarm[["SS2 Patient 8"]] + g1s.swarm[["SS2 Patient 9"]]

# G2M CP -------------
g2m.swarm <- vector("list", length = 2)
names(g2m.swarm) <- SS2.names

for(i in 1:2){
  SS2 <- SS2s[[i]]
  SS2.name <- SS2.names[[i]]
  
  my_comparisons <- list(
    c(glue("{patient[[i]]}.0"), glue("{patient[[i]]}.1")), 
    c(glue("{patient[[i]]}.1"), glue("{patient[[i]]}.2")), 
    c(glue("{patient[[i]]}.0"), glue("{patient[[i]]}.2"))
  )
  
  feature.found <- sum(hallmark.list[["G2M.checkpoint"]] %in% rownames(SS2))
  feature.length <- length(hallmark.list[["G2M.checkpoint"]])
  feature.pFound <- round((feature.found / feature.length)*100, 2)
  numCells <- nrow(SS2@meta.data)
  
  p <- ggplot(SS2@meta.data, aes(x= sample, y = G2M.checkpoint42.centered, color = sample)) +
  geom_quasirandom(groupOnX =TRUE) + 
  theme_bw() + 
  scale_color_manual(values = c("#00AFBB", "#E7B800", "#FC4E07")) + 
    labs(title = glue("{SS2.name} Malignant Cells G2M CP expression across sample"), x = SS2.name, subtitle = glue("{numCells} Malignant Cells, {feature.found} out of {feature.length} G2M CP genes found ({feature.pFound}%)")) + 
    theme(plot.title = element_text(size =10), plot.subtitle = element_text(size = 8)) +
  geom_boxplot(width = 0.10, position = position_dodge(0.9), alpha = 1, show.legend = F, color = "black", outlier.alpha = 0) + 
    geom_text(label = paste(sum(str_detect(SS2$sample, "[:digit:].0")), "cells"), x = glue("{patient[[i]]}.0"), y = min(SS2$G2M.checkpoint42.centered) -0.03, color = "black", show.legend = FALSE) + 
    geom_text(label = paste(sum(str_detect(SS2$sample, "[:digit:].1")), "cells"), x = glue("{patient[[i]]}.1"), y = min(SS2$G2M.checkpoint42.centered) - 0.03, color = "black", show.legend = FALSE) + 
    geom_text(label = paste(sum(str_detect(SS2$sample, "[:digit:].2")), "cells"), x = glue("{patient[[i]]}.2"), y = min(SS2$G2M.checkpoint42.centered) - 0.03, color = "black", show.legend = FALSE) + 
    stat_compare_means(comparisons = my_comparisons, method = "wilcox.test", label = "p.format", step.increase = 0.06) + 
    stat_compare_means(comparisons = my_comparisons, method = "wilcox.test", label = "p.signif", step.increase = 0.06, bracket.size = 0, vjust = 1.8)
  
  g2m.swarm[[SS2.name]] <- p 
}

g2m.swarm[["SS2 Patient 8"]] + g2m.swarm[["SS2 Patient 9"]]

# G2M DNA Damage CP -------------
g2m.dna.swarm <- vector("list", length = 2)
names(g2m.dna.swarm) <- SS2.names

for(i in 1:2){
  SS2 <- SS2s[[i]]
  SS2.name <- SS2.names[[i]]
  
  my_comparisons <- list(
    c(glue("{patient[[i]]}.0"), glue("{patient[[i]]}.1")), 
    c(glue("{patient[[i]]}.1"), glue("{patient[[i]]}.2")), 
    c(glue("{patient[[i]]}.0"), glue("{patient[[i]]}.2"))
  )
  
  feature.found <- sum(hallmark.list[["G2M.DNAdamage"]] %in% rownames(SS2))
  feature.length <- length(hallmark.list[["G2M.DNAdamage"]])
  feature.pFound <- round((feature.found / feature.length)*100, 2)
  numCells <- nrow(SS2@meta.data)
  
  p <- ggplot(SS2@meta.data, aes(x= sample, y = G2M.DNAdamage43.centered, color = sample)) +
  geom_quasirandom(groupOnX =TRUE) + 
  theme_bw() + 
  scale_color_manual(values = c("#00AFBB", "#E7B800", "#FC4E07")) + 
    labs(title = glue("{SS2.name} Malignant Cells G2M DNA Damage CP expression across sample"), x = SS2.name, subtitle = glue("{numCells} Malignant Cells, {feature.found} out of {feature.length} G2M DNA Damage CP genes found ({feature.pFound}%)")) + 
    theme(plot.title = element_text(size =10), plot.subtitle = element_text(size = 8)) + 
  geom_boxplot(width = 0.10, position = position_dodge(0.9), alpha = 1, show.legend = F, color = "black", outlier.alpha = 0) + 
    geom_text(label = paste(sum(str_detect(SS2$sample, "[:digit:].0")), "cells"), x = glue("{patient[[i]]}.0"), y = min(SS2$G2M.DNAdamage43.centered) -0.03, color = "black", show.legend = FALSE) + 
    geom_text(label = paste(sum(str_detect(SS2$sample, "[:digit:].1")), "cells"), x = glue("{patient[[i]]}.1"), y = min(SS2$G2M.DNAdamage43.centered) - 0.03, color = "black", show.legend = FALSE) + 
    geom_text(label = paste(sum(str_detect(SS2$sample, "[:digit:].2")), "cells"), x = glue("{patient[[i]]}.2"), y = min(SS2$G2M.DNAdamage43.centered) - 0.03, color = "black", show.legend = FALSE) + 
    stat_compare_means(comparisons = my_comparisons, method = "wilcox.test", label = "p.format", step.increase = 0.06) + 
    stat_compare_means(comparisons = my_comparisons, method = "wilcox.test", label = "p.signif", step.increase = 0.06, bracket.size = 0, vjust = 1.8)
  
  g2m.dna.swarm[[SS2.name]] <- p 
}

g2m.dna.swarm[["SS2 Patient 8"]] + g2m.dna.swarm[["SS2 Patient 9"]]

```

We hypothesized that samples p.1 will have enriched levels of non/low-cycling genes. 

* Statistically significant results 
    - G0/G1
        * 8.2 > 8.0
    - G1/S DNA Damage Checkpoint 
        * 9.0 > 9.2 
        * 9.1 > 9.2 (supports our hypothesis)
    - G2M Checkpoint 
        * 9.0 > 9.2 
        * 9.1 > 9.2 (supports our hypothesis)
    - G2M DNA Damage Checkpoint 
        * 9.0 > 9.2
    
**2) CYCLING SCORES**

- Cycling Scores 
    - G1/S Specific Transcription 
    - S Score
    - G2M Score

```{r cellcycleVln, message=FALSE, warning=FALSE, fig.align='center', fig.width=12, fig.height=8}
#VlnPlot -------------------------
cc <- c("G1S.transcription41.centered", "S.Score.centered", "G2M.Score.centered")
cc.names <- c("G1/S Specific Transcription", "S Score", "G2M Score")

# G1/S --------
g1s.transcription.swarm <- vector("list", length = 2)
names(g1s.transcription.swarm) <- SS2.names

for(i in 1:2){
  SS2 <- SS2s[[i]]
  SS2.name <- SS2.names[[i]]
  
  my_comparisons <- list(
    c(glue("{patient[[i]]}.0"), glue("{patient[[i]]}.1")), 
    c(glue("{patient[[i]]}.1"), glue("{patient[[i]]}.2")), 
    c(glue("{patient[[i]]}.0"), glue("{patient[[i]]}.2"))
  )
  
  feature.found <- sum(hallmark.list[["G1S.transcription"]] %in% rownames(SS2))
  feature.length <- length(hallmark.list[["G1S.transcription"]])
  feature.pFound <- round((feature.found / feature.length)*100, 2)
  numCells <- nrow(SS2@meta.data)
  
  p <- ggplot(SS2@meta.data, aes(x= sample, y = G1S.transcription41.centered, color = sample)) +
  geom_quasirandom(groupOnX =TRUE) + 
  theme_bw() + 
  scale_color_manual(values = c("#00AFBB", "#E7B800", "#FC4E07")) + 
    labs(title = glue("{SS2.name} Malignant Cells G1/S Specific Transcription expression across sample"), x = SS2.name, subtitle = glue("{numCells} Malignant Cells, {feature.found} out of {feature.length} G1/S genes found ({feature.pFound}%)")) + 
    theme(plot.title = element_text(size =10), plot.subtitle = element_text(size = 8)) + 
  geom_boxplot(width = 0.10, position = position_dodge(0.9), alpha = 1, show.legend = F, color = "black", outlier.alpha = 0) + 
    geom_text(label = paste(sum(str_detect(SS2$sample, "[:digit:].0")), "cells"), x = glue("{patient[[i]]}.0"), y = min(SS2$G1S.transcription41.centered) -0.03, color = "black", show.legend = FALSE) + 
    geom_text(label = paste(sum(str_detect(SS2$sample, "[:digit:].1")), "cells"), x = glue("{patient[[i]]}.1"), y = min(SS2$G1S.transcription41.centered) - 0.03, color = "black", show.legend = FALSE) + 
    geom_text(label = paste(sum(str_detect(SS2$sample, "[:digit:].2")), "cells"), x = glue("{patient[[i]]}.2"), y = min(SS2$G1S.transcription41.centered) - 0.03, color = "black", show.legend = FALSE) + 
    stat_compare_means(comparisons = my_comparisons, method = "wilcox.test", label = "p.format", step.increase = 0.06) + 
    stat_compare_means(comparisons = my_comparisons, method = "wilcox.test", label = "p.signif", step.increase = 0.06, bracket.size = 0, vjust = 1.8)
  
  g1s.transcription.swarm[[SS2.name]] <- p 
}

g1s.transcription.swarm[["SS2 Patient 8"]] + g1s.transcription.swarm[["SS2 Patient 9"]]

# S --------
s.swarm <- vector("list", length = 2)
names(s.swarm) <- SS2.names

for(i in 1:2){
  SS2 <- SS2s[[i]]
  SS2.name <- SS2.names[[i]]
  
  my_comparisons <- list(
    c(glue("{patient[[i]]}.0"), glue("{patient[[i]]}.1")), 
    c(glue("{patient[[i]]}.1"), glue("{patient[[i]]}.2")), 
    c(glue("{patient[[i]]}.0"), glue("{patient[[i]]}.2"))
  )
  
  feature.found <- sum(s.genes %in% rownames(SS2))
  feature.length <- length(s.genes)
  feature.pFound <- round((feature.found / feature.length)*100, 2)
  numCells <- nrow(SS2@meta.data)
  
  p <- ggplot(SS2@meta.data, aes(x= sample, y = S.Score.centered, color = sample)) +
  geom_quasirandom(groupOnX =TRUE) + 
  theme_bw() + 
  scale_color_manual(values = c("#00AFBB", "#E7B800", "#FC4E07")) + 
    labs(title = glue("{SS2.name} Malignant Cells S Phase expression across sample"), x = SS2.name, subtitle = glue("{numCells} Malignant Cells, {feature.found} out of {feature.length} S genes found ({feature.pFound}%)")) + 
    theme(plot.title = element_text(size =10), plot.subtitle = element_text(size = 8)) + 
  geom_boxplot(width = 0.10, position = position_dodge(0.9), alpha = 1, show.legend = F, color = "black", outlier.alpha = 0) + 
    geom_text(label = paste(sum(str_detect(SS2$sample, "[:digit:].0")), "cells"), x = glue("{patient[[i]]}.0"), y = min(SS2$S.Score.centered) -0.03, color = "black", show.legend = FALSE) + 
    geom_text(label = paste(sum(str_detect(SS2$sample, "[:digit:].1")), "cells"), x = glue("{patient[[i]]}.1"), y = min(SS2$S.Score.centered) - 0.03, color = "black", show.legend = FALSE) + 
    geom_text(label = paste(sum(str_detect(SS2$sample, "[:digit:].2")), "cells"), x = glue("{patient[[i]]}.2"), y = min(SS2$S.Score.centered) - 0.03, color = "black", show.legend = FALSE) + 
    stat_compare_means(comparisons = my_comparisons, method = "wilcox.test", label = "p.format", step.increase = 0.06) + 
    stat_compare_means(comparisons = my_comparisons, method = "wilcox.test", label = "p.signif", step.increase = 0.06, bracket.size = 0, vjust = 1.8)
  
  s.swarm[[SS2.name]] <- p 
}

s.swarm[["SS2 Patient 8"]] + s.swarm[["SS2 Patient 9"]]

# G2M --------
g2m.swarm <- vector("list", length = 2)
names(g2m.swarm) <- SS2.names

for(i in 1:2){
  SS2 <- SS2s[[i]]
  SS2.name <- SS2.names[[i]]
  
  my_comparisons <- list(
    c(glue("{patient[[i]]}.0"), glue("{patient[[i]]}.1")), 
    c(glue("{patient[[i]]}.1"), glue("{patient[[i]]}.2")), 
    c(glue("{patient[[i]]}.0"), glue("{patient[[i]]}.2"))
  )
  
  feature.found <- sum(g2m.genes %in% rownames(SS2))
  feature.length <- length(g2m.genes)
  feature.pFound <- round((feature.found / feature.length)*100, 2)
  numCells <- nrow(SS2@meta.data)
  
  p <- ggplot(SS2@meta.data, aes(x= sample, y = G2M.Score.centered, color = sample)) +
  geom_quasirandom(groupOnX =TRUE) + 
  theme_bw() + 
  scale_color_manual(values = c("#00AFBB", "#E7B800", "#FC4E07")) + 
    labs(title = glue("{SS2.name} Malignant Cells G2M Phase expression across sample"), x = SS2.name, subtitle = glue("{numCells} Malignant Cells, {feature.found} out of {feature.length} G2M genes found ({feature.pFound}%)")) + 
    theme(plot.title = element_text(size =10), plot.subtitle = element_text(size = 8)) + 
  geom_boxplot(width = 0.10, position = position_dodge(0.9), alpha = 1, show.legend = F, color = "black", outlier.alpha = 0) + 
    geom_text(label = paste(sum(str_detect(SS2$sample, "[:digit:].0")), "cells"), x = glue("{patient[[i]]}.0"), y = min(SS2$G2M.Score.centered) -0.03, color = "black", show.legend = FALSE) + 
    geom_text(label = paste(sum(str_detect(SS2$sample, "[:digit:].1")), "cells"), x = glue("{patient[[i]]}.1"), y = min(SS2$G2M.Score.centered) - 0.03, color = "black", show.legend = FALSE) + 
    geom_text(label = paste(sum(str_detect(SS2$sample, "[:digit:].2")), "cells"), x = glue("{patient[[i]]}.2"), y = min(SS2$G2M.Score.centered) - 0.03, color = "black", show.legend = FALSE) + 
    stat_compare_means(comparisons = my_comparisons, method = "wilcox.test", label = "p.format", step.increase = 0.06) + 
    stat_compare_means(comparisons = my_comparisons, method = "wilcox.test", label = "p.signif", step.increase = 0.06, bracket.size = 0, vjust = 1.8)
  
  g2m.swarm[[SS2.name]] <- p 
}

g2m.swarm[["SS2 Patient 8"]] + g2m.swarm[["SS2 Patient 9"]]

```

* **CONCLUSIONS** 
    - Statistically significant results: 
        * G1/S Specific Transcription Score
            - 9.1 > 9.2 (rejects our hypothesis)
        * S Score
            - 8.1 > 8.0 (rejects our hypothesis)
            - 8.2 > 8.0 
        * G2M Score
            - 9.1 > 9.2 (rejects our hypothesis)
            - 9.0 > 9.2 

- Statistically significant results from both NONCYCLING and CYCLYING gene score comparisons show us that 9.1 > 9.2, which is very contradictory. 
- It is possible that there is a correlation between nFeature/nCount and gene scores, so that 9.1 expresses higher levels of both types of gene-sets simply because it has more genes in general. We use swarm plots to compare the nFeature and nCount levels across treatment condition within each patient. 

```{r nFnCSwarm, message = FALSE, warning = FALSE, fig.align = 'center', fig.width= 12, fig.height=8}
#nFeature ----
nF.swarm <- vector("list", length = 2)
names(nF.swarm) <- SS2.names

for(i in 1:2){
  SS2 <- SS2s[[i]]
  SS2.name <- SS2.names[[i]]
  
  my_comparisons <- list(
    c(glue("{patient[[i]]}.0"), glue("{patient[[i]]}.1")), 
    c(glue("{patient[[i]]}.1"), glue("{patient[[i]]}.2")), 
    c(glue("{patient[[i]]}.0"), glue("{patient[[i]]}.2"))
  )
  
  numCells <- nrow(SS2@meta.data)
  
  p <- ggplot(SS2@meta.data, aes(x= sample, y = nFeature_RNA, color = sample)) +
  geom_quasirandom(groupOnX =TRUE) + 
  theme_bw() + 
  scale_color_manual(values = c("#00AFBB", "#E7B800", "#FC4E07")) + 
    labs(title = glue("{SS2.name} Malignant Cells nFeature across sample"), x = SS2.name, subtitle = glue("{numCells} Malignant Cells")) + 
    theme(plot.title = element_text(size =10), plot.subtitle = element_text(size = 8)) + 
  geom_boxplot(width = 0.10, position = position_dodge(0.9), alpha = 1, show.legend = F, color = "black", outlier.alpha = 0) + 
    geom_text(label = paste(sum(str_detect(SS2$sample, "[:digit:].0")), "cells"), x = glue("{patient[[i]]}.0"), y = min(SS2$nFeature_RNA) -0.03, color = "black", show.legend = FALSE) + 
    geom_text(label = paste(sum(str_detect(SS2$sample, "[:digit:].1")), "cells"), x = glue("{patient[[i]]}.1"), y = min(SS2$nFeature_RNA) - 0.03, color = "black", show.legend = FALSE) + 
    geom_text(label = paste(sum(str_detect(SS2$sample, "[:digit:].2")), "cells"), x = glue("{patient[[i]]}.2"), y = min(SS2$nFeature_RNA) - 0.03, color = "black", show.legend = FALSE) + 
    stat_compare_means(comparisons = my_comparisons, method = "wilcox.test", label = "p.format", step.increase = 0.06) + 
    stat_compare_means(comparisons = my_comparisons, method = "wilcox.test", label = "p.signif", step.increase = 0.06, bracket.size = 0, vjust = 1.8)
  
  nF.swarm[[SS2.name]] <- p 
}

nF.swarm[["SS2 Patient 8"]] + nF.swarm[["SS2 Patient 9"]]

#nCount ----
nC.swarm <- vector("list", length = 2)
names(nC.swarm) <- SS2.names

for(i in 1:2){
  SS2 <- SS2s[[i]]
  SS2.name <- SS2.names[[i]]
  
  my_comparisons <- list(
    c(glue("{patient[[i]]}.0"), glue("{patient[[i]]}.1")), 
    c(glue("{patient[[i]]}.1"), glue("{patient[[i]]}.2")), 
    c(glue("{patient[[i]]}.0"), glue("{patient[[i]]}.2"))
  )
  
  numCells <- nrow(SS2@meta.data)
  
  p <- ggplot(SS2@meta.data, aes(x= sample, y = nCount_RNA, color = sample)) +
  geom_quasirandom(groupOnX =TRUE) + 
  theme_bw() + 
  scale_color_manual(values = c("#00AFBB", "#E7B800", "#FC4E07")) + 
    labs(title = glue("{SS2.name} Malignant Cells nCount across sample"), x = SS2.name, subtitle = glue("{numCells} Malignant Cells")) + 
    theme(plot.title = element_text(size =10), plot.subtitle = element_text(size = 8)) + 
  geom_boxplot(width = 0.10, position = position_dodge(0.9), alpha = 1, show.legend = F, color = "black", outlier.alpha = 0) + 
    geom_text(label = paste(sum(str_detect(SS2$sample, "[:digit:].0")), "cells"), x = glue("{patient[[i]]}.0"), y = min(SS2$nCount_RNA) -0.03, color = "black", show.legend = FALSE) + 
    geom_text(label = paste(sum(str_detect(SS2$sample, "[:digit:].1")), "cells"), x = glue("{patient[[i]]}.1"), y = min(SS2$nCount_RNA) - 0.03, color = "black", show.legend = FALSE) + 
    geom_text(label = paste(sum(str_detect(SS2$sample, "[:digit:].2")), "cells"), x = glue("{patient[[i]]}.2"), y = min(SS2$nCount_RNA) - 0.03, color = "black", show.legend = FALSE) + 
    stat_compare_means(comparisons = my_comparisons, method = "wilcox.test", label = "p.format", step.increase = 0.06) + 
    stat_compare_means(comparisons = my_comparisons, method = "wilcox.test", label = "p.signif", step.increase = 0.06, bracket.size = 0, vjust = 1.8)
  
  nC.swarm[[SS2.name]] <- p 
}

nC.swarm[["SS2 Patient 8"]] + nC.swarm[["SS2 Patient 9"]]


```

* **OBSERVATIONS** 
    - 9.1 does not have statistically significantly higher nFeature of nCount. This rejects our hypothesis and it seems like there is no correlation between nFeature/nCount and cell cycle gene expression. 

* **STATISTICAL SUMMARY**
```{r cellcycleStat, message=FALSE, warning=FALSE}
#Patient 8 ---------------------------------
SS2Malignant8.0 <- subset(SS2Malignant.8, subset = (sample == "8.0"))
SS2Malignant8.1 <- subset(SS2Malignant.8, subset = (sample == "8.1"))
SS2Malignant8.2 <- subset(SS2Malignant.8, subset = (sample == "8.2"))

SS28M.sscore.df <- data.frame(
  "p.0vsp.1" = wilcox.test(SS2Malignant8.0$S.Score.centered, SS2Malignant8.1$S.Score.centered)$p.value, 
  "p.1vsp.2" = wilcox.test(SS2Malignant8.1$S.Score.centered, SS2Malignant8.2$S.Score.centered)$p.value, 
  "p.0vsp.2" = wilcox.test(SS2Malignant8.0$S.Score.centered, SS2Malignant8.2$S.Score.centered)$p.value
)
SS28M.g2m.df <- 
  data.frame(
  "p.0vsp.1" = wilcox.test(SS2Malignant8.0$G2M.Score.centered, SS2Malignant8.1$G2M.Score.centered)$p.value, 
  "p.1vsp.2" = wilcox.test(SS2Malignant8.1$G2M.Score.centered, SS2Malignant8.2$G2M.Score.centered)$p.value, 
  "p.0vsp.2" = wilcox.test(SS2Malignant8.0$G2M.Score.centered, SS2Malignant8.2$G2M.Score.centered)$p.value
)

#Patient 9 ---------------------------------
SS2Malignant9.0 <- subset(SS2Malignant.9, subset = (sample == "9.0"))
SS2Malignant9.1 <- subset(SS2Malignant.9, subset = (sample == "9.1"))
SS2Malignant9.2 <- subset(SS2Malignant.9, subset = (sample == "9.2"))

SS29M.sscore.df <- data.frame(
  "p.0vsp.1" = wilcox.test(SS2Malignant9.0$S.Score.centered, SS2Malignant9.1$S.Score.centered)$p.value, 
  "p.1vsp.2" = wilcox.test(SS2Malignant9.1$S.Score.centered, SS2Malignant9.2$S.Score.centered)$p.value, 
  "p.0vsp.2" = wilcox.test(SS2Malignant9.0$S.Score.centered, SS2Malignant9.2$S.Score.centered)$p.value
)
SS29M.g2m.df <- 
  data.frame(
  "p.0vsp.1" = wilcox.test(SS2Malignant9.0$G2M.Score.centered, SS2Malignant9.1$G2M.Score.centered)$p.value, 
  "p.1vsp.2" = wilcox.test(SS2Malignant9.1$G2M.Score.centered, SS2Malignant9.2$G2M.Score.centered)$p.value, 
  "p.0vsp.2" = wilcox.test(SS2Malignant9.0$G2M.Score.centered, SS2Malignant9.2$G2M.Score.centered)$p.value
)

#combine ------------------------------
sscore.DF <- rbind(SS28M.sscore.df, SS29M.sscore.df)
rownames(sscore.DF) <- c("S.P8", "S.P9")
g2m.DF <- rbind(SS28M.g2m.df, SS29M.g2m.df)
rownames(g2m.DF) <- c("g2m.P8", "g2m.P9")
both.DF <- rbind(sscore.DF, g2m.DF)
DT::datatable(both.DF) %>% 
  DT::formatSignif(names(both.DF), digits = 2) %>% 
  DT::formatStyle(names(both.DF), color = DT::styleInterval(0.05, c('red', 'black')))
```

Our results from DE Analysis and Cell Cycle Analysis collectively suggest that **OXPHOS and UPR** expression weakly correlate with treatment condition, while cell cycle phase does not significantly correlate with treatment. 
However, considering the idea that cell cycle might influence the expression of signatures, we are now interested in examining whether there is a correlation between the expression of **OXPHOS and UPR** and **cell cycle phase**. 

* **QUESTION #2** Is there a correlation between cell cycle phase and the expression of signatures? 
* **HYPOTHESIS** We hypothesize that low cycling cells (G1) should express higher levels of **OXPHOS and UPR** genes. (G1 > S > G2M)
    - **APPROACH** Violin Plots
        * Create Violin Plots for each hallmark of interest like **DE Analysis #1/APPROACH #1. 
        * Stratify the Violin Plots by cell cycle phase

```{r cellcycle2, message=FALSE, warning=FALSE, fig.align='center', fig.width=10, fig.height = 8}
hms.centered <- c("GO.OXPHOS35.centered", "UNUPDATED.UPR38.centered")
cc.exp.plot <- vector("list", length = 3)
names(cc.exp.plot) <- SS2.names
patient <- c("8", "9")

for(i in 1:length(SS2s)){
  obj = SS2s[[i]]
  name = SS2.names[[i]]
  numCells = nrow(obj@meta.data)
  obj$Phase <- factor(obj$Phase, levels = c("G1", "S", "G2M"))
  
  my_comparisons <- list(
    c(glue("G1"), glue("S")), 
    c(glue("S"), glue("G2M")), 
    c(glue("G1"), glue("G2M"))
  )
  
  if(patient[[i]] == "8"){
    p1 <- VlnPlot(obj, features = hms.centered, group.by = "Phase", pt.size = 0, combine = FALSE, cols = c("#00AFBB", "#E7B800", "#FC4E07"), y.max = 1)
  }
  else{
    p1 <- VlnPlot(obj, features = hms.centered, group.by = "Phase", pt.size = 0, combine = FALSE, cols = c("#00AFBB", "#E7B800", "#FC4E07"), y.max = 1.3)
  }
  p1[[1]] <- p1[[1]] + labs(title = glue("{name} OXPHOS score by Cell Cycle Phase"), subtitle = glue("{numCells} Malignant Cells")) +
    geom_boxplot(width = 0.15, position = position_dodge(0.9), alpha = 0.3, show.legend = F)+  
    geom_text(label = paste(sum(obj$Phase == "G1"), "cells"), x = "G1", y = min(obj$GO.OXPHOS35.centered) -0.03) + 
    geom_text(label = paste(sum(obj$Phase == "S"), "cells"), x = "S", y = min(obj$GO.OXPHOS35.centered) - 0.03) + 
    geom_text(label = paste(sum(obj$Phase == "G2M"), "cells"), x = "G2M", y = min(obj$GO.OXPHOS35.centered) - 0.03) + 
    theme(plot.title = element_text(size = 10)) + 
    stat_compare_means(comparisons = my_comparisons, method = "wilcox.test", label = "p.format", step.increase = 0.12) + 
    stat_compare_means(comparisons = my_comparisons, method = "wilcox.test", label = "p.signif", step.increase = 0.12, bracket.size = 0, vjust = 1.8)
  
  p1[[2]] <- p1[[2]] + labs(title = glue("{name} UPR score by Cell Cycle Phase"), subtitle = glue("{numCells} Malignant Cells")) +
    theme(plot.title = element_text(size = 10)) + 
    geom_boxplot(width = 0.15, position = position_dodge(0.9), alpha = 0.3, show.legend = F) + 
    geom_text(label = paste(sum(obj$Phase == "G1"), "cells"), x = "G1", y = min(obj$UNUPDATED.UPR38.centered) -0.03) + 
    geom_text(label = paste(sum(obj$Phase == "S"), "cells"), x = "S", y = min(obj$UNUPDATED.UPR38.centered) - 0.03) + 
    geom_text(label = paste(sum(obj$Phase == "G2M"), "cells"), x = "G2M", y = min(obj$UNUPDATED.UPR38.centered) - 0.03) +
    stat_compare_means(comparisons = my_comparisons, method = "wilcox.test", label = "p.format", step.increase = 0.12) + 
    stat_compare_means(comparisons = my_comparisons, method = "wilcox.test", label = "p.signif", step.increase = 0.12, bracket.size = 0, vjust = 1.8)

  
  p2 <- VlnPlot(obj, features = hms.centered, group.by = "sample", split.by = "Phase", pt.size = 0, combine = FALSE, c("#00AFBB", "#FC4E07", "#E7B800"))
  p2[[1]] <- p2[[1]] + labs(title = glue("{name} OXPHOS score by Cell Cycle Phase / Treatment"), subtitle = glue("{numCells} Malignant Cells")) +
    geom_boxplot(width = 0.15, position = position_dodge(0.9), alpha = 0.3, show.legend = F) + 
    theme(plot.title = element_text(size = 10))
  p2[[2]] <- p2[[2]] + labs(title = glue("{name} UPR score by Cell Cycle Phase / Treatment"), subtitle = glue("{numCells} Malignant Cells")) +
    geom_boxplot(width = 0.15, position = position_dodge(0.9), alpha = 0.3, show.legend = F) + 
    theme(plot.title = element_text(size = 10))
  
  cc.exp.plot[[i]] <- p1[[1]] + p1[[2]] + p2[[1]]+ p2[[2]] + plot_layout(guides= 'collect')
}

cc.exp.plot[["SS2 Patient 8"]]
cc.exp.plot[["SS2 Patient 9"]]
```

* **CONCLUSIONS** 
    - **Patient 8**
        * OXPHOS: G2M > G1 > S (not statistically significant & does not support our hypothesis)
        * UPR: G2M > S > G1 (not statistically significant & does not support our hypothesis)
    - **Patient 9** 
        * OXPHOS: G2M > G1; S > G1 (statistically significant but does not support our hypothesis)
        * UPR: S > G2M > G1 (not statistically significant and does not support our hypothesis)
    - No specific statistically significant trend identified. However, cells in the non / low cycling G1 phase do not seem to express higher levels of OXPHOS and UPR genes, which does not support our hypothesis. 
    
    